// @harvest-forward/runtime/dist/src/plugins/npm/NpmBuilder.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NpmBuilder = void 0;
const childProcess = __importStar(require("child_process"));
const AbstractSingularBuilder_1 = require("../../runtime/AbstractSingularBuilder");
const System_1 = require("../../system/System");
const SystemComponent_1 = require("../../system/SystemComponent");
const ValueType_1 = require("../../system/ValueType");
const LogAsyncMethod_1 = require("../../util/LogAsyncMethod");
class NpmBuilder extends AbstractSingularBuilder_1.AbstractSingularBuilder {
    constructor(configurationValues) {
        super('NpmBuilder', 'A builder that executes an npm script', {
            scriptName: {
                name: 'scriptName',
                required: true,
                description: 'The name of the npm script to run',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
        }, configurationValues ? configurationValues : []);
    }
    async buildInternal(__system, __component) {
        const scriptName = this.getConfiguredValue('scriptName');
        childProcess.execSync(`npm run ${scriptName}`);
        return [];
    }
}
__decorate([
    (0, LogAsyncMethod_1.LogAsyncMethod)(),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [System_1.System, SystemComponent_1.SystemComponent]),
    __metadata("design:returntype", Promise)
], NpmBuilder.prototype, "buildInternal", null);
exports.NpmBuilder = NpmBuilder;
