// @harvest-forward/runtime/dist/src/plugins/java-bindings/grammar/Java9ParserListener.js
"use strict";
var antlr4 = require('antlr4/index');
function Java9ParserListener() {
    antlr4.tree.ParseTreeListener.call(this);
    return this;
}
Java9ParserListener.prototype = Object.create(antlr4.tree.ParseTreeListener.prototype);
Java9ParserListener.prototype.constructor = Java9ParserListener;
Java9ParserListener.prototype.enterLiteral = function (ctx) {
};
Java9ParserListener.prototype.exitLiteral = function (ctx) {
};
Java9ParserListener.prototype.enterPrimitiveType = function (ctx) {
};
Java9ParserListener.prototype.exitPrimitiveType = function (ctx) {
};
Java9ParserListener.prototype.enterNumericType = function (ctx) {
};
Java9ParserListener.prototype.exitNumericType = function (ctx) {
};
Java9ParserListener.prototype.enterIntegralType = function (ctx) {
};
Java9ParserListener.prototype.exitIntegralType = function (ctx) {
};
Java9ParserListener.prototype.enterFloatingPointType = function (ctx) {
};
Java9ParserListener.prototype.exitFloatingPointType = function (ctx) {
};
Java9ParserListener.prototype.enterReferenceType = function (ctx) {
};
Java9ParserListener.prototype.exitReferenceType = function (ctx) {
};
Java9ParserListener.prototype.enterClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterClassType = function (ctx) {
};
Java9ParserListener.prototype.exitClassType = function (ctx) {
};
Java9ParserListener.prototype.enterClassType_lf_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitClassType_lf_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterClassType_lfno_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitClassType_lfno_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceType_lf_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceType_lf_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceType_lfno_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceType_lfno_classOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterTypeVariable = function (ctx) {
};
Java9ParserListener.prototype.exitTypeVariable = function (ctx) {
};
Java9ParserListener.prototype.enterArrayType = function (ctx) {
};
Java9ParserListener.prototype.exitArrayType = function (ctx) {
};
Java9ParserListener.prototype.enterDims = function (ctx) {
};
Java9ParserListener.prototype.exitDims = function (ctx) {
};
Java9ParserListener.prototype.enterTypeParameter = function (ctx) {
};
Java9ParserListener.prototype.exitTypeParameter = function (ctx) {
};
Java9ParserListener.prototype.enterTypeParameterModifier = function (ctx) {
};
Java9ParserListener.prototype.exitTypeParameterModifier = function (ctx) {
};
Java9ParserListener.prototype.enterTypeBound = function (ctx) {
};
Java9ParserListener.prototype.exitTypeBound = function (ctx) {
};
Java9ParserListener.prototype.enterAdditionalBound = function (ctx) {
};
Java9ParserListener.prototype.exitAdditionalBound = function (ctx) {
};
Java9ParserListener.prototype.enterTypeArguments = function (ctx) {
};
Java9ParserListener.prototype.exitTypeArguments = function (ctx) {
};
Java9ParserListener.prototype.enterTypeArgumentList = function (ctx) {
};
Java9ParserListener.prototype.exitTypeArgumentList = function (ctx) {
};
Java9ParserListener.prototype.enterTypeArgument = function (ctx) {
};
Java9ParserListener.prototype.exitTypeArgument = function (ctx) {
};
Java9ParserListener.prototype.enterWildcard = function (ctx) {
};
Java9ParserListener.prototype.exitWildcard = function (ctx) {
};
Java9ParserListener.prototype.enterWildcardBounds = function (ctx) {
};
Java9ParserListener.prototype.exitWildcardBounds = function (ctx) {
};
Java9ParserListener.prototype.enterModuleName = function (ctx) {
};
Java9ParserListener.prototype.exitModuleName = function (ctx) {
};
Java9ParserListener.prototype.enterPackageName = function (ctx) {
};
Java9ParserListener.prototype.exitPackageName = function (ctx) {
};
Java9ParserListener.prototype.enterTypeName = function (ctx) {
};
Java9ParserListener.prototype.exitTypeName = function (ctx) {
};
Java9ParserListener.prototype.enterPackageOrTypeName = function (ctx) {
};
Java9ParserListener.prototype.exitPackageOrTypeName = function (ctx) {
};
Java9ParserListener.prototype.enterExpressionName = function (ctx) {
};
Java9ParserListener.prototype.exitExpressionName = function (ctx) {
};
Java9ParserListener.prototype.enterMethodName = function (ctx) {
};
Java9ParserListener.prototype.exitMethodName = function (ctx) {
};
Java9ParserListener.prototype.enterAmbiguousName = function (ctx) {
};
Java9ParserListener.prototype.exitAmbiguousName = function (ctx) {
};
Java9ParserListener.prototype.enterCompilationUnit = function (ctx) {
};
Java9ParserListener.prototype.exitCompilationUnit = function (ctx) {
};
Java9ParserListener.prototype.enterOrdinaryCompilation = function (ctx) {
};
Java9ParserListener.prototype.exitOrdinaryCompilation = function (ctx) {
};
Java9ParserListener.prototype.enterModularCompilation = function (ctx) {
};
Java9ParserListener.prototype.exitModularCompilation = function (ctx) {
};
Java9ParserListener.prototype.enterPackageDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitPackageDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterPackageModifier = function (ctx) {
};
Java9ParserListener.prototype.exitPackageModifier = function (ctx) {
};
Java9ParserListener.prototype.enterImportDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitImportDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterSingleTypeImportDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitSingleTypeImportDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterTypeImportOnDemandDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitTypeImportOnDemandDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterSingleStaticImportDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitSingleStaticImportDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterStaticImportOnDemandDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitStaticImportOnDemandDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterTypeDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitTypeDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterModuleDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitModuleDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterModuleDirective = function (ctx) {
};
Java9ParserListener.prototype.exitModuleDirective = function (ctx) {
};
Java9ParserListener.prototype.enterRequiresModifier = function (ctx) {
};
Java9ParserListener.prototype.exitRequiresModifier = function (ctx) {
};
Java9ParserListener.prototype.enterClassDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitClassDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterNormalClassDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitNormalClassDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterClassModifier = function (ctx) {
};
Java9ParserListener.prototype.exitClassModifier = function (ctx) {
};
Java9ParserListener.prototype.enterTypeParameters = function (ctx) {
};
Java9ParserListener.prototype.exitTypeParameters = function (ctx) {
};
Java9ParserListener.prototype.enterTypeParameterList = function (ctx) {
};
Java9ParserListener.prototype.exitTypeParameterList = function (ctx) {
};
Java9ParserListener.prototype.enterSuperclass = function (ctx) {
};
Java9ParserListener.prototype.exitSuperclass = function (ctx) {
};
Java9ParserListener.prototype.enterSuperinterfaces = function (ctx) {
};
Java9ParserListener.prototype.exitSuperinterfaces = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceTypeList = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceTypeList = function (ctx) {
};
Java9ParserListener.prototype.enterClassBody = function (ctx) {
};
Java9ParserListener.prototype.exitClassBody = function (ctx) {
};
Java9ParserListener.prototype.enterClassBodyDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitClassBodyDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterClassMemberDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitClassMemberDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterFieldDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitFieldDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterFieldModifier = function (ctx) {
};
Java9ParserListener.prototype.exitFieldModifier = function (ctx) {
};
Java9ParserListener.prototype.enterVariableDeclaratorList = function (ctx) {
};
Java9ParserListener.prototype.exitVariableDeclaratorList = function (ctx) {
};
Java9ParserListener.prototype.enterVariableDeclarator = function (ctx) {
};
Java9ParserListener.prototype.exitVariableDeclarator = function (ctx) {
};
Java9ParserListener.prototype.enterVariableDeclaratorId = function (ctx) {
};
Java9ParserListener.prototype.exitVariableDeclaratorId = function (ctx) {
};
Java9ParserListener.prototype.enterVariableInitializer = function (ctx) {
};
Java9ParserListener.prototype.exitVariableInitializer = function (ctx) {
};
Java9ParserListener.prototype.enterUnannType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannPrimitiveType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannPrimitiveType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannReferenceType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannReferenceType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannClassType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannClassType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannClassType_lf_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannClassType_lf_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannClassType_lfno_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannClassType_lfno_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannInterfaceType_lf_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannInterfaceType_lf_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannInterfaceType_lfno_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannInterfaceType_lfno_unannClassOrInterfaceType = function (ctx) {
};
Java9ParserListener.prototype.enterUnannTypeVariable = function (ctx) {
};
Java9ParserListener.prototype.exitUnannTypeVariable = function (ctx) {
};
Java9ParserListener.prototype.enterUnannArrayType = function (ctx) {
};
Java9ParserListener.prototype.exitUnannArrayType = function (ctx) {
};
Java9ParserListener.prototype.enterMethodDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitMethodDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterMethodModifier = function (ctx) {
};
Java9ParserListener.prototype.exitMethodModifier = function (ctx) {
};
Java9ParserListener.prototype.enterMethodHeader = function (ctx) {
};
Java9ParserListener.prototype.exitMethodHeader = function (ctx) {
};
Java9ParserListener.prototype.enterResult = function (ctx) {
};
Java9ParserListener.prototype.exitResult = function (ctx) {
};
Java9ParserListener.prototype.enterMethodDeclarator = function (ctx) {
};
Java9ParserListener.prototype.exitMethodDeclarator = function (ctx) {
};
Java9ParserListener.prototype.enterFormalParameterList = function (ctx) {
};
Java9ParserListener.prototype.exitFormalParameterList = function (ctx) {
};
Java9ParserListener.prototype.enterFormalParameters = function (ctx) {
};
Java9ParserListener.prototype.exitFormalParameters = function (ctx) {
};
Java9ParserListener.prototype.enterFormalParameter = function (ctx) {
};
Java9ParserListener.prototype.exitFormalParameter = function (ctx) {
};
Java9ParserListener.prototype.enterVariableModifier = function (ctx) {
};
Java9ParserListener.prototype.exitVariableModifier = function (ctx) {
};
Java9ParserListener.prototype.enterLastFormalParameter = function (ctx) {
};
Java9ParserListener.prototype.exitLastFormalParameter = function (ctx) {
};
Java9ParserListener.prototype.enterReceiverParameter = function (ctx) {
};
Java9ParserListener.prototype.exitReceiverParameter = function (ctx) {
};
Java9ParserListener.prototype.enterThrows_ = function (ctx) {
};
Java9ParserListener.prototype.exitThrows_ = function (ctx) {
};
Java9ParserListener.prototype.enterExceptionTypeList = function (ctx) {
};
Java9ParserListener.prototype.exitExceptionTypeList = function (ctx) {
};
Java9ParserListener.prototype.enterExceptionType = function (ctx) {
};
Java9ParserListener.prototype.exitExceptionType = function (ctx) {
};
Java9ParserListener.prototype.enterMethodBody = function (ctx) {
};
Java9ParserListener.prototype.exitMethodBody = function (ctx) {
};
Java9ParserListener.prototype.enterInstanceInitializer = function (ctx) {
};
Java9ParserListener.prototype.exitInstanceInitializer = function (ctx) {
};
Java9ParserListener.prototype.enterStaticInitializer = function (ctx) {
};
Java9ParserListener.prototype.exitStaticInitializer = function (ctx) {
};
Java9ParserListener.prototype.enterConstructorDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitConstructorDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterConstructorModifier = function (ctx) {
};
Java9ParserListener.prototype.exitConstructorModifier = function (ctx) {
};
Java9ParserListener.prototype.enterConstructorDeclarator = function (ctx) {
};
Java9ParserListener.prototype.exitConstructorDeclarator = function (ctx) {
};
Java9ParserListener.prototype.enterSimpleTypeName = function (ctx) {
};
Java9ParserListener.prototype.exitSimpleTypeName = function (ctx) {
};
Java9ParserListener.prototype.enterConstructorBody = function (ctx) {
};
Java9ParserListener.prototype.exitConstructorBody = function (ctx) {
};
Java9ParserListener.prototype.enterExplicitConstructorInvocation = function (ctx) {
};
Java9ParserListener.prototype.exitExplicitConstructorInvocation = function (ctx) {
};
Java9ParserListener.prototype.enterEnumDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitEnumDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterEnumBody = function (ctx) {
};
Java9ParserListener.prototype.exitEnumBody = function (ctx) {
};
Java9ParserListener.prototype.enterEnumConstantList = function (ctx) {
};
Java9ParserListener.prototype.exitEnumConstantList = function (ctx) {
};
Java9ParserListener.prototype.enterEnumConstant = function (ctx) {
};
Java9ParserListener.prototype.exitEnumConstant = function (ctx) {
};
Java9ParserListener.prototype.enterEnumConstantModifier = function (ctx) {
};
Java9ParserListener.prototype.exitEnumConstantModifier = function (ctx) {
};
Java9ParserListener.prototype.enterEnumBodyDeclarations = function (ctx) {
};
Java9ParserListener.prototype.exitEnumBodyDeclarations = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterNormalInterfaceDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitNormalInterfaceDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceModifier = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceModifier = function (ctx) {
};
Java9ParserListener.prototype.enterExtendsInterfaces = function (ctx) {
};
Java9ParserListener.prototype.exitExtendsInterfaces = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceBody = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceBody = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceMemberDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceMemberDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterConstantDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitConstantDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterConstantModifier = function (ctx) {
};
Java9ParserListener.prototype.exitConstantModifier = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceMethodDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceMethodDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterInterfaceMethodModifier = function (ctx) {
};
Java9ParserListener.prototype.exitInterfaceMethodModifier = function (ctx) {
};
Java9ParserListener.prototype.enterAnnotationTypeDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitAnnotationTypeDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterAnnotationTypeBody = function (ctx) {
};
Java9ParserListener.prototype.exitAnnotationTypeBody = function (ctx) {
};
Java9ParserListener.prototype.enterAnnotationTypeMemberDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitAnnotationTypeMemberDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterAnnotationTypeElementDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitAnnotationTypeElementDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterAnnotationTypeElementModifier = function (ctx) {
};
Java9ParserListener.prototype.exitAnnotationTypeElementModifier = function (ctx) {
};
Java9ParserListener.prototype.enterDefaultValue = function (ctx) {
};
Java9ParserListener.prototype.exitDefaultValue = function (ctx) {
};
Java9ParserListener.prototype.enterAnnotation = function (ctx) {
};
Java9ParserListener.prototype.exitAnnotation = function (ctx) {
};
Java9ParserListener.prototype.enterNormalAnnotation = function (ctx) {
};
Java9ParserListener.prototype.exitNormalAnnotation = function (ctx) {
};
Java9ParserListener.prototype.enterElementValuePairList = function (ctx) {
};
Java9ParserListener.prototype.exitElementValuePairList = function (ctx) {
};
Java9ParserListener.prototype.enterElementValuePair = function (ctx) {
};
Java9ParserListener.prototype.exitElementValuePair = function (ctx) {
};
Java9ParserListener.prototype.enterElementValue = function (ctx) {
};
Java9ParserListener.prototype.exitElementValue = function (ctx) {
};
Java9ParserListener.prototype.enterElementValueArrayInitializer = function (ctx) {
};
Java9ParserListener.prototype.exitElementValueArrayInitializer = function (ctx) {
};
Java9ParserListener.prototype.enterElementValueList = function (ctx) {
};
Java9ParserListener.prototype.exitElementValueList = function (ctx) {
};
Java9ParserListener.prototype.enterMarkerAnnotation = function (ctx) {
};
Java9ParserListener.prototype.exitMarkerAnnotation = function (ctx) {
};
Java9ParserListener.prototype.enterSingleElementAnnotation = function (ctx) {
};
Java9ParserListener.prototype.exitSingleElementAnnotation = function (ctx) {
};
Java9ParserListener.prototype.enterArrayInitializer = function (ctx) {
};
Java9ParserListener.prototype.exitArrayInitializer = function (ctx) {
};
Java9ParserListener.prototype.enterVariableInitializerList = function (ctx) {
};
Java9ParserListener.prototype.exitVariableInitializerList = function (ctx) {
};
Java9ParserListener.prototype.enterBlock = function (ctx) {
};
Java9ParserListener.prototype.exitBlock = function (ctx) {
};
Java9ParserListener.prototype.enterBlockStatements = function (ctx) {
};
Java9ParserListener.prototype.exitBlockStatements = function (ctx) {
};
Java9ParserListener.prototype.enterBlockStatement = function (ctx) {
};
Java9ParserListener.prototype.exitBlockStatement = function (ctx) {
};
Java9ParserListener.prototype.enterLocalVariableDeclarationStatement = function (ctx) {
};
Java9ParserListener.prototype.exitLocalVariableDeclarationStatement = function (ctx) {
};
Java9ParserListener.prototype.enterLocalVariableDeclaration = function (ctx) {
};
Java9ParserListener.prototype.exitLocalVariableDeclaration = function (ctx) {
};
Java9ParserListener.prototype.enterStatement = function (ctx) {
};
Java9ParserListener.prototype.exitStatement = function (ctx) {
};
Java9ParserListener.prototype.enterStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.exitStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.enterStatementWithoutTrailingSubstatement = function (ctx) {
};
Java9ParserListener.prototype.exitStatementWithoutTrailingSubstatement = function (ctx) {
};
Java9ParserListener.prototype.enterEmptyStatement = function (ctx) {
};
Java9ParserListener.prototype.exitEmptyStatement = function (ctx) {
};
Java9ParserListener.prototype.enterLabeledStatement = function (ctx) {
};
Java9ParserListener.prototype.exitLabeledStatement = function (ctx) {
};
Java9ParserListener.prototype.enterLabeledStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.exitLabeledStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.enterExpressionStatement = function (ctx) {
};
Java9ParserListener.prototype.exitExpressionStatement = function (ctx) {
};
Java9ParserListener.prototype.enterStatementExpression = function (ctx) {
};
Java9ParserListener.prototype.exitStatementExpression = function (ctx) {
};
Java9ParserListener.prototype.enterIfThenStatement = function (ctx) {
};
Java9ParserListener.prototype.exitIfThenStatement = function (ctx) {
};
Java9ParserListener.prototype.enterIfThenElseStatement = function (ctx) {
};
Java9ParserListener.prototype.exitIfThenElseStatement = function (ctx) {
};
Java9ParserListener.prototype.enterIfThenElseStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.exitIfThenElseStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.enterAssertStatement = function (ctx) {
};
Java9ParserListener.prototype.exitAssertStatement = function (ctx) {
};
Java9ParserListener.prototype.enterSwitchStatement = function (ctx) {
};
Java9ParserListener.prototype.exitSwitchStatement = function (ctx) {
};
Java9ParserListener.prototype.enterSwitchBlock = function (ctx) {
};
Java9ParserListener.prototype.exitSwitchBlock = function (ctx) {
};
Java9ParserListener.prototype.enterSwitchBlockStatementGroup = function (ctx) {
};
Java9ParserListener.prototype.exitSwitchBlockStatementGroup = function (ctx) {
};
Java9ParserListener.prototype.enterSwitchLabels = function (ctx) {
};
Java9ParserListener.prototype.exitSwitchLabels = function (ctx) {
};
Java9ParserListener.prototype.enterSwitchLabel = function (ctx) {
};
Java9ParserListener.prototype.exitSwitchLabel = function (ctx) {
};
Java9ParserListener.prototype.enterEnumConstantName = function (ctx) {
};
Java9ParserListener.prototype.exitEnumConstantName = function (ctx) {
};
Java9ParserListener.prototype.enterWhileStatement = function (ctx) {
};
Java9ParserListener.prototype.exitWhileStatement = function (ctx) {
};
Java9ParserListener.prototype.enterWhileStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.exitWhileStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.enterDoStatement = function (ctx) {
};
Java9ParserListener.prototype.exitDoStatement = function (ctx) {
};
Java9ParserListener.prototype.enterForStatement = function (ctx) {
};
Java9ParserListener.prototype.exitForStatement = function (ctx) {
};
Java9ParserListener.prototype.enterForStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.exitForStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.enterBasicForStatement = function (ctx) {
};
Java9ParserListener.prototype.exitBasicForStatement = function (ctx) {
};
Java9ParserListener.prototype.enterBasicForStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.exitBasicForStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.enterForInit = function (ctx) {
};
Java9ParserListener.prototype.exitForInit = function (ctx) {
};
Java9ParserListener.prototype.enterForUpdate = function (ctx) {
};
Java9ParserListener.prototype.exitForUpdate = function (ctx) {
};
Java9ParserListener.prototype.enterStatementExpressionList = function (ctx) {
};
Java9ParserListener.prototype.exitStatementExpressionList = function (ctx) {
};
Java9ParserListener.prototype.enterEnhancedForStatement = function (ctx) {
};
Java9ParserListener.prototype.exitEnhancedForStatement = function (ctx) {
};
Java9ParserListener.prototype.enterEnhancedForStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.exitEnhancedForStatementNoShortIf = function (ctx) {
};
Java9ParserListener.prototype.enterBreakStatement = function (ctx) {
};
Java9ParserListener.prototype.exitBreakStatement = function (ctx) {
};
Java9ParserListener.prototype.enterContinueStatement = function (ctx) {
};
Java9ParserListener.prototype.exitContinueStatement = function (ctx) {
};
Java9ParserListener.prototype.enterReturnStatement = function (ctx) {
};
Java9ParserListener.prototype.exitReturnStatement = function (ctx) {
};
Java9ParserListener.prototype.enterThrowStatement = function (ctx) {
};
Java9ParserListener.prototype.exitThrowStatement = function (ctx) {
};
Java9ParserListener.prototype.enterSynchronizedStatement = function (ctx) {
};
Java9ParserListener.prototype.exitSynchronizedStatement = function (ctx) {
};
Java9ParserListener.prototype.enterTryStatement = function (ctx) {
};
Java9ParserListener.prototype.exitTryStatement = function (ctx) {
};
Java9ParserListener.prototype.enterCatches = function (ctx) {
};
Java9ParserListener.prototype.exitCatches = function (ctx) {
};
Java9ParserListener.prototype.enterCatchClause = function (ctx) {
};
Java9ParserListener.prototype.exitCatchClause = function (ctx) {
};
Java9ParserListener.prototype.enterCatchFormalParameter = function (ctx) {
};
Java9ParserListener.prototype.exitCatchFormalParameter = function (ctx) {
};
Java9ParserListener.prototype.enterCatchType = function (ctx) {
};
Java9ParserListener.prototype.exitCatchType = function (ctx) {
};
Java9ParserListener.prototype.enterFinally_ = function (ctx) {
};
Java9ParserListener.prototype.exitFinally_ = function (ctx) {
};
Java9ParserListener.prototype.enterTryWithResourcesStatement = function (ctx) {
};
Java9ParserListener.prototype.exitTryWithResourcesStatement = function (ctx) {
};
Java9ParserListener.prototype.enterResourceSpecification = function (ctx) {
};
Java9ParserListener.prototype.exitResourceSpecification = function (ctx) {
};
Java9ParserListener.prototype.enterResourceList = function (ctx) {
};
Java9ParserListener.prototype.exitResourceList = function (ctx) {
};
Java9ParserListener.prototype.enterResource = function (ctx) {
};
Java9ParserListener.prototype.exitResource = function (ctx) {
};
Java9ParserListener.prototype.enterVariableAccess = function (ctx) {
};
Java9ParserListener.prototype.exitVariableAccess = function (ctx) {
};
Java9ParserListener.prototype.enterPrimary = function (ctx) {
};
Java9ParserListener.prototype.exitPrimary = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lf_arrayAccess = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lf_arrayAccess = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lfno_arrayAccess = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lfno_arrayAccess = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lf_primary_lf_arrayAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lf_primary_lf_arrayAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lf_primary_lfno_arrayAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lf_primary_lfno_arrayAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lfno_primary_lf_arrayAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lfno_primary_lf_arrayAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterPrimaryNoNewArray_lfno_primary_lfno_arrayAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitPrimaryNoNewArray_lfno_primary_lfno_arrayAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterClassLiteral = function (ctx) {
};
Java9ParserListener.prototype.exitClassLiteral = function (ctx) {
};
Java9ParserListener.prototype.enterClassInstanceCreationExpression = function (ctx) {
};
Java9ParserListener.prototype.exitClassInstanceCreationExpression = function (ctx) {
};
Java9ParserListener.prototype.enterClassInstanceCreationExpression_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitClassInstanceCreationExpression_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterClassInstanceCreationExpression_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitClassInstanceCreationExpression_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterTypeArgumentsOrDiamond = function (ctx) {
};
Java9ParserListener.prototype.exitTypeArgumentsOrDiamond = function (ctx) {
};
Java9ParserListener.prototype.enterFieldAccess = function (ctx) {
};
Java9ParserListener.prototype.exitFieldAccess = function (ctx) {
};
Java9ParserListener.prototype.enterFieldAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitFieldAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterFieldAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitFieldAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterArrayAccess = function (ctx) {
};
Java9ParserListener.prototype.exitArrayAccess = function (ctx) {
};
Java9ParserListener.prototype.enterArrayAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitArrayAccess_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterArrayAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitArrayAccess_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterMethodInvocation = function (ctx) {
};
Java9ParserListener.prototype.exitMethodInvocation = function (ctx) {
};
Java9ParserListener.prototype.enterMethodInvocation_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitMethodInvocation_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterMethodInvocation_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitMethodInvocation_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterArgumentList = function (ctx) {
};
Java9ParserListener.prototype.exitArgumentList = function (ctx) {
};
Java9ParserListener.prototype.enterMethodReference = function (ctx) {
};
Java9ParserListener.prototype.exitMethodReference = function (ctx) {
};
Java9ParserListener.prototype.enterMethodReference_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.exitMethodReference_lf_primary = function (ctx) {
};
Java9ParserListener.prototype.enterMethodReference_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.exitMethodReference_lfno_primary = function (ctx) {
};
Java9ParserListener.prototype.enterArrayCreationExpression = function (ctx) {
};
Java9ParserListener.prototype.exitArrayCreationExpression = function (ctx) {
};
Java9ParserListener.prototype.enterDimExprs = function (ctx) {
};
Java9ParserListener.prototype.exitDimExprs = function (ctx) {
};
Java9ParserListener.prototype.enterDimExpr = function (ctx) {
};
Java9ParserListener.prototype.exitDimExpr = function (ctx) {
};
Java9ParserListener.prototype.enterConstantExpression = function (ctx) {
};
Java9ParserListener.prototype.exitConstantExpression = function (ctx) {
};
Java9ParserListener.prototype.enterExpression = function (ctx) {
};
Java9ParserListener.prototype.exitExpression = function (ctx) {
};
Java9ParserListener.prototype.enterLambdaExpression = function (ctx) {
};
Java9ParserListener.prototype.exitLambdaExpression = function (ctx) {
};
Java9ParserListener.prototype.enterLambdaParameters = function (ctx) {
};
Java9ParserListener.prototype.exitLambdaParameters = function (ctx) {
};
Java9ParserListener.prototype.enterInferredFormalParameterList = function (ctx) {
};
Java9ParserListener.prototype.exitInferredFormalParameterList = function (ctx) {
};
Java9ParserListener.prototype.enterLambdaBody = function (ctx) {
};
Java9ParserListener.prototype.exitLambdaBody = function (ctx) {
};
Java9ParserListener.prototype.enterAssignmentExpression = function (ctx) {
};
Java9ParserListener.prototype.exitAssignmentExpression = function (ctx) {
};
Java9ParserListener.prototype.enterAssignment = function (ctx) {
};
Java9ParserListener.prototype.exitAssignment = function (ctx) {
};
Java9ParserListener.prototype.enterLeftHandSide = function (ctx) {
};
Java9ParserListener.prototype.exitLeftHandSide = function (ctx) {
};
Java9ParserListener.prototype.enterAssignmentOperator = function (ctx) {
};
Java9ParserListener.prototype.exitAssignmentOperator = function (ctx) {
};
Java9ParserListener.prototype.enterConditionalExpression = function (ctx) {
};
Java9ParserListener.prototype.exitConditionalExpression = function (ctx) {
};
Java9ParserListener.prototype.enterConditionalOrExpression = function (ctx) {
};
Java9ParserListener.prototype.exitConditionalOrExpression = function (ctx) {
};
Java9ParserListener.prototype.enterConditionalAndExpression = function (ctx) {
};
Java9ParserListener.prototype.exitConditionalAndExpression = function (ctx) {
};
Java9ParserListener.prototype.enterInclusiveOrExpression = function (ctx) {
};
Java9ParserListener.prototype.exitInclusiveOrExpression = function (ctx) {
};
Java9ParserListener.prototype.enterExclusiveOrExpression = function (ctx) {
};
Java9ParserListener.prototype.exitExclusiveOrExpression = function (ctx) {
};
Java9ParserListener.prototype.enterAndExpression = function (ctx) {
};
Java9ParserListener.prototype.exitAndExpression = function (ctx) {
};
Java9ParserListener.prototype.enterEqualityExpression = function (ctx) {
};
Java9ParserListener.prototype.exitEqualityExpression = function (ctx) {
};
Java9ParserListener.prototype.enterRelationalExpression = function (ctx) {
};
Java9ParserListener.prototype.exitRelationalExpression = function (ctx) {
};
Java9ParserListener.prototype.enterShiftExpression = function (ctx) {
};
Java9ParserListener.prototype.exitShiftExpression = function (ctx) {
};
Java9ParserListener.prototype.enterAdditiveExpression = function (ctx) {
};
Java9ParserListener.prototype.exitAdditiveExpression = function (ctx) {
};
Java9ParserListener.prototype.enterMultiplicativeExpression = function (ctx) {
};
Java9ParserListener.prototype.exitMultiplicativeExpression = function (ctx) {
};
Java9ParserListener.prototype.enterUnaryExpression = function (ctx) {
};
Java9ParserListener.prototype.exitUnaryExpression = function (ctx) {
};
Java9ParserListener.prototype.enterPreIncrementExpression = function (ctx) {
};
Java9ParserListener.prototype.exitPreIncrementExpression = function (ctx) {
};
Java9ParserListener.prototype.enterPreDecrementExpression = function (ctx) {
};
Java9ParserListener.prototype.exitPreDecrementExpression = function (ctx) {
};
Java9ParserListener.prototype.enterUnaryExpressionNotPlusMinus = function (ctx) {
};
Java9ParserListener.prototype.exitUnaryExpressionNotPlusMinus = function (ctx) {
};
Java9ParserListener.prototype.enterPostfixExpression = function (ctx) {
};
Java9ParserListener.prototype.exitPostfixExpression = function (ctx) {
};
Java9ParserListener.prototype.enterPostIncrementExpression = function (ctx) {
};
Java9ParserListener.prototype.exitPostIncrementExpression = function (ctx) {
};
Java9ParserListener.prototype.enterPostIncrementExpression_lf_postfixExpression = function (ctx) {
};
Java9ParserListener.prototype.exitPostIncrementExpression_lf_postfixExpression = function (ctx) {
};
Java9ParserListener.prototype.enterPostDecrementExpression = function (ctx) {
};
Java9ParserListener.prototype.exitPostDecrementExpression = function (ctx) {
};
Java9ParserListener.prototype.enterPostDecrementExpression_lf_postfixExpression = function (ctx) {
};
Java9ParserListener.prototype.exitPostDecrementExpression_lf_postfixExpression = function (ctx) {
};
Java9ParserListener.prototype.enterCastExpression = function (ctx) {
};
Java9ParserListener.prototype.exitCastExpression = function (ctx) {
};
Java9ParserListener.prototype.enterIdentifier = function (ctx) {
};
Java9ParserListener.prototype.exitIdentifier = function (ctx) {
};
exports.Java9ParserListener = Java9ParserListener;
