// @harvest-forward/runtime/dist/src/plugins/information-architecture/information/Enumeration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Enumeration = void 0;
const Permanence_1 = require("../../../system/Permanence");
const SystemComponentComparator_1 = require("../../../system/SystemComponentComparator");
const ValueType_1 = require("../../../system/ValueType");
const EnumeratedValue_1 = require("./EnumeratedValue");
class Enumeration extends ValueType_1.ValueType {
    constructor(nameSpace, name, description, values) {
        super('Enumeration', nameSpace, name, description, false);
        this.isEnumeration = true;
        this.permanence = Permanence_1.Permanence.transient;
        this._values = {};
        values.forEach((value) => {
            this.addChild(value);
        });
    }
    get valuesMap() {
        return this._values;
    }
    get values() {
        return Object.values(this._values).sort(SystemComponentComparator_1.SystemComponentComparator);
    }
    addChild(child) {
        super.addChild(child);
        if (child instanceof EnumeratedValue_1.EnumeratedValue) {
            this._values[child.name] = child;
        }
    }
    removeChild(child) {
        if (child instanceof EnumeratedValue_1.EnumeratedValue) {
            delete this._values[child.name];
        }
        return super.removeChild(child);
    }
}
exports.Enumeration = Enumeration;
