// @harvest-forward/runtime/dist/src/plugins/information-architecture/information/ObjectTypeDataMember.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectTypeDataMember = void 0;
const SystemComponentType_1 = require("../../../system/SystemComponentType");
const validation_1 = require("../../validation");
const ObjectTypeMember_1 = require("./ObjectTypeMember");
class ObjectTypeDataMember extends ObjectTypeMember_1.ObjectTypeMember {
    constructor(typeName, nameSpace, name, description, type, length) {
        super(typeName, nameSpace, name, description, SystemComponentType_1.SystemComponentType.informationModel);
        this.optional = false;
        this.significantDigits = 0;
        this.type = type;
        this.length = length;
    }
    addChild(child) {
        super.addChild(child);
        if (child instanceof validation_1.AbstractRule) {
            this.rules.push(child);
        }
    }
    removeChild(child) {
        if (child instanceof validation_1.AbstractRule) {
            const index = this.rules.indexOf(child);
            this.rules.splice(index, 1);
        }
        return super.removeChild(child);
    }
    get allReferences() {
        let referent = this.type;
        while (referent.isCollection && referent.collectedType != null) {
            referent = referent.collectedType;
        }
        return [referent];
    }
    get classTypeField() {
        const mapping = this.orMapping;
        return mapping.discriminator ? true : false;
    }
}
exports.ObjectTypeDataMember = ObjectTypeDataMember;
