// @harvest-forward/runtime/dist/src/plugins/csharp-bindings/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.templates = exports.typeResolvers = void 0;
const fs_1 = __importDefault(require("fs"));
const ProgrammingLanguage_1 = require("../../system/ProgrammingLanguage");
const SimpleMappedValueTypeResolver_1 = require("../../system/SimpleMappedValueTypeResolver");
const ValueType_1 = require("../../system/ValueType");
const CSharpMandatoryValueTypes = {};
CSharpMandatoryValueTypes[ValueType_1.ValueType.OBJECT.name] = 'Object';
CSharpMandatoryValueTypes[ValueType_1.ValueType.STRING.name] = 'string';
CSharpMandatoryValueTypes[ValueType_1.ValueType.INT.name] = 'long';
CSharpMandatoryValueTypes[ValueType_1.ValueType.FLOAT.name] = 'double';
CSharpMandatoryValueTypes[ValueType_1.ValueType.BOOLEAN.name] = 'bool';
CSharpMandatoryValueTypes[ValueType_1.ValueType.DATE_TIME.name] = 'DateTime';
CSharpMandatoryValueTypes[ValueType_1.ValueType.DATE.name] = 'DateTime';
CSharpMandatoryValueTypes[ValueType_1.ValueType.TIME.name] = 'DateTime';
CSharpMandatoryValueTypes[ValueType_1.ValueType.INTERVAL.name] = 'string';
CSharpMandatoryValueTypes[ValueType_1.ValueType.DECIMAL.name] = 'decimal';
CSharpMandatoryValueTypes[ValueType_1.ValueType.VOID.name] = 'void';
const CSharpOptionalValueTypes = {};
CSharpOptionalValueTypes[ValueType_1.ValueType.OBJECT.name] = 'Object?';
CSharpOptionalValueTypes[ValueType_1.ValueType.STRING.name] = 'string?';
CSharpOptionalValueTypes[ValueType_1.ValueType.INT.name] = 'long?';
CSharpOptionalValueTypes[ValueType_1.ValueType.FLOAT.name] = 'double?';
CSharpOptionalValueTypes[ValueType_1.ValueType.BOOLEAN.name] = 'bool?';
CSharpOptionalValueTypes[ValueType_1.ValueType.DATE_TIME.name] = 'DateTime?';
CSharpOptionalValueTypes[ValueType_1.ValueType.DATE.name] = 'DateTime?';
CSharpOptionalValueTypes[ValueType_1.ValueType.TIME.name] = 'DateTime?';
CSharpOptionalValueTypes[ValueType_1.ValueType.INTERVAL.name] = 'string?';
CSharpOptionalValueTypes[ValueType_1.ValueType.DECIMAL.name] = 'decimal?';
CSharpOptionalValueTypes[ValueType_1.ValueType.VOID.name] = 'void';
exports.typeResolvers = {
    CSharp: new SimpleMappedValueTypeResolver_1.SimpleMappedValueTypeResolver(ProgrammingLanguage_1.ProgrammingLanguage.cSharp, CSharpMandatoryValueTypes, CSharpOptionalValueTypes),
};
exports.templates = {
    Enum: { type: 'handlebars', contents: () => fs_1.default.readFileSync(`${__dirname}/Enum.handlebars`).toString() },
    Model: { type: 'handlebars', contents: () => fs_1.default.readFileSync(`${__dirname}/Model.handlebars`).toString() },
    Model_generated: { type: 'handlebars', contents: () => fs_1.default.readFileSync(`${__dirname}/Model.generated.handlebars`).toString() },
};
