// @harvest-forward/runtime/dist/src/plugins/mysql/grammar/MySqlParserListener.js
"use strict";
var antlr4 = require('antlr4/index');
function MySqlParserListener() {
    antlr4.tree.ParseTreeListener.call(this);
    return this;
}
MySqlParserListener.prototype = Object.create(antlr4.tree.ParseTreeListener.prototype);
MySqlParserListener.prototype.constructor = MySqlParserListener;
MySqlParserListener.prototype.enterRoot = function (ctx) {
};
MySqlParserListener.prototype.exitRoot = function (ctx) {
};
MySqlParserListener.prototype.enterSqlStatements = function (ctx) {
};
MySqlParserListener.prototype.exitSqlStatements = function (ctx) {
};
MySqlParserListener.prototype.enterSqlStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSqlStatement = function (ctx) {
};
MySqlParserListener.prototype.enterEmptyStatement = function (ctx) {
};
MySqlParserListener.prototype.exitEmptyStatement = function (ctx) {
};
MySqlParserListener.prototype.enterDdlStatement = function (ctx) {
};
MySqlParserListener.prototype.exitDdlStatement = function (ctx) {
};
MySqlParserListener.prototype.enterDmlStatement = function (ctx) {
};
MySqlParserListener.prototype.exitDmlStatement = function (ctx) {
};
MySqlParserListener.prototype.enterTransactionStatement = function (ctx) {
};
MySqlParserListener.prototype.exitTransactionStatement = function (ctx) {
};
MySqlParserListener.prototype.enterReplicationStatement = function (ctx) {
};
MySqlParserListener.prototype.exitReplicationStatement = function (ctx) {
};
MySqlParserListener.prototype.enterPreparedStatement = function (ctx) {
};
MySqlParserListener.prototype.exitPreparedStatement = function (ctx) {
};
MySqlParserListener.prototype.enterCompoundStatement = function (ctx) {
};
MySqlParserListener.prototype.exitCompoundStatement = function (ctx) {
};
MySqlParserListener.prototype.enterAdministrationStatement = function (ctx) {
};
MySqlParserListener.prototype.exitAdministrationStatement = function (ctx) {
};
MySqlParserListener.prototype.enterUtilityStatement = function (ctx) {
};
MySqlParserListener.prototype.exitUtilityStatement = function (ctx) {
};
MySqlParserListener.prototype.enterCreateDatabase = function (ctx) {
};
MySqlParserListener.prototype.exitCreateDatabase = function (ctx) {
};
MySqlParserListener.prototype.enterCreateEvent = function (ctx) {
};
MySqlParserListener.prototype.exitCreateEvent = function (ctx) {
};
MySqlParserListener.prototype.enterCreateIndex = function (ctx) {
};
MySqlParserListener.prototype.exitCreateIndex = function (ctx) {
};
MySqlParserListener.prototype.enterCreateLogfileGroup = function (ctx) {
};
MySqlParserListener.prototype.exitCreateLogfileGroup = function (ctx) {
};
MySqlParserListener.prototype.enterCreateProcedure = function (ctx) {
};
MySqlParserListener.prototype.exitCreateProcedure = function (ctx) {
};
MySqlParserListener.prototype.enterCreateFunction = function (ctx) {
};
MySqlParserListener.prototype.exitCreateFunction = function (ctx) {
};
MySqlParserListener.prototype.enterCreateServer = function (ctx) {
};
MySqlParserListener.prototype.exitCreateServer = function (ctx) {
};
MySqlParserListener.prototype.enterCopyCreateTable = function (ctx) {
};
MySqlParserListener.prototype.exitCopyCreateTable = function (ctx) {
};
MySqlParserListener.prototype.enterQueryCreateTable = function (ctx) {
};
MySqlParserListener.prototype.exitQueryCreateTable = function (ctx) {
};
MySqlParserListener.prototype.enterColumnCreateTable = function (ctx) {
};
MySqlParserListener.prototype.exitColumnCreateTable = function (ctx) {
};
MySqlParserListener.prototype.enterCreateTablespaceInnodb = function (ctx) {
};
MySqlParserListener.prototype.exitCreateTablespaceInnodb = function (ctx) {
};
MySqlParserListener.prototype.enterCreateTablespaceNdb = function (ctx) {
};
MySqlParserListener.prototype.exitCreateTablespaceNdb = function (ctx) {
};
MySqlParserListener.prototype.enterCreateTrigger = function (ctx) {
};
MySqlParserListener.prototype.exitCreateTrigger = function (ctx) {
};
MySqlParserListener.prototype.enterCreateView = function (ctx) {
};
MySqlParserListener.prototype.exitCreateView = function (ctx) {
};
MySqlParserListener.prototype.enterCreateDatabaseOption = function (ctx) {
};
MySqlParserListener.prototype.exitCreateDatabaseOption = function (ctx) {
};
MySqlParserListener.prototype.enterOwnerStatement = function (ctx) {
};
MySqlParserListener.prototype.exitOwnerStatement = function (ctx) {
};
MySqlParserListener.prototype.enterPreciseSchedule = function (ctx) {
};
MySqlParserListener.prototype.exitPreciseSchedule = function (ctx) {
};
MySqlParserListener.prototype.enterIntervalSchedule = function (ctx) {
};
MySqlParserListener.prototype.exitIntervalSchedule = function (ctx) {
};
MySqlParserListener.prototype.enterTimestampValue = function (ctx) {
};
MySqlParserListener.prototype.exitTimestampValue = function (ctx) {
};
MySqlParserListener.prototype.enterIntervalExpr = function (ctx) {
};
MySqlParserListener.prototype.exitIntervalExpr = function (ctx) {
};
MySqlParserListener.prototype.enterIntervalType = function (ctx) {
};
MySqlParserListener.prototype.exitIntervalType = function (ctx) {
};
MySqlParserListener.prototype.enterEnableType = function (ctx) {
};
MySqlParserListener.prototype.exitEnableType = function (ctx) {
};
MySqlParserListener.prototype.enterIndexType = function (ctx) {
};
MySqlParserListener.prototype.exitIndexType = function (ctx) {
};
MySqlParserListener.prototype.enterIndexOption = function (ctx) {
};
MySqlParserListener.prototype.exitIndexOption = function (ctx) {
};
MySqlParserListener.prototype.enterProcedureParameter = function (ctx) {
};
MySqlParserListener.prototype.exitProcedureParameter = function (ctx) {
};
MySqlParserListener.prototype.enterFunctionParameter = function (ctx) {
};
MySqlParserListener.prototype.exitFunctionParameter = function (ctx) {
};
MySqlParserListener.prototype.enterRoutineComment = function (ctx) {
};
MySqlParserListener.prototype.exitRoutineComment = function (ctx) {
};
MySqlParserListener.prototype.enterRoutineLanguage = function (ctx) {
};
MySqlParserListener.prototype.exitRoutineLanguage = function (ctx) {
};
MySqlParserListener.prototype.enterRoutineBehavior = function (ctx) {
};
MySqlParserListener.prototype.exitRoutineBehavior = function (ctx) {
};
MySqlParserListener.prototype.enterRoutineData = function (ctx) {
};
MySqlParserListener.prototype.exitRoutineData = function (ctx) {
};
MySqlParserListener.prototype.enterRoutineSecurity = function (ctx) {
};
MySqlParserListener.prototype.exitRoutineSecurity = function (ctx) {
};
MySqlParserListener.prototype.enterServerOption = function (ctx) {
};
MySqlParserListener.prototype.exitServerOption = function (ctx) {
};
MySqlParserListener.prototype.enterCreateDefinitions = function (ctx) {
};
MySqlParserListener.prototype.exitCreateDefinitions = function (ctx) {
};
MySqlParserListener.prototype.enterColumnDeclaration = function (ctx) {
};
MySqlParserListener.prototype.exitColumnDeclaration = function (ctx) {
};
MySqlParserListener.prototype.enterConstraintDeclaration = function (ctx) {
};
MySqlParserListener.prototype.exitConstraintDeclaration = function (ctx) {
};
MySqlParserListener.prototype.enterIndexDeclaration = function (ctx) {
};
MySqlParserListener.prototype.exitIndexDeclaration = function (ctx) {
};
MySqlParserListener.prototype.enterColumnDefinition = function (ctx) {
};
MySqlParserListener.prototype.exitColumnDefinition = function (ctx) {
};
MySqlParserListener.prototype.enterNullColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitNullColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterDefaultColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitDefaultColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterAutoIncrementColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitAutoIncrementColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterPrimaryKeyColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitPrimaryKeyColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterUniqueKeyColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitUniqueKeyColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterCommentColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitCommentColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterFormatColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitFormatColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterStorageColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitStorageColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterReferenceColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitReferenceColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterCollateColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitCollateColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterGeneratedColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitGeneratedColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterSerialDefaultColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitSerialDefaultColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterCheckColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitCheckColumnConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterPrimaryKeyTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitPrimaryKeyTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterUniqueKeyTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitUniqueKeyTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterForeignKeyTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitForeignKeyTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterCheckTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitCheckTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterReferenceDefinition = function (ctx) {
};
MySqlParserListener.prototype.exitReferenceDefinition = function (ctx) {
};
MySqlParserListener.prototype.enterReferenceAction = function (ctx) {
};
MySqlParserListener.prototype.exitReferenceAction = function (ctx) {
};
MySqlParserListener.prototype.enterReferenceControlType = function (ctx) {
};
MySqlParserListener.prototype.exitReferenceControlType = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleIndexDeclaration = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleIndexDeclaration = function (ctx) {
};
MySqlParserListener.prototype.enterSpecialIndexDeclaration = function (ctx) {
};
MySqlParserListener.prototype.exitSpecialIndexDeclaration = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionEngine = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionEngine = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionAutoIncrement = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionAutoIncrement = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionAverage = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionAverage = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionCharset = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionCharset = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionChecksum = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionChecksum = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionCollate = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionCollate = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionComment = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionComment = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionCompression = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionCompression = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionConnection = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionConnection = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionDataDirectory = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionDataDirectory = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionDelay = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionDelay = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionEncryption = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionEncryption = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionIndexDirectory = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionIndexDirectory = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionInsertMethod = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionInsertMethod = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionKeyBlockSize = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionKeyBlockSize = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionMaxRows = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionMaxRows = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionMinRows = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionMinRows = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionPackKeys = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionPackKeys = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionPassword = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionPassword = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionRowFormat = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionRowFormat = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionRecalculation = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionRecalculation = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionPersistent = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionPersistent = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionSamplePage = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionSamplePage = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionTablespace = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionTablespace = function (ctx) {
};
MySqlParserListener.prototype.enterTableOptionUnion = function (ctx) {
};
MySqlParserListener.prototype.exitTableOptionUnion = function (ctx) {
};
MySqlParserListener.prototype.enterTablespaceStorage = function (ctx) {
};
MySqlParserListener.prototype.exitTablespaceStorage = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionDefinitions = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionDefinitions = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionFunctionHash = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionFunctionHash = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionFunctionKey = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionFunctionKey = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionFunctionRange = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionFunctionRange = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionFunctionList = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionFunctionList = function (ctx) {
};
MySqlParserListener.prototype.enterSubPartitionFunctionHash = function (ctx) {
};
MySqlParserListener.prototype.exitSubPartitionFunctionHash = function (ctx) {
};
MySqlParserListener.prototype.enterSubPartitionFunctionKey = function (ctx) {
};
MySqlParserListener.prototype.exitSubPartitionFunctionKey = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionComparision = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionComparision = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionListAtom = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionListAtom = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionListVector = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionListVector = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionSimple = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionSimple = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionDefinerAtom = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionDefinerAtom = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionDefinerVector = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionDefinerVector = function (ctx) {
};
MySqlParserListener.prototype.enterSubpartitionDefinition = function (ctx) {
};
MySqlParserListener.prototype.exitSubpartitionDefinition = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionEngine = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionEngine = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionComment = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionComment = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionDataDirectory = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionDataDirectory = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionIndexDirectory = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionIndexDirectory = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionMaxRows = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionMaxRows = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionMinRows = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionMinRows = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionTablespace = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionTablespace = function (ctx) {
};
MySqlParserListener.prototype.enterPartitionOptionNodeGroup = function (ctx) {
};
MySqlParserListener.prototype.exitPartitionOptionNodeGroup = function (ctx) {
};
MySqlParserListener.prototype.enterAlterSimpleDatabase = function (ctx) {
};
MySqlParserListener.prototype.exitAlterSimpleDatabase = function (ctx) {
};
MySqlParserListener.prototype.enterAlterUpgradeName = function (ctx) {
};
MySqlParserListener.prototype.exitAlterUpgradeName = function (ctx) {
};
MySqlParserListener.prototype.enterAlterEvent = function (ctx) {
};
MySqlParserListener.prototype.exitAlterEvent = function (ctx) {
};
MySqlParserListener.prototype.enterAlterFunction = function (ctx) {
};
MySqlParserListener.prototype.exitAlterFunction = function (ctx) {
};
MySqlParserListener.prototype.enterAlterInstance = function (ctx) {
};
MySqlParserListener.prototype.exitAlterInstance = function (ctx) {
};
MySqlParserListener.prototype.enterAlterLogfileGroup = function (ctx) {
};
MySqlParserListener.prototype.exitAlterLogfileGroup = function (ctx) {
};
MySqlParserListener.prototype.enterAlterProcedure = function (ctx) {
};
MySqlParserListener.prototype.exitAlterProcedure = function (ctx) {
};
MySqlParserListener.prototype.enterAlterServer = function (ctx) {
};
MySqlParserListener.prototype.exitAlterServer = function (ctx) {
};
MySqlParserListener.prototype.enterAlterTable = function (ctx) {
};
MySqlParserListener.prototype.exitAlterTable = function (ctx) {
};
MySqlParserListener.prototype.enterAlterTablespace = function (ctx) {
};
MySqlParserListener.prototype.exitAlterTablespace = function (ctx) {
};
MySqlParserListener.prototype.enterAlterView = function (ctx) {
};
MySqlParserListener.prototype.exitAlterView = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByTableOption = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByTableOption = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddColumn = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddColumn = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddColumns = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddColumns = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddIndex = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddIndex = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddPrimaryKey = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddPrimaryKey = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddUniqueKey = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddUniqueKey = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddSpecialIndex = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddSpecialIndex = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddForeignKey = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddForeignKey = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddCheckTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddCheckTableConstraint = function (ctx) {
};
MySqlParserListener.prototype.enterAlterBySetAlgorithm = function (ctx) {
};
MySqlParserListener.prototype.exitAlterBySetAlgorithm = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByChangeDefault = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByChangeDefault = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByChangeColumn = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByChangeColumn = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByRenameColumn = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByRenameColumn = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByLock = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByLock = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByModifyColumn = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByModifyColumn = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDropColumn = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDropColumn = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDropConstraintCheck = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDropConstraintCheck = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDropPrimaryKey = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDropPrimaryKey = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByRenameIndex = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByRenameIndex = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAlterIndexVisibility = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAlterIndexVisibility = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDropIndex = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDropIndex = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDropForeignKey = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDropForeignKey = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDisableKeys = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDisableKeys = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByEnableKeys = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByEnableKeys = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByRename = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByRename = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByOrder = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByOrder = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByConvertCharset = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByConvertCharset = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDefaultCharset = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDefaultCharset = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDiscardTablespace = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDiscardTablespace = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByImportTablespace = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByImportTablespace = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByForce = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByForce = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByValidate = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByValidate = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAddPartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAddPartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDropPartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDropPartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByDiscardPartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByDiscardPartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByImportPartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByImportPartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByTruncatePartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByTruncatePartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByCoalescePartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByCoalescePartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByReorganizePartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByReorganizePartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByExchangePartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByExchangePartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByAnalyzePartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByAnalyzePartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByCheckPartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByCheckPartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByOptimizePartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByOptimizePartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByRebuildPartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByRebuildPartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByRepairPartition = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByRepairPartition = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByRemovePartitioning = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByRemovePartitioning = function (ctx) {
};
MySqlParserListener.prototype.enterAlterByUpgradePartitioning = function (ctx) {
};
MySqlParserListener.prototype.exitAlterByUpgradePartitioning = function (ctx) {
};
MySqlParserListener.prototype.enterDropDatabase = function (ctx) {
};
MySqlParserListener.prototype.exitDropDatabase = function (ctx) {
};
MySqlParserListener.prototype.enterDropEvent = function (ctx) {
};
MySqlParserListener.prototype.exitDropEvent = function (ctx) {
};
MySqlParserListener.prototype.enterDropIndex = function (ctx) {
};
MySqlParserListener.prototype.exitDropIndex = function (ctx) {
};
MySqlParserListener.prototype.enterDropLogfileGroup = function (ctx) {
};
MySqlParserListener.prototype.exitDropLogfileGroup = function (ctx) {
};
MySqlParserListener.prototype.enterDropProcedure = function (ctx) {
};
MySqlParserListener.prototype.exitDropProcedure = function (ctx) {
};
MySqlParserListener.prototype.enterDropFunction = function (ctx) {
};
MySqlParserListener.prototype.exitDropFunction = function (ctx) {
};
MySqlParserListener.prototype.enterDropServer = function (ctx) {
};
MySqlParserListener.prototype.exitDropServer = function (ctx) {
};
MySqlParserListener.prototype.enterDropTable = function (ctx) {
};
MySqlParserListener.prototype.exitDropTable = function (ctx) {
};
MySqlParserListener.prototype.enterDropTablespace = function (ctx) {
};
MySqlParserListener.prototype.exitDropTablespace = function (ctx) {
};
MySqlParserListener.prototype.enterDropTrigger = function (ctx) {
};
MySqlParserListener.prototype.exitDropTrigger = function (ctx) {
};
MySqlParserListener.prototype.enterDropView = function (ctx) {
};
MySqlParserListener.prototype.exitDropView = function (ctx) {
};
MySqlParserListener.prototype.enterRenameTable = function (ctx) {
};
MySqlParserListener.prototype.exitRenameTable = function (ctx) {
};
MySqlParserListener.prototype.enterRenameTableClause = function (ctx) {
};
MySqlParserListener.prototype.exitRenameTableClause = function (ctx) {
};
MySqlParserListener.prototype.enterTruncateTable = function (ctx) {
};
MySqlParserListener.prototype.exitTruncateTable = function (ctx) {
};
MySqlParserListener.prototype.enterCallStatement = function (ctx) {
};
MySqlParserListener.prototype.exitCallStatement = function (ctx) {
};
MySqlParserListener.prototype.enterDeleteStatement = function (ctx) {
};
MySqlParserListener.prototype.exitDeleteStatement = function (ctx) {
};
MySqlParserListener.prototype.enterDoStatement = function (ctx) {
};
MySqlParserListener.prototype.exitDoStatement = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerStatement = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerStatement = function (ctx) {
};
MySqlParserListener.prototype.enterInsertStatement = function (ctx) {
};
MySqlParserListener.prototype.exitInsertStatement = function (ctx) {
};
MySqlParserListener.prototype.enterLoadDataStatement = function (ctx) {
};
MySqlParserListener.prototype.exitLoadDataStatement = function (ctx) {
};
MySqlParserListener.prototype.enterLoadXmlStatement = function (ctx) {
};
MySqlParserListener.prototype.exitLoadXmlStatement = function (ctx) {
};
MySqlParserListener.prototype.enterReplaceStatement = function (ctx) {
};
MySqlParserListener.prototype.exitReplaceStatement = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleSelect = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleSelect = function (ctx) {
};
MySqlParserListener.prototype.enterParenthesisSelect = function (ctx) {
};
MySqlParserListener.prototype.exitParenthesisSelect = function (ctx) {
};
MySqlParserListener.prototype.enterUnionSelect = function (ctx) {
};
MySqlParserListener.prototype.exitUnionSelect = function (ctx) {
};
MySqlParserListener.prototype.enterUnionParenthesisSelect = function (ctx) {
};
MySqlParserListener.prototype.exitUnionParenthesisSelect = function (ctx) {
};
MySqlParserListener.prototype.enterUpdateStatement = function (ctx) {
};
MySqlParserListener.prototype.exitUpdateStatement = function (ctx) {
};
MySqlParserListener.prototype.enterInsertStatementValue = function (ctx) {
};
MySqlParserListener.prototype.exitInsertStatementValue = function (ctx) {
};
MySqlParserListener.prototype.enterUpdatedElement = function (ctx) {
};
MySqlParserListener.prototype.exitUpdatedElement = function (ctx) {
};
MySqlParserListener.prototype.enterAssignmentField = function (ctx) {
};
MySqlParserListener.prototype.exitAssignmentField = function (ctx) {
};
MySqlParserListener.prototype.enterLockClause = function (ctx) {
};
MySqlParserListener.prototype.exitLockClause = function (ctx) {
};
MySqlParserListener.prototype.enterSingleDeleteStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSingleDeleteStatement = function (ctx) {
};
MySqlParserListener.prototype.enterMultipleDeleteStatement = function (ctx) {
};
MySqlParserListener.prototype.exitMultipleDeleteStatement = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerOpenStatement = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerOpenStatement = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerReadIndexStatement = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerReadIndexStatement = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerReadStatement = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerReadStatement = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerCloseStatement = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerCloseStatement = function (ctx) {
};
MySqlParserListener.prototype.enterSingleUpdateStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSingleUpdateStatement = function (ctx) {
};
MySqlParserListener.prototype.enterMultipleUpdateStatement = function (ctx) {
};
MySqlParserListener.prototype.exitMultipleUpdateStatement = function (ctx) {
};
MySqlParserListener.prototype.enterOrderByClause = function (ctx) {
};
MySqlParserListener.prototype.exitOrderByClause = function (ctx) {
};
MySqlParserListener.prototype.enterOrderByExpression = function (ctx) {
};
MySqlParserListener.prototype.exitOrderByExpression = function (ctx) {
};
MySqlParserListener.prototype.enterTableSources = function (ctx) {
};
MySqlParserListener.prototype.exitTableSources = function (ctx) {
};
MySqlParserListener.prototype.enterTableSourceBase = function (ctx) {
};
MySqlParserListener.prototype.exitTableSourceBase = function (ctx) {
};
MySqlParserListener.prototype.enterTableSourceNested = function (ctx) {
};
MySqlParserListener.prototype.exitTableSourceNested = function (ctx) {
};
MySqlParserListener.prototype.enterAtomTableItem = function (ctx) {
};
MySqlParserListener.prototype.exitAtomTableItem = function (ctx) {
};
MySqlParserListener.prototype.enterSubqueryTableItem = function (ctx) {
};
MySqlParserListener.prototype.exitSubqueryTableItem = function (ctx) {
};
MySqlParserListener.prototype.enterTableSourcesItem = function (ctx) {
};
MySqlParserListener.prototype.exitTableSourcesItem = function (ctx) {
};
MySqlParserListener.prototype.enterIndexHint = function (ctx) {
};
MySqlParserListener.prototype.exitIndexHint = function (ctx) {
};
MySqlParserListener.prototype.enterIndexHintType = function (ctx) {
};
MySqlParserListener.prototype.exitIndexHintType = function (ctx) {
};
MySqlParserListener.prototype.enterInnerJoin = function (ctx) {
};
MySqlParserListener.prototype.exitInnerJoin = function (ctx) {
};
MySqlParserListener.prototype.enterStraightJoin = function (ctx) {
};
MySqlParserListener.prototype.exitStraightJoin = function (ctx) {
};
MySqlParserListener.prototype.enterOuterJoin = function (ctx) {
};
MySqlParserListener.prototype.exitOuterJoin = function (ctx) {
};
MySqlParserListener.prototype.enterNaturalJoin = function (ctx) {
};
MySqlParserListener.prototype.exitNaturalJoin = function (ctx) {
};
MySqlParserListener.prototype.enterQueryExpression = function (ctx) {
};
MySqlParserListener.prototype.exitQueryExpression = function (ctx) {
};
MySqlParserListener.prototype.enterQueryExpressionNointo = function (ctx) {
};
MySqlParserListener.prototype.exitQueryExpressionNointo = function (ctx) {
};
MySqlParserListener.prototype.enterQuerySpecification = function (ctx) {
};
MySqlParserListener.prototype.exitQuerySpecification = function (ctx) {
};
MySqlParserListener.prototype.enterQuerySpecificationNointo = function (ctx) {
};
MySqlParserListener.prototype.exitQuerySpecificationNointo = function (ctx) {
};
MySqlParserListener.prototype.enterUnionParenthesis = function (ctx) {
};
MySqlParserListener.prototype.exitUnionParenthesis = function (ctx) {
};
MySqlParserListener.prototype.enterUnionStatement = function (ctx) {
};
MySqlParserListener.prototype.exitUnionStatement = function (ctx) {
};
MySqlParserListener.prototype.enterSelectSpec = function (ctx) {
};
MySqlParserListener.prototype.exitSelectSpec = function (ctx) {
};
MySqlParserListener.prototype.enterSelectElements = function (ctx) {
};
MySqlParserListener.prototype.exitSelectElements = function (ctx) {
};
MySqlParserListener.prototype.enterSelectStarElement = function (ctx) {
};
MySqlParserListener.prototype.exitSelectStarElement = function (ctx) {
};
MySqlParserListener.prototype.enterSelectColumnElement = function (ctx) {
};
MySqlParserListener.prototype.exitSelectColumnElement = function (ctx) {
};
MySqlParserListener.prototype.enterSelectFunctionElement = function (ctx) {
};
MySqlParserListener.prototype.exitSelectFunctionElement = function (ctx) {
};
MySqlParserListener.prototype.enterSelectExpressionElement = function (ctx) {
};
MySqlParserListener.prototype.exitSelectExpressionElement = function (ctx) {
};
MySqlParserListener.prototype.enterSelectIntoVariables = function (ctx) {
};
MySqlParserListener.prototype.exitSelectIntoVariables = function (ctx) {
};
MySqlParserListener.prototype.enterSelectIntoDumpFile = function (ctx) {
};
MySqlParserListener.prototype.exitSelectIntoDumpFile = function (ctx) {
};
MySqlParserListener.prototype.enterSelectIntoTextFile = function (ctx) {
};
MySqlParserListener.prototype.exitSelectIntoTextFile = function (ctx) {
};
MySqlParserListener.prototype.enterSelectFieldsInto = function (ctx) {
};
MySqlParserListener.prototype.exitSelectFieldsInto = function (ctx) {
};
MySqlParserListener.prototype.enterSelectLinesInto = function (ctx) {
};
MySqlParserListener.prototype.exitSelectLinesInto = function (ctx) {
};
MySqlParserListener.prototype.enterFromClause = function (ctx) {
};
MySqlParserListener.prototype.exitFromClause = function (ctx) {
};
MySqlParserListener.prototype.enterGroupByItem = function (ctx) {
};
MySqlParserListener.prototype.exitGroupByItem = function (ctx) {
};
MySqlParserListener.prototype.enterLimitClause = function (ctx) {
};
MySqlParserListener.prototype.exitLimitClause = function (ctx) {
};
MySqlParserListener.prototype.enterLimitClauseAtom = function (ctx) {
};
MySqlParserListener.prototype.exitLimitClauseAtom = function (ctx) {
};
MySqlParserListener.prototype.enterStartTransaction = function (ctx) {
};
MySqlParserListener.prototype.exitStartTransaction = function (ctx) {
};
MySqlParserListener.prototype.enterBeginWork = function (ctx) {
};
MySqlParserListener.prototype.exitBeginWork = function (ctx) {
};
MySqlParserListener.prototype.enterCommitWork = function (ctx) {
};
MySqlParserListener.prototype.exitCommitWork = function (ctx) {
};
MySqlParserListener.prototype.enterRollbackWork = function (ctx) {
};
MySqlParserListener.prototype.exitRollbackWork = function (ctx) {
};
MySqlParserListener.prototype.enterSavepointStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSavepointStatement = function (ctx) {
};
MySqlParserListener.prototype.enterRollbackStatement = function (ctx) {
};
MySqlParserListener.prototype.exitRollbackStatement = function (ctx) {
};
MySqlParserListener.prototype.enterReleaseStatement = function (ctx) {
};
MySqlParserListener.prototype.exitReleaseStatement = function (ctx) {
};
MySqlParserListener.prototype.enterLockTables = function (ctx) {
};
MySqlParserListener.prototype.exitLockTables = function (ctx) {
};
MySqlParserListener.prototype.enterUnlockTables = function (ctx) {
};
MySqlParserListener.prototype.exitUnlockTables = function (ctx) {
};
MySqlParserListener.prototype.enterSetAutocommitStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSetAutocommitStatement = function (ctx) {
};
MySqlParserListener.prototype.enterSetTransactionStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSetTransactionStatement = function (ctx) {
};
MySqlParserListener.prototype.enterTransactionMode = function (ctx) {
};
MySqlParserListener.prototype.exitTransactionMode = function (ctx) {
};
MySqlParserListener.prototype.enterLockTableElement = function (ctx) {
};
MySqlParserListener.prototype.exitLockTableElement = function (ctx) {
};
MySqlParserListener.prototype.enterLockAction = function (ctx) {
};
MySqlParserListener.prototype.exitLockAction = function (ctx) {
};
MySqlParserListener.prototype.enterTransactionOption = function (ctx) {
};
MySqlParserListener.prototype.exitTransactionOption = function (ctx) {
};
MySqlParserListener.prototype.enterTransactionLevel = function (ctx) {
};
MySqlParserListener.prototype.exitTransactionLevel = function (ctx) {
};
MySqlParserListener.prototype.enterChangeMaster = function (ctx) {
};
MySqlParserListener.prototype.exitChangeMaster = function (ctx) {
};
MySqlParserListener.prototype.enterChangeReplicationFilter = function (ctx) {
};
MySqlParserListener.prototype.exitChangeReplicationFilter = function (ctx) {
};
MySqlParserListener.prototype.enterPurgeBinaryLogs = function (ctx) {
};
MySqlParserListener.prototype.exitPurgeBinaryLogs = function (ctx) {
};
MySqlParserListener.prototype.enterResetMaster = function (ctx) {
};
MySqlParserListener.prototype.exitResetMaster = function (ctx) {
};
MySqlParserListener.prototype.enterResetSlave = function (ctx) {
};
MySqlParserListener.prototype.exitResetSlave = function (ctx) {
};
MySqlParserListener.prototype.enterStartSlave = function (ctx) {
};
MySqlParserListener.prototype.exitStartSlave = function (ctx) {
};
MySqlParserListener.prototype.enterStopSlave = function (ctx) {
};
MySqlParserListener.prototype.exitStopSlave = function (ctx) {
};
MySqlParserListener.prototype.enterStartGroupReplication = function (ctx) {
};
MySqlParserListener.prototype.exitStartGroupReplication = function (ctx) {
};
MySqlParserListener.prototype.enterStopGroupReplication = function (ctx) {
};
MySqlParserListener.prototype.exitStopGroupReplication = function (ctx) {
};
MySqlParserListener.prototype.enterMasterStringOption = function (ctx) {
};
MySqlParserListener.prototype.exitMasterStringOption = function (ctx) {
};
MySqlParserListener.prototype.enterMasterDecimalOption = function (ctx) {
};
MySqlParserListener.prototype.exitMasterDecimalOption = function (ctx) {
};
MySqlParserListener.prototype.enterMasterBoolOption = function (ctx) {
};
MySqlParserListener.prototype.exitMasterBoolOption = function (ctx) {
};
MySqlParserListener.prototype.enterMasterRealOption = function (ctx) {
};
MySqlParserListener.prototype.exitMasterRealOption = function (ctx) {
};
MySqlParserListener.prototype.enterMasterUidListOption = function (ctx) {
};
MySqlParserListener.prototype.exitMasterUidListOption = function (ctx) {
};
MySqlParserListener.prototype.enterStringMasterOption = function (ctx) {
};
MySqlParserListener.prototype.exitStringMasterOption = function (ctx) {
};
MySqlParserListener.prototype.enterDecimalMasterOption = function (ctx) {
};
MySqlParserListener.prototype.exitDecimalMasterOption = function (ctx) {
};
MySqlParserListener.prototype.enterBoolMasterOption = function (ctx) {
};
MySqlParserListener.prototype.exitBoolMasterOption = function (ctx) {
};
MySqlParserListener.prototype.enterChannelOption = function (ctx) {
};
MySqlParserListener.prototype.exitChannelOption = function (ctx) {
};
MySqlParserListener.prototype.enterDoDbReplication = function (ctx) {
};
MySqlParserListener.prototype.exitDoDbReplication = function (ctx) {
};
MySqlParserListener.prototype.enterIgnoreDbReplication = function (ctx) {
};
MySqlParserListener.prototype.exitIgnoreDbReplication = function (ctx) {
};
MySqlParserListener.prototype.enterDoTableReplication = function (ctx) {
};
MySqlParserListener.prototype.exitDoTableReplication = function (ctx) {
};
MySqlParserListener.prototype.enterIgnoreTableReplication = function (ctx) {
};
MySqlParserListener.prototype.exitIgnoreTableReplication = function (ctx) {
};
MySqlParserListener.prototype.enterWildDoTableReplication = function (ctx) {
};
MySqlParserListener.prototype.exitWildDoTableReplication = function (ctx) {
};
MySqlParserListener.prototype.enterWildIgnoreTableReplication = function (ctx) {
};
MySqlParserListener.prototype.exitWildIgnoreTableReplication = function (ctx) {
};
MySqlParserListener.prototype.enterRewriteDbReplication = function (ctx) {
};
MySqlParserListener.prototype.exitRewriteDbReplication = function (ctx) {
};
MySqlParserListener.prototype.enterTablePair = function (ctx) {
};
MySqlParserListener.prototype.exitTablePair = function (ctx) {
};
MySqlParserListener.prototype.enterThreadType = function (ctx) {
};
MySqlParserListener.prototype.exitThreadType = function (ctx) {
};
MySqlParserListener.prototype.enterGtidsUntilOption = function (ctx) {
};
MySqlParserListener.prototype.exitGtidsUntilOption = function (ctx) {
};
MySqlParserListener.prototype.enterMasterLogUntilOption = function (ctx) {
};
MySqlParserListener.prototype.exitMasterLogUntilOption = function (ctx) {
};
MySqlParserListener.prototype.enterRelayLogUntilOption = function (ctx) {
};
MySqlParserListener.prototype.exitRelayLogUntilOption = function (ctx) {
};
MySqlParserListener.prototype.enterSqlGapsUntilOption = function (ctx) {
};
MySqlParserListener.prototype.exitSqlGapsUntilOption = function (ctx) {
};
MySqlParserListener.prototype.enterUserConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.exitUserConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.enterPasswordConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.exitPasswordConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.enterDefaultAuthConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.exitDefaultAuthConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.enterPluginDirConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.exitPluginDirConnectionOption = function (ctx) {
};
MySqlParserListener.prototype.enterGtuidSet = function (ctx) {
};
MySqlParserListener.prototype.exitGtuidSet = function (ctx) {
};
MySqlParserListener.prototype.enterXaStartTransaction = function (ctx) {
};
MySqlParserListener.prototype.exitXaStartTransaction = function (ctx) {
};
MySqlParserListener.prototype.enterXaEndTransaction = function (ctx) {
};
MySqlParserListener.prototype.exitXaEndTransaction = function (ctx) {
};
MySqlParserListener.prototype.enterXaPrepareStatement = function (ctx) {
};
MySqlParserListener.prototype.exitXaPrepareStatement = function (ctx) {
};
MySqlParserListener.prototype.enterXaCommitWork = function (ctx) {
};
MySqlParserListener.prototype.exitXaCommitWork = function (ctx) {
};
MySqlParserListener.prototype.enterXaRollbackWork = function (ctx) {
};
MySqlParserListener.prototype.exitXaRollbackWork = function (ctx) {
};
MySqlParserListener.prototype.enterXaRecoverWork = function (ctx) {
};
MySqlParserListener.prototype.exitXaRecoverWork = function (ctx) {
};
MySqlParserListener.prototype.enterPrepareStatement = function (ctx) {
};
MySqlParserListener.prototype.exitPrepareStatement = function (ctx) {
};
MySqlParserListener.prototype.enterExecuteStatement = function (ctx) {
};
MySqlParserListener.prototype.exitExecuteStatement = function (ctx) {
};
MySqlParserListener.prototype.enterDeallocatePrepare = function (ctx) {
};
MySqlParserListener.prototype.exitDeallocatePrepare = function (ctx) {
};
MySqlParserListener.prototype.enterRoutineBody = function (ctx) {
};
MySqlParserListener.prototype.exitRoutineBody = function (ctx) {
};
MySqlParserListener.prototype.enterBlockStatement = function (ctx) {
};
MySqlParserListener.prototype.exitBlockStatement = function (ctx) {
};
MySqlParserListener.prototype.enterCaseStatement = function (ctx) {
};
MySqlParserListener.prototype.exitCaseStatement = function (ctx) {
};
MySqlParserListener.prototype.enterIfStatement = function (ctx) {
};
MySqlParserListener.prototype.exitIfStatement = function (ctx) {
};
MySqlParserListener.prototype.enterIterateStatement = function (ctx) {
};
MySqlParserListener.prototype.exitIterateStatement = function (ctx) {
};
MySqlParserListener.prototype.enterLeaveStatement = function (ctx) {
};
MySqlParserListener.prototype.exitLeaveStatement = function (ctx) {
};
MySqlParserListener.prototype.enterLoopStatement = function (ctx) {
};
MySqlParserListener.prototype.exitLoopStatement = function (ctx) {
};
MySqlParserListener.prototype.enterRepeatStatement = function (ctx) {
};
MySqlParserListener.prototype.exitRepeatStatement = function (ctx) {
};
MySqlParserListener.prototype.enterReturnStatement = function (ctx) {
};
MySqlParserListener.prototype.exitReturnStatement = function (ctx) {
};
MySqlParserListener.prototype.enterWhileStatement = function (ctx) {
};
MySqlParserListener.prototype.exitWhileStatement = function (ctx) {
};
MySqlParserListener.prototype.enterCloseCursor = function (ctx) {
};
MySqlParserListener.prototype.exitCloseCursor = function (ctx) {
};
MySqlParserListener.prototype.enterFetchCursor = function (ctx) {
};
MySqlParserListener.prototype.exitFetchCursor = function (ctx) {
};
MySqlParserListener.prototype.enterOpenCursor = function (ctx) {
};
MySqlParserListener.prototype.exitOpenCursor = function (ctx) {
};
MySqlParserListener.prototype.enterDeclareVariable = function (ctx) {
};
MySqlParserListener.prototype.exitDeclareVariable = function (ctx) {
};
MySqlParserListener.prototype.enterDeclareCondition = function (ctx) {
};
MySqlParserListener.prototype.exitDeclareCondition = function (ctx) {
};
MySqlParserListener.prototype.enterDeclareCursor = function (ctx) {
};
MySqlParserListener.prototype.exitDeclareCursor = function (ctx) {
};
MySqlParserListener.prototype.enterDeclareHandler = function (ctx) {
};
MySqlParserListener.prototype.exitDeclareHandler = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerConditionCode = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerConditionCode = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerConditionState = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerConditionState = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerConditionName = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerConditionName = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerConditionWarning = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerConditionWarning = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerConditionNotfound = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerConditionNotfound = function (ctx) {
};
MySqlParserListener.prototype.enterHandlerConditionException = function (ctx) {
};
MySqlParserListener.prototype.exitHandlerConditionException = function (ctx) {
};
MySqlParserListener.prototype.enterProcedureSqlStatement = function (ctx) {
};
MySqlParserListener.prototype.exitProcedureSqlStatement = function (ctx) {
};
MySqlParserListener.prototype.enterCaseAlternative = function (ctx) {
};
MySqlParserListener.prototype.exitCaseAlternative = function (ctx) {
};
MySqlParserListener.prototype.enterElifAlternative = function (ctx) {
};
MySqlParserListener.prototype.exitElifAlternative = function (ctx) {
};
MySqlParserListener.prototype.enterAlterUserMysqlV56 = function (ctx) {
};
MySqlParserListener.prototype.exitAlterUserMysqlV56 = function (ctx) {
};
MySqlParserListener.prototype.enterAlterUserMysqlV57 = function (ctx) {
};
MySqlParserListener.prototype.exitAlterUserMysqlV57 = function (ctx) {
};
MySqlParserListener.prototype.enterCreateUserMysqlV56 = function (ctx) {
};
MySqlParserListener.prototype.exitCreateUserMysqlV56 = function (ctx) {
};
MySqlParserListener.prototype.enterCreateUserMysqlV57 = function (ctx) {
};
MySqlParserListener.prototype.exitCreateUserMysqlV57 = function (ctx) {
};
MySqlParserListener.prototype.enterDropUser = function (ctx) {
};
MySqlParserListener.prototype.exitDropUser = function (ctx) {
};
MySqlParserListener.prototype.enterGrantStatement = function (ctx) {
};
MySqlParserListener.prototype.exitGrantStatement = function (ctx) {
};
MySqlParserListener.prototype.enterGrantProxy = function (ctx) {
};
MySqlParserListener.prototype.exitGrantProxy = function (ctx) {
};
MySqlParserListener.prototype.enterRenameUser = function (ctx) {
};
MySqlParserListener.prototype.exitRenameUser = function (ctx) {
};
MySqlParserListener.prototype.enterDetailRevoke = function (ctx) {
};
MySqlParserListener.prototype.exitDetailRevoke = function (ctx) {
};
MySqlParserListener.prototype.enterShortRevoke = function (ctx) {
};
MySqlParserListener.prototype.exitShortRevoke = function (ctx) {
};
MySqlParserListener.prototype.enterRevokeProxy = function (ctx) {
};
MySqlParserListener.prototype.exitRevokeProxy = function (ctx) {
};
MySqlParserListener.prototype.enterSetPasswordStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSetPasswordStatement = function (ctx) {
};
MySqlParserListener.prototype.enterUserSpecification = function (ctx) {
};
MySqlParserListener.prototype.exitUserSpecification = function (ctx) {
};
MySqlParserListener.prototype.enterPasswordAuthOption = function (ctx) {
};
MySqlParserListener.prototype.exitPasswordAuthOption = function (ctx) {
};
MySqlParserListener.prototype.enterStringAuthOption = function (ctx) {
};
MySqlParserListener.prototype.exitStringAuthOption = function (ctx) {
};
MySqlParserListener.prototype.enterHashAuthOption = function (ctx) {
};
MySqlParserListener.prototype.exitHashAuthOption = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleAuthOption = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleAuthOption = function (ctx) {
};
MySqlParserListener.prototype.enterTlsOption = function (ctx) {
};
MySqlParserListener.prototype.exitTlsOption = function (ctx) {
};
MySqlParserListener.prototype.enterUserResourceOption = function (ctx) {
};
MySqlParserListener.prototype.exitUserResourceOption = function (ctx) {
};
MySqlParserListener.prototype.enterUserPasswordOption = function (ctx) {
};
MySqlParserListener.prototype.exitUserPasswordOption = function (ctx) {
};
MySqlParserListener.prototype.enterUserLockOption = function (ctx) {
};
MySqlParserListener.prototype.exitUserLockOption = function (ctx) {
};
MySqlParserListener.prototype.enterPrivelegeClause = function (ctx) {
};
MySqlParserListener.prototype.exitPrivelegeClause = function (ctx) {
};
MySqlParserListener.prototype.enterPrivilege = function (ctx) {
};
MySqlParserListener.prototype.exitPrivilege = function (ctx) {
};
MySqlParserListener.prototype.enterCurrentSchemaPriviLevel = function (ctx) {
};
MySqlParserListener.prototype.exitCurrentSchemaPriviLevel = function (ctx) {
};
MySqlParserListener.prototype.enterGlobalPrivLevel = function (ctx) {
};
MySqlParserListener.prototype.exitGlobalPrivLevel = function (ctx) {
};
MySqlParserListener.prototype.enterDefiniteSchemaPrivLevel = function (ctx) {
};
MySqlParserListener.prototype.exitDefiniteSchemaPrivLevel = function (ctx) {
};
MySqlParserListener.prototype.enterDefiniteFullTablePrivLevel = function (ctx) {
};
MySqlParserListener.prototype.exitDefiniteFullTablePrivLevel = function (ctx) {
};
MySqlParserListener.prototype.enterDefiniteFullTablePrivLevel2 = function (ctx) {
};
MySqlParserListener.prototype.exitDefiniteFullTablePrivLevel2 = function (ctx) {
};
MySqlParserListener.prototype.enterDefiniteTablePrivLevel = function (ctx) {
};
MySqlParserListener.prototype.exitDefiniteTablePrivLevel = function (ctx) {
};
MySqlParserListener.prototype.enterRenameUserClause = function (ctx) {
};
MySqlParserListener.prototype.exitRenameUserClause = function (ctx) {
};
MySqlParserListener.prototype.enterAnalyzeTable = function (ctx) {
};
MySqlParserListener.prototype.exitAnalyzeTable = function (ctx) {
};
MySqlParserListener.prototype.enterCheckTable = function (ctx) {
};
MySqlParserListener.prototype.exitCheckTable = function (ctx) {
};
MySqlParserListener.prototype.enterChecksumTable = function (ctx) {
};
MySqlParserListener.prototype.exitChecksumTable = function (ctx) {
};
MySqlParserListener.prototype.enterOptimizeTable = function (ctx) {
};
MySqlParserListener.prototype.exitOptimizeTable = function (ctx) {
};
MySqlParserListener.prototype.enterRepairTable = function (ctx) {
};
MySqlParserListener.prototype.exitRepairTable = function (ctx) {
};
MySqlParserListener.prototype.enterCheckTableOption = function (ctx) {
};
MySqlParserListener.prototype.exitCheckTableOption = function (ctx) {
};
MySqlParserListener.prototype.enterCreateUdfunction = function (ctx) {
};
MySqlParserListener.prototype.exitCreateUdfunction = function (ctx) {
};
MySqlParserListener.prototype.enterInstallPlugin = function (ctx) {
};
MySqlParserListener.prototype.exitInstallPlugin = function (ctx) {
};
MySqlParserListener.prototype.enterUninstallPlugin = function (ctx) {
};
MySqlParserListener.prototype.exitUninstallPlugin = function (ctx) {
};
MySqlParserListener.prototype.enterSetVariable = function (ctx) {
};
MySqlParserListener.prototype.exitSetVariable = function (ctx) {
};
MySqlParserListener.prototype.enterSetCharset = function (ctx) {
};
MySqlParserListener.prototype.exitSetCharset = function (ctx) {
};
MySqlParserListener.prototype.enterSetNames = function (ctx) {
};
MySqlParserListener.prototype.exitSetNames = function (ctx) {
};
MySqlParserListener.prototype.enterSetPassword = function (ctx) {
};
MySqlParserListener.prototype.exitSetPassword = function (ctx) {
};
MySqlParserListener.prototype.enterSetTransaction = function (ctx) {
};
MySqlParserListener.prototype.exitSetTransaction = function (ctx) {
};
MySqlParserListener.prototype.enterSetAutocommit = function (ctx) {
};
MySqlParserListener.prototype.exitSetAutocommit = function (ctx) {
};
MySqlParserListener.prototype.enterSetNewValueInsideTrigger = function (ctx) {
};
MySqlParserListener.prototype.exitSetNewValueInsideTrigger = function (ctx) {
};
MySqlParserListener.prototype.enterShowMasterLogs = function (ctx) {
};
MySqlParserListener.prototype.exitShowMasterLogs = function (ctx) {
};
MySqlParserListener.prototype.enterShowLogEvents = function (ctx) {
};
MySqlParserListener.prototype.exitShowLogEvents = function (ctx) {
};
MySqlParserListener.prototype.enterShowObjectFilter = function (ctx) {
};
MySqlParserListener.prototype.exitShowObjectFilter = function (ctx) {
};
MySqlParserListener.prototype.enterShowColumns = function (ctx) {
};
MySqlParserListener.prototype.exitShowColumns = function (ctx) {
};
MySqlParserListener.prototype.enterShowCreateDb = function (ctx) {
};
MySqlParserListener.prototype.exitShowCreateDb = function (ctx) {
};
MySqlParserListener.prototype.enterShowCreateFullIdObject = function (ctx) {
};
MySqlParserListener.prototype.exitShowCreateFullIdObject = function (ctx) {
};
MySqlParserListener.prototype.enterShowCreateUser = function (ctx) {
};
MySqlParserListener.prototype.exitShowCreateUser = function (ctx) {
};
MySqlParserListener.prototype.enterShowEngine = function (ctx) {
};
MySqlParserListener.prototype.exitShowEngine = function (ctx) {
};
MySqlParserListener.prototype.enterShowGlobalInfo = function (ctx) {
};
MySqlParserListener.prototype.exitShowGlobalInfo = function (ctx) {
};
MySqlParserListener.prototype.enterShowErrors = function (ctx) {
};
MySqlParserListener.prototype.exitShowErrors = function (ctx) {
};
MySqlParserListener.prototype.enterShowCountErrors = function (ctx) {
};
MySqlParserListener.prototype.exitShowCountErrors = function (ctx) {
};
MySqlParserListener.prototype.enterShowSchemaFilter = function (ctx) {
};
MySqlParserListener.prototype.exitShowSchemaFilter = function (ctx) {
};
MySqlParserListener.prototype.enterShowRoutine = function (ctx) {
};
MySqlParserListener.prototype.exitShowRoutine = function (ctx) {
};
MySqlParserListener.prototype.enterShowGrants = function (ctx) {
};
MySqlParserListener.prototype.exitShowGrants = function (ctx) {
};
MySqlParserListener.prototype.enterShowIndexes = function (ctx) {
};
MySqlParserListener.prototype.exitShowIndexes = function (ctx) {
};
MySqlParserListener.prototype.enterShowOpenTables = function (ctx) {
};
MySqlParserListener.prototype.exitShowOpenTables = function (ctx) {
};
MySqlParserListener.prototype.enterShowProfile = function (ctx) {
};
MySqlParserListener.prototype.exitShowProfile = function (ctx) {
};
MySqlParserListener.prototype.enterShowSlaveStatus = function (ctx) {
};
MySqlParserListener.prototype.exitShowSlaveStatus = function (ctx) {
};
MySqlParserListener.prototype.enterVariableClause = function (ctx) {
};
MySqlParserListener.prototype.exitVariableClause = function (ctx) {
};
MySqlParserListener.prototype.enterShowCommonEntity = function (ctx) {
};
MySqlParserListener.prototype.exitShowCommonEntity = function (ctx) {
};
MySqlParserListener.prototype.enterShowFilter = function (ctx) {
};
MySqlParserListener.prototype.exitShowFilter = function (ctx) {
};
MySqlParserListener.prototype.enterShowGlobalInfoClause = function (ctx) {
};
MySqlParserListener.prototype.exitShowGlobalInfoClause = function (ctx) {
};
MySqlParserListener.prototype.enterShowSchemaEntity = function (ctx) {
};
MySqlParserListener.prototype.exitShowSchemaEntity = function (ctx) {
};
MySqlParserListener.prototype.enterShowProfileType = function (ctx) {
};
MySqlParserListener.prototype.exitShowProfileType = function (ctx) {
};
MySqlParserListener.prototype.enterBinlogStatement = function (ctx) {
};
MySqlParserListener.prototype.exitBinlogStatement = function (ctx) {
};
MySqlParserListener.prototype.enterCacheIndexStatement = function (ctx) {
};
MySqlParserListener.prototype.exitCacheIndexStatement = function (ctx) {
};
MySqlParserListener.prototype.enterFlushStatement = function (ctx) {
};
MySqlParserListener.prototype.exitFlushStatement = function (ctx) {
};
MySqlParserListener.prototype.enterKillStatement = function (ctx) {
};
MySqlParserListener.prototype.exitKillStatement = function (ctx) {
};
MySqlParserListener.prototype.enterLoadIndexIntoCache = function (ctx) {
};
MySqlParserListener.prototype.exitLoadIndexIntoCache = function (ctx) {
};
MySqlParserListener.prototype.enterResetStatement = function (ctx) {
};
MySqlParserListener.prototype.exitResetStatement = function (ctx) {
};
MySqlParserListener.prototype.enterShutdownStatement = function (ctx) {
};
MySqlParserListener.prototype.exitShutdownStatement = function (ctx) {
};
MySqlParserListener.prototype.enterTableIndexes = function (ctx) {
};
MySqlParserListener.prototype.exitTableIndexes = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleFlushOption = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleFlushOption = function (ctx) {
};
MySqlParserListener.prototype.enterChannelFlushOption = function (ctx) {
};
MySqlParserListener.prototype.exitChannelFlushOption = function (ctx) {
};
MySqlParserListener.prototype.enterTableFlushOption = function (ctx) {
};
MySqlParserListener.prototype.exitTableFlushOption = function (ctx) {
};
MySqlParserListener.prototype.enterFlushTableOption = function (ctx) {
};
MySqlParserListener.prototype.exitFlushTableOption = function (ctx) {
};
MySqlParserListener.prototype.enterLoadedTableIndexes = function (ctx) {
};
MySqlParserListener.prototype.exitLoadedTableIndexes = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleDescribeStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleDescribeStatement = function (ctx) {
};
MySqlParserListener.prototype.enterFullDescribeStatement = function (ctx) {
};
MySqlParserListener.prototype.exitFullDescribeStatement = function (ctx) {
};
MySqlParserListener.prototype.enterHelpStatement = function (ctx) {
};
MySqlParserListener.prototype.exitHelpStatement = function (ctx) {
};
MySqlParserListener.prototype.enterUseStatement = function (ctx) {
};
MySqlParserListener.prototype.exitUseStatement = function (ctx) {
};
MySqlParserListener.prototype.enterSignalStatement = function (ctx) {
};
MySqlParserListener.prototype.exitSignalStatement = function (ctx) {
};
MySqlParserListener.prototype.enterResignalStatement = function (ctx) {
};
MySqlParserListener.prototype.exitResignalStatement = function (ctx) {
};
MySqlParserListener.prototype.enterSignalConditionInformation = function (ctx) {
};
MySqlParserListener.prototype.exitSignalConditionInformation = function (ctx) {
};
MySqlParserListener.prototype.enterDiagnosticsStatement = function (ctx) {
};
MySqlParserListener.prototype.exitDiagnosticsStatement = function (ctx) {
};
MySqlParserListener.prototype.enterDiagnosticsConditionInformationName = function (ctx) {
};
MySqlParserListener.prototype.exitDiagnosticsConditionInformationName = function (ctx) {
};
MySqlParserListener.prototype.enterDescribeStatements = function (ctx) {
};
MySqlParserListener.prototype.exitDescribeStatements = function (ctx) {
};
MySqlParserListener.prototype.enterDescribeConnection = function (ctx) {
};
MySqlParserListener.prototype.exitDescribeConnection = function (ctx) {
};
MySqlParserListener.prototype.enterFullId = function (ctx) {
};
MySqlParserListener.prototype.exitFullId = function (ctx) {
};
MySqlParserListener.prototype.enterTableName = function (ctx) {
};
MySqlParserListener.prototype.exitTableName = function (ctx) {
};
MySqlParserListener.prototype.enterFullColumnName = function (ctx) {
};
MySqlParserListener.prototype.exitFullColumnName = function (ctx) {
};
MySqlParserListener.prototype.enterIndexColumnName = function (ctx) {
};
MySqlParserListener.prototype.exitIndexColumnName = function (ctx) {
};
MySqlParserListener.prototype.enterUserName = function (ctx) {
};
MySqlParserListener.prototype.exitUserName = function (ctx) {
};
MySqlParserListener.prototype.enterMysqlVariable = function (ctx) {
};
MySqlParserListener.prototype.exitMysqlVariable = function (ctx) {
};
MySqlParserListener.prototype.enterCharsetName = function (ctx) {
};
MySqlParserListener.prototype.exitCharsetName = function (ctx) {
};
MySqlParserListener.prototype.enterCollationName = function (ctx) {
};
MySqlParserListener.prototype.exitCollationName = function (ctx) {
};
MySqlParserListener.prototype.enterEngineName = function (ctx) {
};
MySqlParserListener.prototype.exitEngineName = function (ctx) {
};
MySqlParserListener.prototype.enterUuidSet = function (ctx) {
};
MySqlParserListener.prototype.exitUuidSet = function (ctx) {
};
MySqlParserListener.prototype.enterXid = function (ctx) {
};
MySqlParserListener.prototype.exitXid = function (ctx) {
};
MySqlParserListener.prototype.enterXuidStringId = function (ctx) {
};
MySqlParserListener.prototype.exitXuidStringId = function (ctx) {
};
MySqlParserListener.prototype.enterAuthPlugin = function (ctx) {
};
MySqlParserListener.prototype.exitAuthPlugin = function (ctx) {
};
MySqlParserListener.prototype.enterUid = function (ctx) {
};
MySqlParserListener.prototype.exitUid = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleId = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleId = function (ctx) {
};
MySqlParserListener.prototype.enterDottedId = function (ctx) {
};
MySqlParserListener.prototype.exitDottedId = function (ctx) {
};
MySqlParserListener.prototype.enterDecimalLiteral = function (ctx) {
};
MySqlParserListener.prototype.exitDecimalLiteral = function (ctx) {
};
MySqlParserListener.prototype.enterFileSizeLiteral = function (ctx) {
};
MySqlParserListener.prototype.exitFileSizeLiteral = function (ctx) {
};
MySqlParserListener.prototype.enterStringLiteral = function (ctx) {
};
MySqlParserListener.prototype.exitStringLiteral = function (ctx) {
};
MySqlParserListener.prototype.enterBooleanLiteral = function (ctx) {
};
MySqlParserListener.prototype.exitBooleanLiteral = function (ctx) {
};
MySqlParserListener.prototype.enterHexadecimalLiteral = function (ctx) {
};
MySqlParserListener.prototype.exitHexadecimalLiteral = function (ctx) {
};
MySqlParserListener.prototype.enterNullNotnull = function (ctx) {
};
MySqlParserListener.prototype.exitNullNotnull = function (ctx) {
};
MySqlParserListener.prototype.enterConstant = function (ctx) {
};
MySqlParserListener.prototype.exitConstant = function (ctx) {
};
MySqlParserListener.prototype.enterStringDataType = function (ctx) {
};
MySqlParserListener.prototype.exitStringDataType = function (ctx) {
};
MySqlParserListener.prototype.enterNationalStringDataType = function (ctx) {
};
MySqlParserListener.prototype.exitNationalStringDataType = function (ctx) {
};
MySqlParserListener.prototype.enterNationalVaryingStringDataType = function (ctx) {
};
MySqlParserListener.prototype.exitNationalVaryingStringDataType = function (ctx) {
};
MySqlParserListener.prototype.enterDimensionDataType = function (ctx) {
};
MySqlParserListener.prototype.exitDimensionDataType = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleDataType = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleDataType = function (ctx) {
};
MySqlParserListener.prototype.enterCollectionDataType = function (ctx) {
};
MySqlParserListener.prototype.exitCollectionDataType = function (ctx) {
};
MySqlParserListener.prototype.enterSpatialDataType = function (ctx) {
};
MySqlParserListener.prototype.exitSpatialDataType = function (ctx) {
};
MySqlParserListener.prototype.enterLongVarcharDataType = function (ctx) {
};
MySqlParserListener.prototype.exitLongVarcharDataType = function (ctx) {
};
MySqlParserListener.prototype.enterLongVarbinaryDataType = function (ctx) {
};
MySqlParserListener.prototype.exitLongVarbinaryDataType = function (ctx) {
};
MySqlParserListener.prototype.enterCollectionOptions = function (ctx) {
};
MySqlParserListener.prototype.exitCollectionOptions = function (ctx) {
};
MySqlParserListener.prototype.enterConvertedDataType = function (ctx) {
};
MySqlParserListener.prototype.exitConvertedDataType = function (ctx) {
};
MySqlParserListener.prototype.enterLengthOneDimension = function (ctx) {
};
MySqlParserListener.prototype.exitLengthOneDimension = function (ctx) {
};
MySqlParserListener.prototype.enterLengthTwoDimension = function (ctx) {
};
MySqlParserListener.prototype.exitLengthTwoDimension = function (ctx) {
};
MySqlParserListener.prototype.enterLengthTwoOptionalDimension = function (ctx) {
};
MySqlParserListener.prototype.exitLengthTwoOptionalDimension = function (ctx) {
};
MySqlParserListener.prototype.enterUidList = function (ctx) {
};
MySqlParserListener.prototype.exitUidList = function (ctx) {
};
MySqlParserListener.prototype.enterTables = function (ctx) {
};
MySqlParserListener.prototype.exitTables = function (ctx) {
};
MySqlParserListener.prototype.enterIndexColumnNames = function (ctx) {
};
MySqlParserListener.prototype.exitIndexColumnNames = function (ctx) {
};
MySqlParserListener.prototype.enterExpressions = function (ctx) {
};
MySqlParserListener.prototype.exitExpressions = function (ctx) {
};
MySqlParserListener.prototype.enterExpressionsWithDefaults = function (ctx) {
};
MySqlParserListener.prototype.exitExpressionsWithDefaults = function (ctx) {
};
MySqlParserListener.prototype.enterConstants = function (ctx) {
};
MySqlParserListener.prototype.exitConstants = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleStrings = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleStrings = function (ctx) {
};
MySqlParserListener.prototype.enterUserVariables = function (ctx) {
};
MySqlParserListener.prototype.exitUserVariables = function (ctx) {
};
MySqlParserListener.prototype.enterDefaultValue = function (ctx) {
};
MySqlParserListener.prototype.exitDefaultValue = function (ctx) {
};
MySqlParserListener.prototype.enterCurrentTimestamp = function (ctx) {
};
MySqlParserListener.prototype.exitCurrentTimestamp = function (ctx) {
};
MySqlParserListener.prototype.enterExpressionOrDefault = function (ctx) {
};
MySqlParserListener.prototype.exitExpressionOrDefault = function (ctx) {
};
MySqlParserListener.prototype.enterIfExists = function (ctx) {
};
MySqlParserListener.prototype.exitIfExists = function (ctx) {
};
MySqlParserListener.prototype.enterIfNotExists = function (ctx) {
};
MySqlParserListener.prototype.exitIfNotExists = function (ctx) {
};
MySqlParserListener.prototype.enterSpecificFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitSpecificFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterAggregateFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitAggregateFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterScalarFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitScalarFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterUdfFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitUdfFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterPasswordFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitPasswordFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterSimpleFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitSimpleFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterDataTypeFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitDataTypeFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterValuesFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitValuesFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterCaseFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitCaseFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterCharFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitCharFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterPositionFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitPositionFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterSubstrFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitSubstrFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterTrimFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitTrimFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterWeightFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitWeightFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterExtractFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitExtractFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterGetFormatFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.exitGetFormatFunctionCall = function (ctx) {
};
MySqlParserListener.prototype.enterCaseFuncAlternative = function (ctx) {
};
MySqlParserListener.prototype.exitCaseFuncAlternative = function (ctx) {
};
MySqlParserListener.prototype.enterLevelWeightList = function (ctx) {
};
MySqlParserListener.prototype.exitLevelWeightList = function (ctx) {
};
MySqlParserListener.prototype.enterLevelWeightRange = function (ctx) {
};
MySqlParserListener.prototype.exitLevelWeightRange = function (ctx) {
};
MySqlParserListener.prototype.enterLevelInWeightListElement = function (ctx) {
};
MySqlParserListener.prototype.exitLevelInWeightListElement = function (ctx) {
};
MySqlParserListener.prototype.enterAggregateWindowedFunction = function (ctx) {
};
MySqlParserListener.prototype.exitAggregateWindowedFunction = function (ctx) {
};
MySqlParserListener.prototype.enterScalarFunctionName = function (ctx) {
};
MySqlParserListener.prototype.exitScalarFunctionName = function (ctx) {
};
MySqlParserListener.prototype.enterPasswordFunctionClause = function (ctx) {
};
MySqlParserListener.prototype.exitPasswordFunctionClause = function (ctx) {
};
MySqlParserListener.prototype.enterFunctionArgs = function (ctx) {
};
MySqlParserListener.prototype.exitFunctionArgs = function (ctx) {
};
MySqlParserListener.prototype.enterFunctionArg = function (ctx) {
};
MySqlParserListener.prototype.exitFunctionArg = function (ctx) {
};
MySqlParserListener.prototype.enterIsExpression = function (ctx) {
};
MySqlParserListener.prototype.exitIsExpression = function (ctx) {
};
MySqlParserListener.prototype.enterNotExpression = function (ctx) {
};
MySqlParserListener.prototype.exitNotExpression = function (ctx) {
};
MySqlParserListener.prototype.enterLogicalExpression = function (ctx) {
};
MySqlParserListener.prototype.exitLogicalExpression = function (ctx) {
};
MySqlParserListener.prototype.enterPredicateExpression = function (ctx) {
};
MySqlParserListener.prototype.exitPredicateExpression = function (ctx) {
};
MySqlParserListener.prototype.enterSoundsLikePredicate = function (ctx) {
};
MySqlParserListener.prototype.exitSoundsLikePredicate = function (ctx) {
};
MySqlParserListener.prototype.enterExpressionAtomPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitExpressionAtomPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterJsonMemberOfPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitJsonMemberOfPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterInPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitInPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterSubqueryComparasionPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitSubqueryComparasionPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterBetweenPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitBetweenPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterBinaryComparasionPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitBinaryComparasionPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterIsNullPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitIsNullPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterLikePredicate = function (ctx) {
};
MySqlParserListener.prototype.exitLikePredicate = function (ctx) {
};
MySqlParserListener.prototype.enterRegexpPredicate = function (ctx) {
};
MySqlParserListener.prototype.exitRegexpPredicate = function (ctx) {
};
MySqlParserListener.prototype.enterUnaryExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitUnaryExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterCollateExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitCollateExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterMysqlVariableExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitMysqlVariableExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterNestedExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitNestedExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterNestedRowExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitNestedRowExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterMathExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitMathExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterExistsExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitExistsExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterIntervalExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitIntervalExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterJsonExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitJsonExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterSubqueryExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitSubqueryExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterConstantExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitConstantExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterFunctionCallExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitFunctionCallExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterBinaryExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitBinaryExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterFullColumnNameExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitFullColumnNameExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterBitExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.exitBitExpressionAtom = function (ctx) {
};
MySqlParserListener.prototype.enterUnaryOperator = function (ctx) {
};
MySqlParserListener.prototype.exitUnaryOperator = function (ctx) {
};
MySqlParserListener.prototype.enterComparisonOperator = function (ctx) {
};
MySqlParserListener.prototype.exitComparisonOperator = function (ctx) {
};
MySqlParserListener.prototype.enterLogicalOperator = function (ctx) {
};
MySqlParserListener.prototype.exitLogicalOperator = function (ctx) {
};
MySqlParserListener.prototype.enterBitOperator = function (ctx) {
};
MySqlParserListener.prototype.exitBitOperator = function (ctx) {
};
MySqlParserListener.prototype.enterMathOperator = function (ctx) {
};
MySqlParserListener.prototype.exitMathOperator = function (ctx) {
};
MySqlParserListener.prototype.enterJsonOperator = function (ctx) {
};
MySqlParserListener.prototype.exitJsonOperator = function (ctx) {
};
MySqlParserListener.prototype.enterCharsetNameBase = function (ctx) {
};
MySqlParserListener.prototype.exitCharsetNameBase = function (ctx) {
};
MySqlParserListener.prototype.enterTransactionLevelBase = function (ctx) {
};
MySqlParserListener.prototype.exitTransactionLevelBase = function (ctx) {
};
MySqlParserListener.prototype.enterPrivilegesBase = function (ctx) {
};
MySqlParserListener.prototype.exitPrivilegesBase = function (ctx) {
};
MySqlParserListener.prototype.enterIntervalTypeBase = function (ctx) {
};
MySqlParserListener.prototype.exitIntervalTypeBase = function (ctx) {
};
MySqlParserListener.prototype.enterDataTypeBase = function (ctx) {
};
MySqlParserListener.prototype.exitDataTypeBase = function (ctx) {
};
MySqlParserListener.prototype.enterKeywordsCanBeId = function (ctx) {
};
MySqlParserListener.prototype.exitKeywordsCanBeId = function (ctx) {
};
MySqlParserListener.prototype.enterFunctionNameBase = function (ctx) {
};
MySqlParserListener.prototype.exitFunctionNameBase = function (ctx) {
};
exports.MySqlParserListener = MySqlParserListener;
