// @harvest-forward/runtime/dist/src/plugins/mysql/grammar/MySqlParserVisitor.js
"use strict";
var antlr4 = require('antlr4/index');
function MySqlParserVisitor() {
    antlr4.tree.ParseTreeVisitor.call(this);
    return this;
}
MySqlParserVisitor.prototype = Object.create(antlr4.tree.ParseTreeVisitor.prototype);
MySqlParserVisitor.prototype.constructor = MySqlParserVisitor;
MySqlParserVisitor.prototype.visitRoot = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSqlStatements = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSqlStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitEmptyStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDdlStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDmlStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTransactionStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReplicationStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPreparedStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCompoundStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAdministrationStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUtilityStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateDatabase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateEvent = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateIndex = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateLogfileGroup = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateProcedure = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateFunction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateServer = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCopyCreateTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitQueryCreateTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitColumnCreateTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateTablespaceInnodb = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateTablespaceNdb = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateTrigger = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateView = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateDatabaseOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitOwnerStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPreciseSchedule = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIntervalSchedule = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTimestampValue = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIntervalExpr = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIntervalType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitEnableType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIndexType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIndexOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitProcedureParameter = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFunctionParameter = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRoutineComment = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRoutineLanguage = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRoutineBehavior = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRoutineData = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRoutineSecurity = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitServerOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateDefinitions = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitColumnDeclaration = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitConstraintDeclaration = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIndexDeclaration = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitColumnDefinition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNullColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDefaultColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAutoIncrementColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPrimaryKeyColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUniqueKeyColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCommentColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFormatColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStorageColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReferenceColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCollateColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGeneratedColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSerialDefaultColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCheckColumnConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPrimaryKeyTableConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUniqueKeyTableConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitForeignKeyTableConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCheckTableConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReferenceDefinition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReferenceAction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReferenceControlType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleIndexDeclaration = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSpecialIndexDeclaration = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionEngine = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionAutoIncrement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionAverage = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionCharset = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionChecksum = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionCollate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionComment = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionCompression = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionConnection = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionDataDirectory = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionDelay = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionEncryption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionIndexDirectory = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionInsertMethod = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionKeyBlockSize = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionMaxRows = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionMinRows = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionPackKeys = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionPassword = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionRowFormat = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionRecalculation = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionPersistent = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionSamplePage = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionTablespace = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableOptionUnion = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTablespaceStorage = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionDefinitions = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionFunctionHash = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionFunctionKey = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionFunctionRange = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionFunctionList = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSubPartitionFunctionHash = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSubPartitionFunctionKey = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionComparision = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionListAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionListVector = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionSimple = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionDefinerAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionDefinerVector = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSubpartitionDefinition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionEngine = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionComment = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionDataDirectory = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionIndexDirectory = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionMaxRows = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionMinRows = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionTablespace = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPartitionOptionNodeGroup = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterSimpleDatabase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterUpgradeName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterEvent = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterFunction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterInstance = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterLogfileGroup = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterProcedure = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterServer = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterTablespace = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterView = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByTableOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddColumn = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddColumns = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddIndex = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddPrimaryKey = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddUniqueKey = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddSpecialIndex = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddForeignKey = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddCheckTableConstraint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterBySetAlgorithm = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByChangeDefault = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByChangeColumn = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByRenameColumn = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByLock = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByModifyColumn = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDropColumn = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDropConstraintCheck = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDropPrimaryKey = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByRenameIndex = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAlterIndexVisibility = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDropIndex = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDropForeignKey = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDisableKeys = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByEnableKeys = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByRename = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByOrder = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByConvertCharset = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDefaultCharset = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDiscardTablespace = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByImportTablespace = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByForce = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByValidate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAddPartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDropPartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByDiscardPartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByImportPartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByTruncatePartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByCoalescePartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByReorganizePartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByExchangePartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByAnalyzePartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByCheckPartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByOptimizePartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByRebuildPartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByRepairPartition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByRemovePartitioning = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterByUpgradePartitioning = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropDatabase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropEvent = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropIndex = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropLogfileGroup = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropProcedure = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropFunction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropServer = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropTablespace = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropTrigger = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropView = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRenameTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRenameTableClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTruncateTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCallStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDeleteStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDoStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitInsertStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLoadDataStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLoadXmlStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReplaceStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleSelect = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitParenthesisSelect = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUnionSelect = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUnionParenthesisSelect = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUpdateStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitInsertStatementValue = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUpdatedElement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAssignmentField = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLockClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSingleDeleteStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMultipleDeleteStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerOpenStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerReadIndexStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerReadStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerCloseStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSingleUpdateStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMultipleUpdateStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitOrderByClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitOrderByExpression = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableSources = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableSourceBase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableSourceNested = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAtomTableItem = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSubqueryTableItem = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableSourcesItem = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIndexHint = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIndexHintType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitInnerJoin = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStraightJoin = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitOuterJoin = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNaturalJoin = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitQueryExpression = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitQueryExpressionNointo = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitQuerySpecification = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitQuerySpecificationNointo = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUnionParenthesis = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUnionStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectSpec = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectElements = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectStarElement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectColumnElement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectFunctionElement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectExpressionElement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectIntoVariables = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectIntoDumpFile = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectIntoTextFile = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectFieldsInto = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSelectLinesInto = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFromClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGroupByItem = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLimitClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLimitClauseAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStartTransaction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBeginWork = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCommitWork = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRollbackWork = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSavepointStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRollbackStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReleaseStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLockTables = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUnlockTables = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetAutocommitStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetTransactionStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTransactionMode = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLockTableElement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLockAction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTransactionOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTransactionLevel = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitChangeMaster = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitChangeReplicationFilter = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPurgeBinaryLogs = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitResetMaster = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitResetSlave = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStartSlave = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStopSlave = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStartGroupReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStopGroupReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMasterStringOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMasterDecimalOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMasterBoolOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMasterRealOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMasterUidListOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStringMasterOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDecimalMasterOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBoolMasterOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitChannelOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDoDbReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIgnoreDbReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDoTableReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIgnoreTableReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitWildDoTableReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitWildIgnoreTableReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRewriteDbReplication = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTablePair = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitThreadType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGtidsUntilOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMasterLogUntilOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRelayLogUntilOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSqlGapsUntilOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUserConnectionOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPasswordConnectionOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDefaultAuthConnectionOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPluginDirConnectionOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGtuidSet = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXaStartTransaction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXaEndTransaction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXaPrepareStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXaCommitWork = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXaRollbackWork = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXaRecoverWork = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPrepareStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitExecuteStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDeallocatePrepare = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRoutineBody = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBlockStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCaseStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIfStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIterateStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLeaveStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLoopStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRepeatStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitReturnStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitWhileStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCloseCursor = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFetchCursor = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitOpenCursor = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDeclareVariable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDeclareCondition = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDeclareCursor = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDeclareHandler = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerConditionCode = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerConditionState = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerConditionName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerConditionWarning = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerConditionNotfound = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHandlerConditionException = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitProcedureSqlStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCaseAlternative = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitElifAlternative = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterUserMysqlV56 = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAlterUserMysqlV57 = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateUserMysqlV56 = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateUserMysqlV57 = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDropUser = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGrantStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGrantProxy = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRenameUser = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDetailRevoke = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShortRevoke = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRevokeProxy = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetPasswordStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUserSpecification = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPasswordAuthOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStringAuthOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHashAuthOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleAuthOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTlsOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUserResourceOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUserPasswordOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUserLockOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPrivelegeClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPrivilege = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCurrentSchemaPriviLevel = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGlobalPrivLevel = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDefiniteSchemaPrivLevel = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDefiniteFullTablePrivLevel = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDefiniteFullTablePrivLevel2 = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDefiniteTablePrivLevel = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRenameUserClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAnalyzeTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCheckTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitChecksumTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitOptimizeTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRepairTable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCheckTableOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCreateUdfunction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitInstallPlugin = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUninstallPlugin = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetVariable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetCharset = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetNames = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetPassword = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetTransaction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetAutocommit = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSetNewValueInsideTrigger = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowMasterLogs = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowLogEvents = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowObjectFilter = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowColumns = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowCreateDb = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowCreateFullIdObject = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowCreateUser = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowEngine = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowGlobalInfo = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowErrors = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowCountErrors = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowSchemaFilter = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowRoutine = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowGrants = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowIndexes = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowOpenTables = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowProfile = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowSlaveStatus = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitVariableClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowCommonEntity = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowFilter = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowGlobalInfoClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowSchemaEntity = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShowProfileType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBinlogStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCacheIndexStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFlushStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitKillStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLoadIndexIntoCache = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitResetStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitShutdownStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableIndexes = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleFlushOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitChannelFlushOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableFlushOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFlushTableOption = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLoadedTableIndexes = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleDescribeStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFullDescribeStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHelpStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUseStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSignalStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitResignalStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSignalConditionInformation = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDiagnosticsStatement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDiagnosticsConditionInformationName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDescribeStatements = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDescribeConnection = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFullId = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTableName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFullColumnName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIndexColumnName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUserName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMysqlVariable = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCharsetName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCollationName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitEngineName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUuidSet = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXid = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitXuidStringId = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAuthPlugin = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUid = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleId = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDottedId = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDecimalLiteral = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFileSizeLiteral = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStringLiteral = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBooleanLiteral = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitHexadecimalLiteral = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNullNotnull = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitConstant = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitStringDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNationalStringDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNationalVaryingStringDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDimensionDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCollectionDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSpatialDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLongVarcharDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLongVarbinaryDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCollectionOptions = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitConvertedDataType = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLengthOneDimension = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLengthTwoDimension = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLengthTwoOptionalDimension = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUidList = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTables = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIndexColumnNames = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitExpressions = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitExpressionsWithDefaults = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitConstants = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleStrings = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUserVariables = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDefaultValue = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCurrentTimestamp = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitExpressionOrDefault = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIfExists = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIfNotExists = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSpecificFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAggregateFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitScalarFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUdfFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPasswordFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSimpleFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDataTypeFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitValuesFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCaseFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCharFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPositionFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSubstrFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTrimFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitWeightFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitExtractFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitGetFormatFunctionCall = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCaseFuncAlternative = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLevelWeightList = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLevelWeightRange = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLevelInWeightListElement = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitAggregateWindowedFunction = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitScalarFunctionName = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPasswordFunctionClause = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFunctionArgs = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFunctionArg = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIsExpression = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNotExpression = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLogicalExpression = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPredicateExpression = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSoundsLikePredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitExpressionAtomPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitJsonMemberOfPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitInPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSubqueryComparasionPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBetweenPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBinaryComparasionPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIsNullPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLikePredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitRegexpPredicate = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUnaryExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCollateExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMysqlVariableExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNestedExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitNestedRowExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMathExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitExistsExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIntervalExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitJsonExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitSubqueryExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitConstantExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFunctionCallExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBinaryExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFullColumnNameExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBitExpressionAtom = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitUnaryOperator = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitComparisonOperator = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitLogicalOperator = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitBitOperator = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitMathOperator = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitJsonOperator = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitCharsetNameBase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitTransactionLevelBase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitPrivilegesBase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitIntervalTypeBase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitDataTypeBase = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitKeywordsCanBeId = function (ctx) {
    return this.visitChildren(ctx);
};
MySqlParserVisitor.prototype.visitFunctionNameBase = function (ctx) {
    return this.visitChildren(ctx);
};
exports.MySqlParserVisitor = MySqlParserVisitor;
