// @harvest-forward/runtime/dist/src/plugins/node/NodeBuilder.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeBuilder = void 0;
const childProcess = __importStar(require("child_process"));
const AbstractSingularBuilder_1 = require("../../runtime/AbstractSingularBuilder");
const System_1 = require("../../system/System");
const SystemComponent_1 = require("../../system/SystemComponent");
const ValueType_1 = require("../../system/ValueType");
const LogAsyncMethod_1 = require("../../util/LogAsyncMethod");
class NodeBuilder extends AbstractSingularBuilder_1.AbstractSingularBuilder {
    constructor(configurationValues) {
        super('NodeBuilder', 'A builder that executes a node.js application', {
            fileToRun: {
                name: 'FileTorun',
                required: true,
                description: 'A full path to a node.js main file to run',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
            args: {
                name: 'args',
                required: true,
                description: 'A stringified array of the arguments to pass to the script',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
        }, configurationValues ? configurationValues : []);
    }
    async runScript(scriptPath, args = []) {
        const _thisThis = this;
        const result = new Promise((resolve, reject) => {
            let invoked = false;
            const process = childProcess.fork(scriptPath, args);
            process.on('error', function (err) {
                _thisThis.logger.isLevelEnabled('debug') ? _thisThis.logger.debug(`runScript(${scriptPath}: ${args}) failed ${JSON.stringify(err)}`) : '';
                if (invoked)
                    return;
                invoked = true;
                reject(err);
            });
            process.on('exit', function (code) {
                _thisThis.logger.isLevelEnabled('debug') ? _thisThis.logger.debug(`runScript(${scriptPath}: ${args}) success ${JSON.stringify(code)}`) : '';
                if (invoked)
                    return;
                invoked = true;
                const err = code === 0 ? null : new Error('exit code ' + code);
                resolve(err);
            });
        });
        return result;
    }
    async buildInternal(__system, __component) {
        const targetPath = this.getConfiguredValue('fileToRun');
        const argsConfig = this.getConfiguredValue('args');
        const args = argsConfig ? JSON.parse(argsConfig) : [];
        await this.runScript(targetPath, args);
        return [];
    }
}
__decorate([
    (0, LogAsyncMethod_1.LogAsyncMethod)(),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [System_1.System, SystemComponent_1.SystemComponent]),
    __metadata("design:returntype", Promise)
], NodeBuilder.prototype, "buildInternal", null);
exports.NodeBuilder = NodeBuilder;
