// @harvest-forward/runtime/dist/src/plugins/orm-basic/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.builders = void 0;
const OrmColumnMappingCollector_1 = require("./builders/OrmColumnMappingCollector");
const OrmDefaultSortColumnHarvester_1 = require("./builders/OrmDefaultSortColumnHarvester");
const OrmDerivedClassPropertyConsolidator_1 = require("./builders/OrmDerivedClassPropertyConsolidator");
const OrmDomainToEnumHarvester_1 = require("./builders/OrmDomainToEnumHarvester");
const OrmIncomingForeignKeyToPropertyHarvester_1 = require("./builders/OrmIncomingForeignKeyToPropertyHarvester");
const OrmOutgoingForeignKeyToPropertyHarvester_1 = require("./builders/OrmOutgoingForeignKeyToPropertyHarvester");
const OrmStorageColumnToPropertyHarvester_1 = require("./builders/OrmStorageColumnToPropertyHarvester");
const OrmTableToClassHarvester_1 = require("./builders/OrmTableToClassHarvester");
const OrmViewToClassHarvester_1 = require("./builders/OrmViewToClassHarvester");
const OrmViewToDerivedClassTransformer_1 = require("./builders/OrmViewToDerivedClassTransformer");
const RdbmsRuleHarvester_1 = require("./builders/RdbmsRuleHarvester");
exports.builders = {
    ColumnMappingCollector: (settings) => new OrmColumnMappingCollector_1.OrmColumnMappingCollector(settings),
    DerivedClassPropertyConsolidator: (settings) => new OrmDerivedClassPropertyConsolidator_1.OrmDerivedClassPropertyConsolidator(settings),
    DefaultSortColumnHarvester: (settings) => new OrmDefaultSortColumnHarvester_1.OrmDefaultSortColumnHarvester(settings),
    DomainToEnumHarvester: (settings) => new OrmDomainToEnumHarvester_1.OrmDomainToEnumHarvester(settings),
    IncomingForeignKeyToPropertyHarvester: (settings) => new OrmIncomingForeignKeyToPropertyHarvester_1.OrmIncomingForeignKeyToPropertyHarvester(settings),
    OutgoingForeignKeyToPropertyHarvester: (settings) => new OrmOutgoingForeignKeyToPropertyHarvester_1.OrmOutgoingForeignKeyToPropertyHarvester(settings),
    RuleHarvester: (settings) => new RdbmsRuleHarvester_1.RdbmsRuleHarvester(settings),
    StorageColumnToPropertyHarvester: (settings) => new OrmStorageColumnToPropertyHarvester_1.OrmStorageColumnToPropertyHarvester(settings),
    TableToClassHarvester: (settings) => new OrmTableToClassHarvester_1.OrmTableToClassHarvester(settings),
    ViewToClassHarvester: (settings) => new OrmViewToClassHarvester_1.OrmViewToClassHarvester(settings),
    ViewToDerivedClassTransformer: (settings) => new OrmViewToDerivedClassTransformer_1.OrmViewToDerivedClassTransformer(settings),
};
__exportStar(require("./builders/AbstractSqlOrmHarvester"), exports);
__exportStar(require("./builders/OrmColumnMappingCollector"), exports);
__exportStar(require("./builders/OrmDefaultSortColumnHarvester"), exports);
__exportStar(require("./builders/OrmDerivedClassPropertyConsolidator"), exports);
__exportStar(require("./builders/OrmDomainToEnumHarvester"), exports);
__exportStar(require("./builders/OrmIncomingForeignKeyToPropertyHarvester"), exports);
__exportStar(require("./builders/OrmOutgoingForeignKeyToPropertyHarvester"), exports);
__exportStar(require("./builders/OrmStorageColumnToPropertyHarvester"), exports);
__exportStar(require("./builders/OrmTableToClassHarvester"), exports);
__exportStar(require("./builders/OrmViewToClassHarvester"), exports);
__exportStar(require("./builders/OrmViewToDerivedClassTransformer"), exports);
__exportStar(require("./builders/RdbmsDiscriminatorHarvester"), exports);
__exportStar(require("./builders/RdbmsRuleHarvester"), exports);
__exportStar(require("./DataMemberColumnMapping"), exports);
__exportStar(require("./MappableDataMember"), exports);
__exportStar(require("./MappableDataType"), exports);
__exportStar(require("./ObjectRelationalMapping"), exports);
__exportStar(require("./ObjectTypeTableMapping"), exports);
