// @harvest-forward/runtime/dist/src/plugins/rdbms-basic/builders/RdbmsPkStitcher.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RdbmsPkStitcher = void 0;
const AbstractSingularBuilder_1 = require("../../../runtime/AbstractSingularBuilder");
const System_1 = require("../../../system/System");
const SystemComponent_1 = require("../../../system/SystemComponent");
const ValueType_1 = require("../../../system/ValueType");
class RdbmsPkStitcher extends AbstractSingularBuilder_1.AbstractSingularBuilder {
    constructor(configurationValues) {
        const nonNullConfigurationValues = configurationValues ? configurationValues : [];
        nonNullConfigurationValues['repeatFor'] = JSON.stringify({
            objectTypeName: 'PrimaryKey',
        });
        super('RdbmsPkStitcher', 'A transformer that ensure primary keys are properly connected to their respective table and columns', {
            repeatFor: {
                name: 'Repeat for',
                required: true,
                description: 'A parsable string of JSON that represents the properties values of the component that should be iterated on, review the documentation for SystemDescendantComponent and derivations for most available for use',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
        }, nonNullConfigurationValues);
    }
    async buildInternal(system, component) {
        const pk = component;
        const schema = pk.parent;
        const fullTableName = SystemComponent_1.SystemComponent.fullConstantCase(schema.fullConstantCaseName, pk.tableName);
        const table = system.descendants[fullTableName];
        if (table != null) {
            const pkColumn = table.columns[System_1.System.constantCase(pk.columnName)];
            if (pkColumn != null) {
                table.primaryKey = pkColumn;
                table.addChild(pk);
                pk.column = pkColumn;
            }
            else
                this.logger.warn(`PK ${pk.constantCaseFullName} has no matching column ${pk.columnName}`);
        }
        else
            this.logger.warn(`PK ${pk.constantCaseFullName} has no matching table ${fullTableName}`);
        return [];
    }
}
exports.RdbmsPkStitcher = RdbmsPkStitcher;
