// @harvest-forward/runtime/dist/src/plugins/rdbms-basic/builders/RdbmsConstraintStitcher.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RdbmsConstraintStitcher = void 0;
const AbstractSingularBuilder_1 = require("../../../runtime/AbstractSingularBuilder");
const System_1 = require("../../../system/System");
const SystemComponent_1 = require("../../../system/SystemComponent");
const ValueType_1 = require("../../../system/ValueType");
class RdbmsConstraintStitcher extends AbstractSingularBuilder_1.AbstractSingularBuilder {
    constructor(configurationValues) {
        const nonNullConfigurationValues = configurationValues ? configurationValues : [];
        nonNullConfigurationValues['repeatFor'] = JSON.stringify({
            objectTypeName: 'Constraint',
        });
        super('RdbmsConstraintStitcher', 'A transformer that ensures that constraints are properly connected to their related tables and values', {
            repeatFor: {
                name: 'Repeat for',
                required: true,
                description: 'A parsable string of JSON that represents the properties values of the component that should be iterated on, review the documentation for SystemDescendantComponent and derivations for most available for use',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
        }, nonNullConfigurationValues);
    }
    async buildInternal(system, component) {
        const constraint = component;
        const schema = constraint.parent;
        const fullTableName = SystemComponent_1.SystemComponent.fullConstantCase(schema.constantCaseFullName, constraint.schemaName);
        const table = system.descendants[fullTableName];
        if (table != null) {
            if (constraint.columnName != null) {
                const constraintColumn = table.columns[System_1.System.constantCase(constraint.columnName)];
                constraintColumn.constraint = constraint;
                table.addChild(constraint);
                constraint.table = table;
                constraint.column = constraintColumn;
            }
            else
                table.addChild(constraint);
        }
        else
            this.logger.warn(`Constraint has no matching table ${constraint.constantCaseFullName}`);
        return [];
    }
}
exports.RdbmsConstraintStitcher = RdbmsConstraintStitcher;
