// @harvest-forward/runtime/dist/src/plugins/orm-basic/builders/OrmTableToClassHarvester.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrmTableToClassHarvester = void 0;
const changeCase = __importStar(require("change-case"));
const AbstractSingularBuilder_1 = require("../../../runtime/AbstractSingularBuilder");
const ComponentOrigin_1 = require("../../../system/ComponentOrigin");
const Permanence_1 = require("../../../system/Permanence");
const SystemComponent_1 = require("../../../system/SystemComponent");
const SystemComponentArtifact_1 = require("../../../system/SystemComponentArtifact");
const SystemComponentType_1 = require("../../../system/SystemComponentType");
const ValueType_1 = require("../../../system/ValueType");
const information_architecture_1 = require("../../information-architecture");
const ObjectTypeTableMapping_1 = require("../ObjectTypeTableMapping");
class OrmTableToClassHarvester extends AbstractSingularBuilder_1.AbstractSingularBuilder {
    constructor(configurationValues) {
        const nonNullConfigurationValues = configurationValues ? configurationValues : [];
        nonNullConfigurationValues['repeatFor'] = JSON.stringify({
            objectTypeName: 'Table',
        });
        super('OrmTableToClassHarvester', 'A harvester that takes tables in a sql schema and creates classes in an information model', {
            repeatFor: {
                name: 'Repeat for',
                required: true,
                description: 'A parsable string of JSON that represents the properties values of the component that should be iterated on, review the documentation for SystemDescendantComponent and derivations for most available for use',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
        }, nonNullConfigurationValues);
    }
    async buildInternal(system, component) {
        const results = [];
        const table = component;
        let mapping = table.orMapping;
        let typeName;
        if (mapping != null) {
            typeName = mapping.typeName;
            mapping.storage = table;
        }
        if (typeName == null || typeName == '')
            typeName = table.singularName == null ? table.name : table.singularName;
        typeName = changeCase.pascalCase(typeName);
        const model = information_architecture_1.InformationModel.fromSystem(system);
        const fullName = SystemComponent_1.SystemComponent.fullConstantCase(model.fullConstantCaseName, typeName);
        const valueType = system.descendants[fullName];
        if (valueType instanceof information_architecture_1.Class || valueType == null) {
            let informationalClass = valueType;
            if (mapping == null) {
                const schema = table.parent;
                const db = schema.parent;
                const orm = system.findOneComponent({
                    objectTypeName: 'ObjectRelationalMapping',
                });
                mapping = new ObjectTypeTableMapping_1.ObjectTypeTableMapping(model.fullConstantCaseName, typeName, db.fullConstantCaseName, schema.name, table.name);
                mapping.permanence = Permanence_1.Permanence.constant;
                mapping.informational = true;
                mapping.functional = false;
                mapping.origin = ComponentOrigin_1.ComponentOrigin.manufactured;
                orm.addChild(mapping);
                results.push(new SystemComponentArtifact_1.SystemComponentArtifact(mapping));
                mapping.storage = table;
            }
            else {
                mapping.typeName = typeName;
            }
            ;
            table.orMapping = mapping;
            let pluralName = table.pluralName == null ? table.name : table.pluralName;
            pluralName = changeCase.pascalCase(pluralName);
            if (informationalClass == null) {
                informationalClass = new information_architecture_1.Class(model.fullConstantCaseName, typeName, table.description, SystemComponentType_1.SystemComponentType.informationModel);
                informationalClass.pluralName = pluralName;
                informationalClass.orMapping = mapping;
                informationalClass.primitive = false;
                informationalClass.permanence = Permanence_1.Permanence.persistent;
                informationalClass.informational = true;
                informationalClass.functional = false;
                informationalClass.origin = ComponentOrigin_1.ComponentOrigin.manufactured;
                model.addChild(informationalClass);
                results.push(new SystemComponentArtifact_1.SystemComponentArtifact(informationalClass));
                mapping.type = informationalClass;
            }
            this.logger.isLevelEnabled('debug') ? this.logger.debug(`translateTableToClass(${table.constantCaseFullName}) exit`) : '';
        }
        return results;
    }
}
exports.OrmTableToClassHarvester = OrmTableToClassHarvester;
