// @harvest-forward/runtime/dist/src/plugins/orm-basic/builders/OrmViewToClassHarvester.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrmViewToClassHarvester = void 0;
const changeCase = __importStar(require("change-case"));
const AbstractSingularBuilder_1 = require("../../../runtime/AbstractSingularBuilder");
const ComponentOrigin_1 = require("../../../system/ComponentOrigin");
const Permanence_1 = require("../../../system/Permanence");
const SystemComponent_1 = require("../../../system/SystemComponent");
const SystemComponentArtifact_1 = require("../../../system/SystemComponentArtifact");
const SystemComponentType_1 = require("../../../system/SystemComponentType");
const ValueType_1 = require("../../../system/ValueType");
const information_architecture_1 = require("../../information-architecture");
const DataMemberColumnMapping_1 = require("../DataMemberColumnMapping");
const ObjectTypeTableMapping_1 = require("../ObjectTypeTableMapping");
class OrmViewToClassHarvester extends AbstractSingularBuilder_1.AbstractSingularBuilder {
    constructor(configurationValues) {
        const nonNullConfigurationValues = configurationValues ? configurationValues : [];
        nonNullConfigurationValues['repeatFor'] = JSON.stringify({
            objectTypeName: 'View',
        });
        super('OrmViewToClassHarvester', 'A harvester that takes views from a sql schema and creates classes in an information model', {
            repeatFor: {
                name: 'Repeat for',
                required: true,
                description: 'A parsable string of JSON that represents the properties values of the component that should be iterated on, review the documentation for SystemDescendantComponent and derivations for most available for use',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
        }, nonNullConfigurationValues);
    }
    async buildInternal(system, component) {
        const results = [];
        const view = component;
        let viewMapping = view.orMapping;
        let typeName;
        if (viewMapping != null) {
            typeName = viewMapping.typeName;
            viewMapping.storage = view;
        }
        if (typeName == null || typeName == '')
            typeName = view.singularName == null ? view.name : view.singularName;
        typeName = changeCase.pascalCase(typeName);
        let pluralName = view.pluralName == null ? view.name : view.pluralName;
        pluralName = changeCase.pascalCase(pluralName);
        const model = information_architecture_1.InformationModel.fromSystem(system);
        const fullName = SystemComponent_1.SystemComponent.fullConstantCase(model.fullConstantCaseName, typeName);
        const valueType = system.descendants[fullName];
        if (valueType instanceof information_architecture_1.Class || valueType == null) {
            let informationalClass = valueType;
            const schema = view.parent;
            const db = schema.parent;
            const orm = system.findOneComponent({
                objectTypeName: 'ObjectRelationalMapping',
            });
            viewMapping = orm.tableNameToMapping[view.constantCaseFullName];
            if (viewMapping == null) {
                viewMapping = new ObjectTypeTableMapping_1.ObjectTypeTableMapping(model.fullConstantCaseName, typeName, db.fullConstantCaseName, schema.name, view.name);
                viewMapping.permanence = Permanence_1.Permanence.constant;
                viewMapping.informational = true;
                viewMapping.functional = false;
                viewMapping.origin = ComponentOrigin_1.ComponentOrigin.manufactured;
                orm.addChild(viewMapping);
                results.push(new SystemComponentArtifact_1.SystemComponentArtifact(viewMapping));
                viewMapping.storage = view;
                viewMapping.type = informationalClass;
            }
            else {
                viewMapping.typeName = typeName;
            }
            ;
            view.orMapping = viewMapping;
            if (informationalClass == null) {
                informationalClass = new information_architecture_1.Class(model.fullConstantCaseName, typeName, view.description, SystemComponentType_1.SystemComponentType.informationModel);
                informationalClass.pluralName = pluralName;
                informationalClass.orMapping = viewMapping;
                informationalClass.primitive = false;
                informationalClass.updateable = false;
                informationalClass.permanence = Permanence_1.Permanence.persistent;
                informationalClass.informational = true;
                informationalClass.functional = false;
                informationalClass.origin = ComponentOrigin_1.ComponentOrigin.manufactured;
                let pkColumn = view.columns['ID'];
                Object.values(view.columns).forEach((column) => {
                    if (column.name.toUpperCase() == 'ID')
                        pkColumn = column;
                });
                let pkProperty;
                if (pkColumn != null) {
                    const mappable = pkColumn;
                    if (mappable.orMapping?.property == null) {
                        const pkPropertyName = 'id';
                        const pkPropertyType = ValueType_1.ValueType.INT;
                        const pkPropertyDescription = 'Unique identifier for each instance';
                        pkProperty = new information_architecture_1.Property(informationalClass.constantCaseFullName, pkPropertyName, pkPropertyDescription, pkPropertyType, 0);
                        informationalClass.addChild(pkProperty);
                        results.push(new SystemComponentArtifact_1.SystemComponentArtifact(pkProperty));
                    }
                    else
                        pkProperty = mappable.orMapping.property;
                    const dataMember = pkProperty;
                    if (mappable.orMapping == null) {
                        dataMember.orMapping = new DataMemberColumnMapping_1.DataMemberColumnMapping(pkProperty.name, pkColumn.name, pkProperty.type.name, pkProperty.type);
                        mappable.orMapping = dataMember.orMapping;
                        dataMember.orMapping.column = pkColumn;
                        dataMember.orMapping.property = pkProperty;
                        dataMember.orMapping.permanence = Permanence_1.Permanence.constant;
                        dataMember.orMapping.informational = true;
                        dataMember.orMapping.functional = false;
                        dataMember.orMapping.origin = ComponentOrigin_1.ComponentOrigin.manufactured;
                        viewMapping.addChild(dataMember.orMapping);
                        results.push(new SystemComponentArtifact_1.SystemComponentArtifact(dataMember.orMapping));
                    }
                    else
                        dataMember.orMapping = mappable.orMapping;
                    informationalClass.identifyingProperty = pkProperty;
                }
                model.addChild(informationalClass);
                results.push(new SystemComponentArtifact_1.SystemComponentArtifact(informationalClass));
                viewMapping.type = informationalClass;
            }
        }
        return results;
    }
}
exports.OrmViewToClassHarvester = OrmViewToClassHarvester;
