// @harvest-forward/runtime/dist/src/plugins/rdbms-basic/ForeignKey.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForeignKey = void 0;
const System_1 = require("../../system/System");
const SystemComponentType_1 = require("../../system/SystemComponentType");
const RelationalComponent_1 = require("./RelationalComponent");
class ForeignKey extends RelationalComponent_1.RelationalComponent {
    constructor(sourceSchemaName, sourceTableName, name, sourceColumnName, targetSchemaName, targetTableName, description) {
        super('ForeignKey', System_1.System.fullConstantCase(sourceSchemaName, sourceTableName), sourceSchemaName, name, SystemComponentType_1.SystemComponentType.storage, description);
        this.sourceTableName = sourceTableName;
        this.sourceColumnName = sourceColumnName;
        this.targetSchemaName = targetSchemaName;
        this.targetTableName = targetTableName;
        this.sourceTableFullName = System_1.System.fullConstantCase(sourceSchemaName, sourceTableName);
        this.targetTableFullName = System_1.System.fullConstantCase(targetSchemaName, targetTableName);
    }
}
exports.ForeignKey = ForeignKey;
