// @harvest-forward/runtime/dist/src/plugins/rdbms-basic/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.typeResolvers = exports.builders = void 0;
const RdbmsColumnStitcher_1 = require("./builders/RdbmsColumnStitcher");
const RdbmsConstraintStitcher_1 = require("./builders/RdbmsConstraintStitcher");
const RdbmsFkStitcher_1 = require("./builders/RdbmsFkStitcher");
const RdbmsPkStitcher_1 = require("./builders/RdbmsPkStitcher");
const Sql92Types_1 = require("./Sql92Types");
exports.builders = {
    ColumnStitcher: (settings) => new RdbmsColumnStitcher_1.RdbmsColumnStitcher(settings),
    ConstraintStitcher: (settings) => new RdbmsConstraintStitcher_1.RdbmsConstraintStitcher(settings),
    FkStitcher: (settings) => new RdbmsFkStitcher_1.RdbmsFkStitcher(settings),
    PkStitcher: (settings) => new RdbmsPkStitcher_1.RdbmsPkStitcher(settings),
};
exports.typeResolvers = { Sql: Sql92Types_1.Sql92ValueTypeResolver };
__exportStar(require("./builders/AbstractSqlSchemaHarvester"), exports);
__exportStar(require("./builders/RdbmsColumnStitcher"), exports);
__exportStar(require("./builders/RdbmsConstraintStitcher"), exports);
__exportStar(require("./builders/RdbmsFkStitcher"), exports);
__exportStar(require("./builders/RdbmsPkStitcher"), exports);
__exportStar(require("./Column"), exports);
__exportStar(require("./Constraint"), exports);
__exportStar(require("./Domain"), exports);
__exportStar(require("./DomainValue"), exports);
__exportStar(require("./ForeignKey"), exports);
__exportStar(require("./PrimaryKey"), exports);
__exportStar(require("./RelationalComponent"), exports);
__exportStar(require("./RelationalDatabase"), exports);
__exportStar(require("./RelationalStorage"), exports);
__exportStar(require("./Schema"), exports);
__exportStar(require("./Sql92Types"), exports);
__exportStar(require("./Table"), exports);
__exportStar(require("./TableType"), exports);
__exportStar(require("./TableTypeInfo"), exports);
__exportStar(require("./View"), exports);
