// @harvest-forward/runtime/dist/src/plugins/python-bindings/grammar/Python3ParserListener.js
"use strict";
var antlr4 = require('antlr4/index');
function Python3ParserListener() {
    antlr4.tree.ParseTreeListener.call(this);
    return this;
}
Python3ParserListener.prototype = Object.create(antlr4.tree.ParseTreeListener.prototype);
Python3ParserListener.prototype.constructor = Python3ParserListener;
Python3ParserListener.prototype.enterSingle_input = function (ctx) {
};
Python3ParserListener.prototype.exitSingle_input = function (ctx) {
};
Python3ParserListener.prototype.enterFile_input = function (ctx) {
};
Python3ParserListener.prototype.exitFile_input = function (ctx) {
};
Python3ParserListener.prototype.enterEval_input = function (ctx) {
};
Python3ParserListener.prototype.exitEval_input = function (ctx) {
};
Python3ParserListener.prototype.enterDecorator = function (ctx) {
};
Python3ParserListener.prototype.exitDecorator = function (ctx) {
};
Python3ParserListener.prototype.enterDecorators = function (ctx) {
};
Python3ParserListener.prototype.exitDecorators = function (ctx) {
};
Python3ParserListener.prototype.enterDecorated = function (ctx) {
};
Python3ParserListener.prototype.exitDecorated = function (ctx) {
};
Python3ParserListener.prototype.enterAsync_funcdef = function (ctx) {
};
Python3ParserListener.prototype.exitAsync_funcdef = function (ctx) {
};
Python3ParserListener.prototype.enterFuncdef = function (ctx) {
};
Python3ParserListener.prototype.exitFuncdef = function (ctx) {
};
Python3ParserListener.prototype.enterParameters = function (ctx) {
};
Python3ParserListener.prototype.exitParameters = function (ctx) {
};
Python3ParserListener.prototype.enterTypedargslist = function (ctx) {
};
Python3ParserListener.prototype.exitTypedargslist = function (ctx) {
};
Python3ParserListener.prototype.enterTfpdef = function (ctx) {
};
Python3ParserListener.prototype.exitTfpdef = function (ctx) {
};
Python3ParserListener.prototype.enterVarargslist = function (ctx) {
};
Python3ParserListener.prototype.exitVarargslist = function (ctx) {
};
Python3ParserListener.prototype.enterVfpdef = function (ctx) {
};
Python3ParserListener.prototype.exitVfpdef = function (ctx) {
};
Python3ParserListener.prototype.enterStmt = function (ctx) {
};
Python3ParserListener.prototype.exitStmt = function (ctx) {
};
Python3ParserListener.prototype.enterSimple_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitSimple_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterSmall_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitSmall_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterExpr_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitExpr_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterAnnassign = function (ctx) {
};
Python3ParserListener.prototype.exitAnnassign = function (ctx) {
};
Python3ParserListener.prototype.enterTestlist_star_expr = function (ctx) {
};
Python3ParserListener.prototype.exitTestlist_star_expr = function (ctx) {
};
Python3ParserListener.prototype.enterAugassign = function (ctx) {
};
Python3ParserListener.prototype.exitAugassign = function (ctx) {
};
Python3ParserListener.prototype.enterDel_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitDel_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterPass_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitPass_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterFlow_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitFlow_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterBreak_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitBreak_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterContinue_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitContinue_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterReturn_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitReturn_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterYield_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitYield_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterRaise_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitRaise_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterImport_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitImport_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterImport_name = function (ctx) {
};
Python3ParserListener.prototype.exitImport_name = function (ctx) {
};
Python3ParserListener.prototype.enterImport_from = function (ctx) {
};
Python3ParserListener.prototype.exitImport_from = function (ctx) {
};
Python3ParserListener.prototype.enterImport_as_name = function (ctx) {
};
Python3ParserListener.prototype.exitImport_as_name = function (ctx) {
};
Python3ParserListener.prototype.enterDotted_as_name = function (ctx) {
};
Python3ParserListener.prototype.exitDotted_as_name = function (ctx) {
};
Python3ParserListener.prototype.enterImport_as_names = function (ctx) {
};
Python3ParserListener.prototype.exitImport_as_names = function (ctx) {
};
Python3ParserListener.prototype.enterDotted_as_names = function (ctx) {
};
Python3ParserListener.prototype.exitDotted_as_names = function (ctx) {
};
Python3ParserListener.prototype.enterDotted_name = function (ctx) {
};
Python3ParserListener.prototype.exitDotted_name = function (ctx) {
};
Python3ParserListener.prototype.enterGlobal_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitGlobal_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterNonlocal_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitNonlocal_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterAssert_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitAssert_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterCompound_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitCompound_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterAsync_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitAsync_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterIf_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitIf_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterWhile_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitWhile_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterFor_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitFor_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterTry_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitTry_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterWith_stmt = function (ctx) {
};
Python3ParserListener.prototype.exitWith_stmt = function (ctx) {
};
Python3ParserListener.prototype.enterWith_item = function (ctx) {
};
Python3ParserListener.prototype.exitWith_item = function (ctx) {
};
Python3ParserListener.prototype.enterExcept_clause = function (ctx) {
};
Python3ParserListener.prototype.exitExcept_clause = function (ctx) {
};
Python3ParserListener.prototype.enterSuite = function (ctx) {
};
Python3ParserListener.prototype.exitSuite = function (ctx) {
};
Python3ParserListener.prototype.enterTest = function (ctx) {
};
Python3ParserListener.prototype.exitTest = function (ctx) {
};
Python3ParserListener.prototype.enterTest_nocond = function (ctx) {
};
Python3ParserListener.prototype.exitTest_nocond = function (ctx) {
};
Python3ParserListener.prototype.enterLambdef = function (ctx) {
};
Python3ParserListener.prototype.exitLambdef = function (ctx) {
};
Python3ParserListener.prototype.enterLambdef_nocond = function (ctx) {
};
Python3ParserListener.prototype.exitLambdef_nocond = function (ctx) {
};
Python3ParserListener.prototype.enterOr_test = function (ctx) {
};
Python3ParserListener.prototype.exitOr_test = function (ctx) {
};
Python3ParserListener.prototype.enterAnd_test = function (ctx) {
};
Python3ParserListener.prototype.exitAnd_test = function (ctx) {
};
Python3ParserListener.prototype.enterNot_test = function (ctx) {
};
Python3ParserListener.prototype.exitNot_test = function (ctx) {
};
Python3ParserListener.prototype.enterComparison = function (ctx) {
};
Python3ParserListener.prototype.exitComparison = function (ctx) {
};
Python3ParserListener.prototype.enterComp_op = function (ctx) {
};
Python3ParserListener.prototype.exitComp_op = function (ctx) {
};
Python3ParserListener.prototype.enterStar_expr = function (ctx) {
};
Python3ParserListener.prototype.exitStar_expr = function (ctx) {
};
Python3ParserListener.prototype.enterExpr = function (ctx) {
};
Python3ParserListener.prototype.exitExpr = function (ctx) {
};
Python3ParserListener.prototype.enterXor_expr = function (ctx) {
};
Python3ParserListener.prototype.exitXor_expr = function (ctx) {
};
Python3ParserListener.prototype.enterAnd_expr = function (ctx) {
};
Python3ParserListener.prototype.exitAnd_expr = function (ctx) {
};
Python3ParserListener.prototype.enterShift_expr = function (ctx) {
};
Python3ParserListener.prototype.exitShift_expr = function (ctx) {
};
Python3ParserListener.prototype.enterArith_expr = function (ctx) {
};
Python3ParserListener.prototype.exitArith_expr = function (ctx) {
};
Python3ParserListener.prototype.enterTerm = function (ctx) {
};
Python3ParserListener.prototype.exitTerm = function (ctx) {
};
Python3ParserListener.prototype.enterFactor = function (ctx) {
};
Python3ParserListener.prototype.exitFactor = function (ctx) {
};
Python3ParserListener.prototype.enterPower = function (ctx) {
};
Python3ParserListener.prototype.exitPower = function (ctx) {
};
Python3ParserListener.prototype.enterAtom_expr = function (ctx) {
};
Python3ParserListener.prototype.exitAtom_expr = function (ctx) {
};
Python3ParserListener.prototype.enterAtom = function (ctx) {
};
Python3ParserListener.prototype.exitAtom = function (ctx) {
};
Python3ParserListener.prototype.enterTestlist_comp = function (ctx) {
};
Python3ParserListener.prototype.exitTestlist_comp = function (ctx) {
};
Python3ParserListener.prototype.enterTrailer = function (ctx) {
};
Python3ParserListener.prototype.exitTrailer = function (ctx) {
};
Python3ParserListener.prototype.enterSubscriptlist = function (ctx) {
};
Python3ParserListener.prototype.exitSubscriptlist = function (ctx) {
};
Python3ParserListener.prototype.enterSubscript = function (ctx) {
};
Python3ParserListener.prototype.exitSubscript = function (ctx) {
};
Python3ParserListener.prototype.enterSliceop = function (ctx) {
};
Python3ParserListener.prototype.exitSliceop = function (ctx) {
};
Python3ParserListener.prototype.enterExprlist = function (ctx) {
};
Python3ParserListener.prototype.exitExprlist = function (ctx) {
};
Python3ParserListener.prototype.enterTestlist = function (ctx) {
};
Python3ParserListener.prototype.exitTestlist = function (ctx) {
};
Python3ParserListener.prototype.enterDictorsetmaker = function (ctx) {
};
Python3ParserListener.prototype.exitDictorsetmaker = function (ctx) {
};
Python3ParserListener.prototype.enterClassdef = function (ctx) {
};
Python3ParserListener.prototype.exitClassdef = function (ctx) {
};
Python3ParserListener.prototype.enterArglist = function (ctx) {
};
Python3ParserListener.prototype.exitArglist = function (ctx) {
};
Python3ParserListener.prototype.enterArgument = function (ctx) {
};
Python3ParserListener.prototype.exitArgument = function (ctx) {
};
Python3ParserListener.prototype.enterComp_iter = function (ctx) {
};
Python3ParserListener.prototype.exitComp_iter = function (ctx) {
};
Python3ParserListener.prototype.enterComp_for = function (ctx) {
};
Python3ParserListener.prototype.exitComp_for = function (ctx) {
};
Python3ParserListener.prototype.enterComp_if = function (ctx) {
};
Python3ParserListener.prototype.exitComp_if = function (ctx) {
};
Python3ParserListener.prototype.enterEncoding_decl = function (ctx) {
};
Python3ParserListener.prototype.exitEncoding_decl = function (ctx) {
};
Python3ParserListener.prototype.enterYield_expr = function (ctx) {
};
Python3ParserListener.prototype.exitYield_expr = function (ctx) {
};
Python3ParserListener.prototype.enterYield_arg = function (ctx) {
};
Python3ParserListener.prototype.exitYield_arg = function (ctx) {
};
exports.Python3ParserListener = Python3ParserListener;
