// @harvest-forward/runtime/dist/src/system/SystemComponentType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemComponentType = void 0;
var SystemComponentType;
(function (SystemComponentType) {
    SystemComponentType["system"] = "system";
    SystemComponentType["storage"] = "storage";
    SystemComponentType["storageProjection"] = "storageProjection";
    SystemComponentType["informationModel"] = "informationModel";
    SystemComponentType["informationModelMapping"] = "informationModelMapping";
    SystemComponentType["informationModelProjection"] = "informationModelProjection";
    SystemComponentType["informationModelProjectionMapping"] = "informationModelProjectionMapping";
    SystemComponentType["dataAccess"] = "dataAccess";
    SystemComponentType["service"] = "service";
    SystemComponentType["serviceEndpoint"] = "serviceEndpoint";
    SystemComponentType["clientApiProxy"] = "clientApiProxy";
    SystemComponentType["clientApplication"] = "clientApplication";
    SystemComponentType["visualization"] = "visualization";
    SystemComponentType["accessControlMapping"] = "accessControlMapping";
    SystemComponentType["validationRule"] = "validationRule";
})(SystemComponentType = exports.SystemComponentType || (exports.SystemComponentType = {}));
