// @harvest-forward/runtime/dist/src/plugins/application-architecture/builders/DalToServiceTransformer.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DalToServiceTransformer = void 0;
const pluralize_1 = __importDefault(require("pluralize"));
const AbstractSingularBuilder_1 = require("../../../runtime/AbstractSingularBuilder");
const ComponentOrigin_1 = require("../../../system/ComponentOrigin");
const Permanence_1 = require("../../../system/Permanence");
const SystemComponentArtifact_1 = require("../../../system/SystemComponentArtifact");
const SystemComponentType_1 = require("../../../system/SystemComponentType");
const ValueType_1 = require("../../../system/ValueType");
const information_architecture_1 = require("../../information-architecture");
class DalToServiceTransformer extends AbstractSingularBuilder_1.AbstractSingularBuilder {
    constructor(configurationValues) {
        const nonNullConfigurationValues = configurationValues ? configurationValues : [];
        nonNullConfigurationValues['repeatFor'] = JSON.stringify({
            objectTypeName: 'Class',
            componentType: 'dataAccess',
        });
        super('DalToServiceTransformer', 'A transformer that creates a service layer based on the information model and data access layer', {
            repeatFor: {
                name: 'Repeat for',
                required: true,
                description: 'A parsable string of JSON that represents the properties values of the component that should be iterated on, review the documentation for SystemDescendantComponent and derivations for most available for use',
                valueType: ValueType_1.ValueType.STRING,
                defaultValue: ``,
            },
        }, nonNullConfigurationValues);
    }
    async buildInternal(system, component) {
        const dalClass = component;
        const bllName = `${dalClass.name.replace(/Dal$/, '')}Service`;
        const bllDescription = `A class to performs actions against the model class ${dalClass.name}. Many actions expose access to the data itself`;
        const bllClass = new information_architecture_1.Class('service', bllName, bllDescription, SystemComponentType_1.SystemComponentType.service);
        bllClass.singularName = bllName;
        bllClass.pluralName = pluralize_1.default.plural(dalClass.name);
        bllClass.operatesOnBehalfOf = dalClass.operatesOnBehalfOf;
        bllClass.permanence = Permanence_1.Permanence.constant;
        bllClass.informational = false;
        bllClass.functional = true;
        bllClass.origin = ComponentOrigin_1.ComponentOrigin.manufactured;
        system.addChild(bllClass);
        return [new SystemComponentArtifact_1.SystemComponentArtifact(bllClass)];
    }
}
exports.DalToServiceTransformer = DalToServiceTransformer;
