// @harvest-forward/runtime/dist/src/plugins/php-bindings/grammar/PhpParserListener.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const antlr4_1 = __importDefault(require("antlr4"));
class PhpParserListener extends antlr4_1.default.tree.ParseTreeListener {
    enterHtmlDocument(ctx) {
    }
    exitHtmlDocument(ctx) {
    }
    enterInlineHtml(ctx) {
    }
    exitInlineHtml(ctx) {
    }
    enterHtmlElement(ctx) {
    }
    exitHtmlElement(ctx) {
    }
    enterScriptText(ctx) {
    }
    exitScriptText(ctx) {
    }
    enterPhpBlock(ctx) {
    }
    exitPhpBlock(ctx) {
    }
    enterImportStatement(ctx) {
    }
    exitImportStatement(ctx) {
    }
    enterTopStatement(ctx) {
    }
    exitTopStatement(ctx) {
    }
    enterUseDeclaration(ctx) {
    }
    exitUseDeclaration(ctx) {
    }
    enterUseDeclarationContentList(ctx) {
    }
    exitUseDeclarationContentList(ctx) {
    }
    enterUseDeclarationContent(ctx) {
    }
    exitUseDeclarationContent(ctx) {
    }
    enterNamespaceDeclaration(ctx) {
    }
    exitNamespaceDeclaration(ctx) {
    }
    enterNamespaceStatement(ctx) {
    }
    exitNamespaceStatement(ctx) {
    }
    enterFunctionDeclaration(ctx) {
    }
    exitFunctionDeclaration(ctx) {
    }
    enterClassDeclaration(ctx) {
    }
    exitClassDeclaration(ctx) {
    }
    enterClassEntryType(ctx) {
    }
    exitClassEntryType(ctx) {
    }
    enterInterfaceList(ctx) {
    }
    exitInterfaceList(ctx) {
    }
    enterTypeParameterListInBrackets(ctx) {
    }
    exitTypeParameterListInBrackets(ctx) {
    }
    enterTypeParameterList(ctx) {
    }
    exitTypeParameterList(ctx) {
    }
    enterTypeParameterWithDefaultsList(ctx) {
    }
    exitTypeParameterWithDefaultsList(ctx) {
    }
    enterTypeParameterDecl(ctx) {
    }
    exitTypeParameterDecl(ctx) {
    }
    enterTypeParameterWithDefaultDecl(ctx) {
    }
    exitTypeParameterWithDefaultDecl(ctx) {
    }
    enterGenericDynamicArgs(ctx) {
    }
    exitGenericDynamicArgs(ctx) {
    }
    enterAttributes(ctx) {
    }
    exitAttributes(ctx) {
    }
    enterAttributeGroup(ctx) {
    }
    exitAttributeGroup(ctx) {
    }
    enterAttribute(ctx) {
    }
    exitAttribute(ctx) {
    }
    enterInnerStatementList(ctx) {
    }
    exitInnerStatementList(ctx) {
    }
    enterInnerStatement(ctx) {
    }
    exitInnerStatement(ctx) {
    }
    enterStatement(ctx) {
    }
    exitStatement(ctx) {
    }
    enterEmptyStatement(ctx) {
    }
    exitEmptyStatement(ctx) {
    }
    enterBlockStatement(ctx) {
    }
    exitBlockStatement(ctx) {
    }
    enterIfStatement(ctx) {
    }
    exitIfStatement(ctx) {
    }
    enterElseIfStatement(ctx) {
    }
    exitElseIfStatement(ctx) {
    }
    enterElseIfColonStatement(ctx) {
    }
    exitElseIfColonStatement(ctx) {
    }
    enterElseStatement(ctx) {
    }
    exitElseStatement(ctx) {
    }
    enterElseColonStatement(ctx) {
    }
    exitElseColonStatement(ctx) {
    }
    enterWhileStatement(ctx) {
    }
    exitWhileStatement(ctx) {
    }
    enterDoWhileStatement(ctx) {
    }
    exitDoWhileStatement(ctx) {
    }
    enterForStatement(ctx) {
    }
    exitForStatement(ctx) {
    }
    enterForInit(ctx) {
    }
    exitForInit(ctx) {
    }
    enterForUpdate(ctx) {
    }
    exitForUpdate(ctx) {
    }
    enterSwitchStatement(ctx) {
    }
    exitSwitchStatement(ctx) {
    }
    enterSwitchBlock(ctx) {
    }
    exitSwitchBlock(ctx) {
    }
    enterBreakStatement(ctx) {
    }
    exitBreakStatement(ctx) {
    }
    enterContinueStatement(ctx) {
    }
    exitContinueStatement(ctx) {
    }
    enterReturnStatement(ctx) {
    }
    exitReturnStatement(ctx) {
    }
    enterExpressionStatement(ctx) {
    }
    exitExpressionStatement(ctx) {
    }
    enterUnsetStatement(ctx) {
    }
    exitUnsetStatement(ctx) {
    }
    enterForeachStatement(ctx) {
    }
    exitForeachStatement(ctx) {
    }
    enterTryCatchFinally(ctx) {
    }
    exitTryCatchFinally(ctx) {
    }
    enterCatchClause(ctx) {
    }
    exitCatchClause(ctx) {
    }
    enterFinallyStatement(ctx) {
    }
    exitFinallyStatement(ctx) {
    }
    enterThrowStatement(ctx) {
    }
    exitThrowStatement(ctx) {
    }
    enterGotoStatement(ctx) {
    }
    exitGotoStatement(ctx) {
    }
    enterDeclareStatement(ctx) {
    }
    exitDeclareStatement(ctx) {
    }
    enterInlineHtmlStatement(ctx) {
    }
    exitInlineHtmlStatement(ctx) {
    }
    enterDeclareList(ctx) {
    }
    exitDeclareList(ctx) {
    }
    enterFormalParameterList(ctx) {
    }
    exitFormalParameterList(ctx) {
    }
    enterFormalParameter(ctx) {
    }
    exitFormalParameter(ctx) {
    }
    enterTypeHint(ctx) {
    }
    exitTypeHint(ctx) {
    }
    enterGlobalStatement(ctx) {
    }
    exitGlobalStatement(ctx) {
    }
    enterGlobalVar(ctx) {
    }
    exitGlobalVar(ctx) {
    }
    enterEchoStatement(ctx) {
    }
    exitEchoStatement(ctx) {
    }
    enterStaticVariableStatement(ctx) {
    }
    exitStaticVariableStatement(ctx) {
    }
    enterClassStatement(ctx) {
    }
    exitClassStatement(ctx) {
    }
    enterTraitAdaptations(ctx) {
    }
    exitTraitAdaptations(ctx) {
    }
    enterTraitAdaptationStatement(ctx) {
    }
    exitTraitAdaptationStatement(ctx) {
    }
    enterTraitPrecedence(ctx) {
    }
    exitTraitPrecedence(ctx) {
    }
    enterTraitAlias(ctx) {
    }
    exitTraitAlias(ctx) {
    }
    enterTraitMethodReference(ctx) {
    }
    exitTraitMethodReference(ctx) {
    }
    enterBaseCtorCall(ctx) {
    }
    exitBaseCtorCall(ctx) {
    }
    enterMethodBody(ctx) {
    }
    exitMethodBody(ctx) {
    }
    enterPropertyModifiers(ctx) {
    }
    exitPropertyModifiers(ctx) {
    }
    enterMemberModifiers(ctx) {
    }
    exitMemberModifiers(ctx) {
    }
    enterVariableInitializer(ctx) {
    }
    exitVariableInitializer(ctx) {
    }
    enterIdentifierInitializer(ctx) {
    }
    exitIdentifierInitializer(ctx) {
    }
    enterGlobalConstantDeclaration(ctx) {
    }
    exitGlobalConstantDeclaration(ctx) {
    }
    enterExpressionList(ctx) {
    }
    exitExpressionList(ctx) {
    }
    enterParentheses(ctx) {
    }
    exitParentheses(ctx) {
    }
    enterChainExpression(ctx) {
    }
    exitChainExpression(ctx) {
    }
    enterSpecialWordExpression(ctx) {
    }
    exitSpecialWordExpression(ctx) {
    }
    enterArrayCreationExpression(ctx) {
    }
    exitArrayCreationExpression(ctx) {
    }
    enterBackQuoteStringExpression(ctx) {
    }
    exitBackQuoteStringExpression(ctx) {
    }
    enterMatchExpression(ctx) {
    }
    exitMatchExpression(ctx) {
    }
    enterLogicalExpression(ctx) {
    }
    exitLogicalExpression(ctx) {
    }
    enterPrintExpression(ctx) {
    }
    exitPrintExpression(ctx) {
    }
    enterAssignmentExpression(ctx) {
    }
    exitAssignmentExpression(ctx) {
    }
    enterPostfixIncDecExpression(ctx) {
    }
    exitPostfixIncDecExpression(ctx) {
    }
    enterCloneExpression(ctx) {
    }
    exitCloneExpression(ctx) {
    }
    enterUnaryOperatorExpression(ctx) {
    }
    exitUnaryOperatorExpression(ctx) {
    }
    enterNewExpression(ctx) {
    }
    exitNewExpression(ctx) {
    }
    enterParenthesisExpression(ctx) {
    }
    exitParenthesisExpression(ctx) {
    }
    enterSpaceshipExpression(ctx) {
    }
    exitSpaceshipExpression(ctx) {
    }
    enterConditionalExpression(ctx) {
    }
    exitConditionalExpression(ctx) {
    }
    enterNullCoalescingExpression(ctx) {
    }
    exitNullCoalescingExpression(ctx) {
    }
    enterArithmeticExpression(ctx) {
    }
    exitArithmeticExpression(ctx) {
    }
    enterIndexerExpression(ctx) {
    }
    exitIndexerExpression(ctx) {
    }
    enterScalarExpression(ctx) {
    }
    exitScalarExpression(ctx) {
    }
    enterPrefixIncDecExpression(ctx) {
    }
    exitPrefixIncDecExpression(ctx) {
    }
    enterComparisonExpression(ctx) {
    }
    exitComparisonExpression(ctx) {
    }
    enterCastExpression(ctx) {
    }
    exitCastExpression(ctx) {
    }
    enterInstanceOfExpression(ctx) {
    }
    exitInstanceOfExpression(ctx) {
    }
    enterLambdaFunctionExpression(ctx) {
    }
    exitLambdaFunctionExpression(ctx) {
    }
    enterBitwiseExpression(ctx) {
    }
    exitBitwiseExpression(ctx) {
    }
    enterAssignable(ctx) {
    }
    exitAssignable(ctx) {
    }
    enterArrayCreation(ctx) {
    }
    exitArrayCreation(ctx) {
    }
    enterLambdaFunctionExpr(ctx) {
    }
    exitLambdaFunctionExpr(ctx) {
    }
    enterMatchExpr(ctx) {
    }
    exitMatchExpr(ctx) {
    }
    enterMatchItem(ctx) {
    }
    exitMatchItem(ctx) {
    }
    enterNewExpr(ctx) {
    }
    exitNewExpr(ctx) {
    }
    enterAssignmentOperator(ctx) {
    }
    exitAssignmentOperator(ctx) {
    }
    enterYieldExpression(ctx) {
    }
    exitYieldExpression(ctx) {
    }
    enterArrayItemList(ctx) {
    }
    exitArrayItemList(ctx) {
    }
    enterArrayItem(ctx) {
    }
    exitArrayItem(ctx) {
    }
    enterLambdaFunctionUseVars(ctx) {
    }
    exitLambdaFunctionUseVars(ctx) {
    }
    enterLambdaFunctionUseVar(ctx) {
    }
    exitLambdaFunctionUseVar(ctx) {
    }
    enterQualifiedStaticTypeRef(ctx) {
    }
    exitQualifiedStaticTypeRef(ctx) {
    }
    enterTypeRef(ctx) {
    }
    exitTypeRef(ctx) {
    }
    enterAnonymousClass(ctx) {
    }
    exitAnonymousClass(ctx) {
    }
    enterIndirectTypeRef(ctx) {
    }
    exitIndirectTypeRef(ctx) {
    }
    enterQualifiedNamespaceName(ctx) {
    }
    exitQualifiedNamespaceName(ctx) {
    }
    enterNamespaceNameList(ctx) {
    }
    exitNamespaceNameList(ctx) {
    }
    enterNamespaceNameTail(ctx) {
    }
    exitNamespaceNameTail(ctx) {
    }
    enterQualifiedNamespaceNameList(ctx) {
    }
    exitQualifiedNamespaceNameList(ctx) {
    }
    enterArguments(ctx) {
    }
    exitArguments(ctx) {
    }
    enterActualArgument(ctx) {
    }
    exitActualArgument(ctx) {
    }
    enterArgumentName(ctx) {
    }
    exitArgumentName(ctx) {
    }
    enterConstantInitializer(ctx) {
    }
    exitConstantInitializer(ctx) {
    }
    enterConstant(ctx) {
    }
    exitConstant(ctx) {
    }
    enterLiteralConstant(ctx) {
    }
    exitLiteralConstant(ctx) {
    }
    enterNumericConstant(ctx) {
    }
    exitNumericConstant(ctx) {
    }
    enterClassConstant(ctx) {
    }
    exitClassConstant(ctx) {
    }
    enterStringConstant(ctx) {
    }
    exitStringConstant(ctx) {
    }
    enterString(ctx) {
    }
    exitString(ctx) {
    }
    enterInterpolatedStringPart(ctx) {
    }
    exitInterpolatedStringPart(ctx) {
    }
    enterChainList(ctx) {
    }
    exitChainList(ctx) {
    }
    enterChain(ctx) {
    }
    exitChain(ctx) {
    }
    enterChainOrigin(ctx) {
    }
    exitChainOrigin(ctx) {
    }
    enterMemberAccess(ctx) {
    }
    exitMemberAccess(ctx) {
    }
    enterFunctionCall(ctx) {
    }
    exitFunctionCall(ctx) {
    }
    enterFunctionCallName(ctx) {
    }
    exitFunctionCallName(ctx) {
    }
    enterActualArguments(ctx) {
    }
    exitActualArguments(ctx) {
    }
    enterChainBase(ctx) {
    }
    exitChainBase(ctx) {
    }
    enterKeyedFieldName(ctx) {
    }
    exitKeyedFieldName(ctx) {
    }
    enterKeyedSimpleFieldName(ctx) {
    }
    exitKeyedSimpleFieldName(ctx) {
    }
    enterKeyedVariable(ctx) {
    }
    exitKeyedVariable(ctx) {
    }
    enterSquareCurlyExpression(ctx) {
    }
    exitSquareCurlyExpression(ctx) {
    }
    enterAssignmentList(ctx) {
    }
    exitAssignmentList(ctx) {
    }
    enterAssignmentListElement(ctx) {
    }
    exitAssignmentListElement(ctx) {
    }
    enterModifier(ctx) {
    }
    exitModifier(ctx) {
    }
    enterIdentifier(ctx) {
    }
    exitIdentifier(ctx) {
    }
    enterMemberModifier(ctx) {
    }
    exitMemberModifier(ctx) {
    }
    enterMagicConstant(ctx) {
    }
    exitMagicConstant(ctx) {
    }
    enterMagicMethod(ctx) {
    }
    exitMagicMethod(ctx) {
    }
    enterPrimitiveType(ctx) {
    }
    exitPrimitiveType(ctx) {
    }
    enterCastOperation(ctx) {
    }
    exitCastOperation(ctx) {
    }
}
exports.default = PhpParserListener;
