// @harvest-forward/runtime/dist/src/plugins/vbnet-bindings/grammar/vbParserListener.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const antlr4_1 = __importDefault(require("antlr4"));
class vbParserListener extends antlr4_1.default.tree.ParseTreeListener {
    enterStart(ctx) {
    }
    exitStart(ctx) {
    }
    enterHexDigit(ctx) {
    }
    exitHexDigit(ctx) {
    }
    enterDecimalDigit(ctx) {
    }
    exitDecimalDigit(ctx) {
    }
    enterOctalDigit(ctx) {
    }
    exitOctalDigit(ctx) {
    }
    enterAlphaCharacter(ctx) {
    }
    exitAlphaCharacter(ctx) {
    }
    enterBooleanLiteral(ctx) {
    }
    exitBooleanLiteral(ctx) {
    }
    enterIntLiteral(ctx) {
    }
    exitIntLiteral(ctx) {
    }
    enterHexLiteral(ctx) {
    }
    exitHexLiteral(ctx) {
    }
    enterOctalLiteral(ctx) {
    }
    exitOctalLiteral(ctx) {
    }
    enterIntegerLiteral(ctx) {
    }
    exitIntegerLiteral(ctx) {
    }
    enterIntegerLiteralValue(ctx) {
    }
    exitIntegerLiteralValue(ctx) {
    }
    enterFloatingPointLiteral(ctx) {
    }
    exitFloatingPointLiteral(ctx) {
    }
    enterFloatingPointLiteralValue(ctx) {
    }
    exitFloatingPointLiteralValue(ctx) {
    }
    enterExponent(ctx) {
    }
    exitExponent(ctx) {
    }
    enterSign(ctx) {
    }
    exitSign(ctx) {
    }
    enterSingleTypeCharacter(ctx) {
    }
    exitSingleTypeCharacter(ctx) {
    }
    enterDoubleTypeCharacter(ctx) {
    }
    exitDoubleTypeCharacter(ctx) {
    }
    enterLongTypeCharacter(ctx) {
    }
    exitLongTypeCharacter(ctx) {
    }
    enterStringTypeCharacter(ctx) {
    }
    exitStringTypeCharacter(ctx) {
    }
    enterIntegerTypeCharacter(ctx) {
    }
    exitIntegerTypeCharacter(ctx) {
    }
    enterDecimalTypeCharacter(ctx) {
    }
    exitDecimalTypeCharacter(ctx) {
    }
    enterIntegralTypeCharacter(ctx) {
    }
    exitIntegralTypeCharacter(ctx) {
    }
    enterFloatingPointTypeCharacter(ctx) {
    }
    exitFloatingPointTypeCharacter(ctx) {
    }
    enterMonthValue(ctx) {
    }
    exitMonthValue(ctx) {
    }
    enterDayValue(ctx) {
    }
    exitDayValue(ctx) {
    }
    enterYearValue(ctx) {
    }
    exitYearValue(ctx) {
    }
    enterHourValue(ctx) {
    }
    exitHourValue(ctx) {
    }
    enterMinuteValue(ctx) {
    }
    exitMinuteValue(ctx) {
    }
    enterSecondValue(ctx) {
    }
    exitSecondValue(ctx) {
    }
    enterAmpm(ctx) {
    }
    exitAmpm(ctx) {
    }
    enterDateValue(ctx) {
    }
    exitDateValue(ctx) {
    }
    enterTimeValue(ctx) {
    }
    exitTimeValue(ctx) {
    }
    enterDateOrTime(ctx) {
    }
    exitDateOrTime(ctx) {
    }
    enterDateLiteral(ctx) {
    }
    exitDateLiteral(ctx) {
    }
    enterStringCharacter(ctx) {
    }
    exitStringCharacter(ctx) {
    }
    enterStringLiteral(ctx) {
    }
    exitStringLiteral(ctx) {
    }
    enterCharacterLiteral(ctx) {
    }
    exitCharacterLiteral(ctx) {
    }
    enterIdentifierStart(ctx) {
    }
    exitIdentifierStart(ctx) {
    }
    enterIdentifierName(ctx) {
    }
    exitIdentifierName(ctx) {
    }
    enterIdentifierCharacter(ctx) {
    }
    exitIdentifierCharacter(ctx) {
    }
    enterIdentifier(ctx) {
    }
    exitIdentifier(ctx) {
    }
    enterIdentifierOrKeyword(ctx) {
    }
    exitIdentifierOrKeyword(ctx) {
    }
    enterTypeCharacter(ctx) {
    }
    exitTypeCharacter(ctx) {
    }
    enterLiteral(ctx) {
    }
    exitLiteral(ctx) {
    }
    enterElseIf(ctx) {
    }
    exitElseIf(ctx) {
    }
    enterCCStart(ctx) {
    }
    exitCCStart(ctx) {
    }
    enterCCStatement(ctx) {
    }
    exitCCStatement(ctx) {
    }
    enterCCExpression(ctx) {
    }
    exitCCExpression(ctx) {
    }
    enterCCParenthesizedExpression(ctx) {
    }
    exitCCParenthesizedExpression(ctx) {
    }
    enterCCSimpleNameExpression(ctx) {
    }
    exitCCSimpleNameExpression(ctx) {
    }
    enterCCCastExpression(ctx) {
    }
    exitCCCastExpression(ctx) {
    }
    enterCCPlusMinus(ctx) {
    }
    exitCCPlusMinus(ctx) {
    }
    enterCCUnaryOperator(ctx) {
    }
    exitCCUnaryOperator(ctx) {
    }
    enterCCBinaryOperator(ctx) {
    }
    exitCCBinaryOperator(ctx) {
    }
    enterCCConditionalExpression(ctx) {
    }
    exitCCConditionalExpression(ctx) {
    }
    enterCCConstantDeclaration(ctx) {
    }
    exitCCConstantDeclaration(ctx) {
    }
    enterCCIfGroup(ctx) {
    }
    exitCCIfGroup(ctx) {
    }
    enterCCElseIfGroup(ctx) {
    }
    exitCCElseIfGroup(ctx) {
    }
    enterCCElseGroup(ctx) {
    }
    exitCCElseGroup(ctx) {
    }
    enterRegionStart(ctx) {
    }
    exitRegionStart(ctx) {
    }
    enterRegionStatement(ctx) {
    }
    exitRegionStatement(ctx) {
    }
    enterRegionGroup(ctx) {
    }
    exitRegionGroup(ctx) {
    }
    enterAccessModifier(ctx) {
    }
    exitAccessModifier(ctx) {
    }
    enterTypeParameterList(ctx) {
    }
    exitTypeParameterList(ctx) {
    }
    enterTypeParameter(ctx) {
    }
    exitTypeParameter(ctx) {
    }
    enterVarianceModifier(ctx) {
    }
    exitVarianceModifier(ctx) {
    }
    enterTypeParameterConstraints(ctx) {
    }
    exitTypeParameterConstraints(ctx) {
    }
    enterConstraintList(ctx) {
    }
    exitConstraintList(ctx) {
    }
    enterConstraint(ctx) {
    }
    exitConstraint(ctx) {
    }
    enterAttributes(ctx) {
    }
    exitAttributes(ctx) {
    }
    enterAttributeBlock(ctx) {
    }
    exitAttributeBlock(ctx) {
    }
    enterAttributeList(ctx) {
    }
    exitAttributeList(ctx) {
    }
    enterAttribute(ctx) {
    }
    exitAttribute(ctx) {
    }
    enterAttributeModifier(ctx) {
    }
    exitAttributeModifier(ctx) {
    }
    enterAttributeArguments(ctx) {
    }
    exitAttributeArguments(ctx) {
    }
    enterAttributePositionalArgumentList(ctx) {
    }
    exitAttributePositionalArgumentList(ctx) {
    }
    enterVariablePropertyInitializerList(ctx) {
    }
    exitVariablePropertyInitializerList(ctx) {
    }
    enterVariablePropertyInitializer(ctx) {
    }
    exitVariablePropertyInitializer(ctx) {
    }
    enterAttributeArgumentExpression(ctx) {
    }
    exitAttributeArgumentExpression(ctx) {
    }
    enterStatementTerminator(ctx) {
    }
    exitStatementTerminator(ctx) {
    }
    enterAttributesStatement(ctx) {
    }
    exitAttributesStatement(ctx) {
    }
    enterOptionStatement(ctx) {
    }
    exitOptionStatement(ctx) {
    }
    enterOptionExplicitStatement(ctx) {
    }
    exitOptionExplicitStatement(ctx) {
    }
    enterOnOff(ctx) {
    }
    exitOnOff(ctx) {
    }
    enterOptionStrictStatement(ctx) {
    }
    exitOptionStrictStatement(ctx) {
    }
    enterOptionCompareStatement(ctx) {
    }
    exitOptionCompareStatement(ctx) {
    }
    enterCompareOption(ctx) {
    }
    exitCompareOption(ctx) {
    }
    enterOptionInferStatement(ctx) {
    }
    exitOptionInferStatement(ctx) {
    }
    enterImportsStatement(ctx) {
    }
    exitImportsStatement(ctx) {
    }
    enterImportsClauses(ctx) {
    }
    exitImportsClauses(ctx) {
    }
    enterImportsClause(ctx) {
    }
    exitImportsClause(ctx) {
    }
    enterAliasImportsClause(ctx) {
    }
    exitAliasImportsClause(ctx) {
    }
    enterMembersImportsClause(ctx) {
    }
    exitMembersImportsClause(ctx) {
    }
    enterXMLNamespaceImportsClause(ctx) {
    }
    exitXMLNamespaceImportsClause(ctx) {
    }
    enterXMLNamespaceValue(ctx) {
    }
    exitXMLNamespaceValue(ctx) {
    }
    enterNamespaceDeclaration(ctx) {
    }
    exitNamespaceDeclaration(ctx) {
    }
    enterNamespaceName(ctx) {
    }
    exitNamespaceName(ctx) {
    }
    enterRelativeNamespaceName(ctx) {
    }
    exitRelativeNamespaceName(ctx) {
    }
    enterNamespaceMemberDeclaration(ctx) {
    }
    exitNamespaceMemberDeclaration(ctx) {
    }
    enterTypeDeclaration(ctx) {
    }
    exitTypeDeclaration(ctx) {
    }
    enterNonModuleDeclaration(ctx) {
    }
    exitNonModuleDeclaration(ctx) {
    }
    enterTypeName(ctx) {
    }
    exitTypeName(ctx) {
    }
    enterNonArrayTypeName(ctx) {
    }
    exitNonArrayTypeName(ctx) {
    }
    enterSimpleTypeName(ctx) {
    }
    exitSimpleTypeName(ctx) {
    }
    enterQualifiedTypeName(ctx) {
    }
    exitQualifiedTypeName(ctx) {
    }
    enterTypeArguments(ctx) {
    }
    exitTypeArguments(ctx) {
    }
    enterTypeArgumentList(ctx) {
    }
    exitTypeArgumentList(ctx) {
    }
    enterBuiltInTypeName(ctx) {
    }
    exitBuiltInTypeName(ctx) {
    }
    enterTypeModifier(ctx) {
    }
    exitTypeModifier(ctx) {
    }
    enterIdentifierModifiers(ctx) {
    }
    exitIdentifierModifiers(ctx) {
    }
    enterNullableNameModifier(ctx) {
    }
    exitNullableNameModifier(ctx) {
    }
    enterTypeImplementsClause(ctx) {
    }
    exitTypeImplementsClause(ctx) {
    }
    enterTypeImplements(ctx) {
    }
    exitTypeImplements(ctx) {
    }
    enterPrimitiveTypeName(ctx) {
    }
    exitPrimitiveTypeName(ctx) {
    }
    enterNumericTypeName(ctx) {
    }
    exitNumericTypeName(ctx) {
    }
    enterIntegralTypeName(ctx) {
    }
    exitIntegralTypeName(ctx) {
    }
    enterFloatingPointTypeName(ctx) {
    }
    exitFloatingPointTypeName(ctx) {
    }
    enterEnumDeclaration(ctx) {
    }
    exitEnumDeclaration(ctx) {
    }
    enterEnumMemberDeclaration(ctx) {
    }
    exitEnumMemberDeclaration(ctx) {
    }
    enterClassDeclaration(ctx) {
    }
    exitClassDeclaration(ctx) {
    }
    enterClassModifier(ctx) {
    }
    exitClassModifier(ctx) {
    }
    enterClassBase(ctx) {
    }
    exitClassBase(ctx) {
    }
    enterClassMemberDeclaration(ctx) {
    }
    exitClassMemberDeclaration(ctx) {
    }
    enterStructureDeclaration(ctx) {
    }
    exitStructureDeclaration(ctx) {
    }
    enterStructureModifier(ctx) {
    }
    exitStructureModifier(ctx) {
    }
    enterStructMemberDeclaration(ctx) {
    }
    exitStructMemberDeclaration(ctx) {
    }
    enterModuleDeclaration(ctx) {
    }
    exitModuleDeclaration(ctx) {
    }
    enterModuleMemberDeclaration(ctx) {
    }
    exitModuleMemberDeclaration(ctx) {
    }
    enterInterfaceDeclaration(ctx) {
    }
    exitInterfaceDeclaration(ctx) {
    }
    enterInterfaceBase(ctx) {
    }
    exitInterfaceBase(ctx) {
    }
    enterInterfaceBases(ctx) {
    }
    exitInterfaceBases(ctx) {
    }
    enterInterfaceMemberDeclaration(ctx) {
    }
    exitInterfaceMemberDeclaration(ctx) {
    }
    enterArrayTypeName(ctx) {
    }
    exitArrayTypeName(ctx) {
    }
    enterArrayTypeModifiers(ctx) {
    }
    exitArrayTypeModifiers(ctx) {
    }
    enterArrayTypeModifier(ctx) {
    }
    exitArrayTypeModifier(ctx) {
    }
    enterRankList(ctx) {
    }
    exitRankList(ctx) {
    }
    enterArrayNameModifier(ctx) {
    }
    exitArrayNameModifier(ctx) {
    }
    enterDelegateDeclaration(ctx) {
    }
    exitDelegateDeclaration(ctx) {
    }
    enterMethodSignature(ctx) {
    }
    exitMethodSignature(ctx) {
    }
    enterImplementsClause(ctx) {
    }
    exitImplementsClause(ctx) {
    }
    enterImplementsList(ctx) {
    }
    exitImplementsList(ctx) {
    }
    enterInterfaceMemberSpecifier(ctx) {
    }
    exitInterfaceMemberSpecifier(ctx) {
    }
    enterMethodMemberDeclaration(ctx) {
    }
    exitMethodMemberDeclaration(ctx) {
    }
    enterInterfaceMethodMemberDeclaration(ctx) {
    }
    exitInterfaceMethodMemberDeclaration(ctx) {
    }
    enterMethodDeclaration(ctx) {
    }
    exitMethodDeclaration(ctx) {
    }
    enterInterfaceMethodDeclaration(ctx) {
    }
    exitInterfaceMethodDeclaration(ctx) {
    }
    enterSubSignature(ctx) {
    }
    exitSubSignature(ctx) {
    }
    enterFunctionSignature(ctx) {
    }
    exitFunctionSignature(ctx) {
    }
    enterSubDeclaration(ctx) {
    }
    exitSubDeclaration(ctx) {
    }
    enterMustOverrideSubDeclaration(ctx) {
    }
    exitMustOverrideSubDeclaration(ctx) {
    }
    enterInterfaceSubDeclaration(ctx) {
    }
    exitInterfaceSubDeclaration(ctx) {
    }
    enterFunctionDeclaration(ctx) {
    }
    exitFunctionDeclaration(ctx) {
    }
    enterMustOverrideFunctionDeclaration(ctx) {
    }
    exitMustOverrideFunctionDeclaration(ctx) {
    }
    enterInterfaceFunctionDeclaration(ctx) {
    }
    exitInterfaceFunctionDeclaration(ctx) {
    }
    enterProcedureModifier(ctx) {
    }
    exitProcedureModifier(ctx) {
    }
    enterMustOverrideProcedureModifier(ctx) {
    }
    exitMustOverrideProcedureModifier(ctx) {
    }
    enterInterfaceProcedureModifier(ctx) {
    }
    exitInterfaceProcedureModifier(ctx) {
    }
    enterHandlesOrImplements(ctx) {
    }
    exitHandlesOrImplements(ctx) {
    }
    enterExternalMethodDeclaration(ctx) {
    }
    exitExternalMethodDeclaration(ctx) {
    }
    enterExternalSubDeclaration(ctx) {
    }
    exitExternalSubDeclaration(ctx) {
    }
    enterExternalFunctionDeclaration(ctx) {
    }
    exitExternalFunctionDeclaration(ctx) {
    }
    enterExternalMethodModifier(ctx) {
    }
    exitExternalMethodModifier(ctx) {
    }
    enterCharsetModifier(ctx) {
    }
    exitCharsetModifier(ctx) {
    }
    enterLibraryClause(ctx) {
    }
    exitLibraryClause(ctx) {
    }
    enterAliasClause(ctx) {
    }
    exitAliasClause(ctx) {
    }
    enterParameterList(ctx) {
    }
    exitParameterList(ctx) {
    }
    enterParameter(ctx) {
    }
    exitParameter(ctx) {
    }
    enterParameterModifier(ctx) {
    }
    exitParameterModifier(ctx) {
    }
    enterParameterIdentifier(ctx) {
    }
    exitParameterIdentifier(ctx) {
    }
    enterHandlesClause(ctx) {
    }
    exitHandlesClause(ctx) {
    }
    enterEventHandlesList(ctx) {
    }
    exitEventHandlesList(ctx) {
    }
    enterEventMemberSpecifier(ctx) {
    }
    exitEventMemberSpecifier(ctx) {
    }
    enterConstructorMemberDeclaration(ctx) {
    }
    exitConstructorMemberDeclaration(ctx) {
    }
    enterConstructorModifier(ctx) {
    }
    exitConstructorModifier(ctx) {
    }
    enterEventMemberDeclaration(ctx) {
    }
    exitEventMemberDeclaration(ctx) {
    }
    enterRegularEventMemberDeclaration(ctx) {
    }
    exitRegularEventMemberDeclaration(ctx) {
    }
    enterInterfaceEventMemberDeclaration(ctx) {
    }
    exitInterfaceEventMemberDeclaration(ctx) {
    }
    enterParametersOrType(ctx) {
    }
    exitParametersOrType(ctx) {
    }
    enterEventModifiers(ctx) {
    }
    exitEventModifiers(ctx) {
    }
    enterInterfaceEventModifiers(ctx) {
    }
    exitInterfaceEventModifiers(ctx) {
    }
    enterCustomEventMemberDeclaration(ctx) {
    }
    exitCustomEventMemberDeclaration(ctx) {
    }
    enterEventAccessorDeclaration(ctx) {
    }
    exitEventAccessorDeclaration(ctx) {
    }
    enterAddHandlerDeclaration(ctx) {
    }
    exitAddHandlerDeclaration(ctx) {
    }
    enterRemoveHandlerDeclaration(ctx) {
    }
    exitRemoveHandlerDeclaration(ctx) {
    }
    enterRaiseEventDeclaration(ctx) {
    }
    exitRaiseEventDeclaration(ctx) {
    }
    enterConstantMemberDeclaration(ctx) {
    }
    exitConstantMemberDeclaration(ctx) {
    }
    enterConstantModifier(ctx) {
    }
    exitConstantModifier(ctx) {
    }
    enterConstantDeclarators(ctx) {
    }
    exitConstantDeclarators(ctx) {
    }
    enterConstantDeclarator(ctx) {
    }
    exitConstantDeclarator(ctx) {
    }
    enterVariableMemberDeclaration(ctx) {
    }
    exitVariableMemberDeclaration(ctx) {
    }
    enterVariableModifier(ctx) {
    }
    exitVariableModifier(ctx) {
    }
    enterVariableDeclarators(ctx) {
    }
    exitVariableDeclarators(ctx) {
    }
    enterVariableDeclarator(ctx) {
    }
    exitVariableDeclarator(ctx) {
    }
    enterVariableIdentifiers(ctx) {
    }
    exitVariableIdentifiers(ctx) {
    }
    enterVariableIdentifier(ctx) {
    }
    exitVariableIdentifier(ctx) {
    }
    enterArraySizeInitializationModifier(ctx) {
    }
    exitArraySizeInitializationModifier(ctx) {
    }
    enterBoundList(ctx) {
    }
    exitBoundList(ctx) {
    }
    enterBound(ctx) {
    }
    exitBound(ctx) {
    }
    enterPropertyMemberDeclaration(ctx) {
    }
    exitPropertyMemberDeclaration(ctx) {
    }
    enterPropertySignature(ctx) {
    }
    exitPropertySignature(ctx) {
    }
    enterRegularPropertyMemberDeclaration(ctx) {
    }
    exitRegularPropertyMemberDeclaration(ctx) {
    }
    enterMustOverridePropertyMemberDeclaration(ctx) {
    }
    exitMustOverridePropertyMemberDeclaration(ctx) {
    }
    enterAutoPropertyMemberDeclaration(ctx) {
    }
    exitAutoPropertyMemberDeclaration(ctx) {
    }
    enterInterfacePropertyMemberDeclaration(ctx) {
    }
    exitInterfacePropertyMemberDeclaration(ctx) {
    }
    enterAutoPropertyModifier(ctx) {
    }
    exitAutoPropertyModifier(ctx) {
    }
    enterPropertyModifier(ctx) {
    }
    exitPropertyModifier(ctx) {
    }
    enterMustOverridePropertyModifier(ctx) {
    }
    exitMustOverridePropertyModifier(ctx) {
    }
    enterInterfacePropertyModifier(ctx) {
    }
    exitInterfacePropertyModifier(ctx) {
    }
    enterPropertyAccessorDeclaration(ctx) {
    }
    exitPropertyAccessorDeclaration(ctx) {
    }
    enterPropertyGetDeclaration(ctx) {
    }
    exitPropertyGetDeclaration(ctx) {
    }
    enterPropertySetDeclaration(ctx) {
    }
    exitPropertySetDeclaration(ctx) {
    }
    enterOperatorDeclaration(ctx) {
    }
    exitOperatorDeclaration(ctx) {
    }
    enterOperatorModifier(ctx) {
    }
    exitOperatorModifier(ctx) {
    }
    enterOverloadableOperator(ctx) {
    }
    exitOverloadableOperator(ctx) {
    }
    enterStatement(ctx) {
    }
    exitStatement(ctx) {
    }
    enterBlock(ctx) {
    }
    exitBlock(ctx) {
    }
    enterLabelDeclarationStatement(ctx) {
    }
    exitLabelDeclarationStatement(ctx) {
    }
    enterLabelName(ctx) {
    }
    exitLabelName(ctx) {
    }
    enterStatements(ctx) {
    }
    exitStatements(ctx) {
    }
    enterLocalDeclarationStatement(ctx) {
    }
    exitLocalDeclarationStatement(ctx) {
    }
    enterLocalModifier(ctx) {
    }
    exitLocalModifier(ctx) {
    }
    enterWithStatement(ctx) {
    }
    exitWithStatement(ctx) {
    }
    enterSyncLockStatement(ctx) {
    }
    exitSyncLockStatement(ctx) {
    }
    enterEventStatement(ctx) {
    }
    exitEventStatement(ctx) {
    }
    enterRaiseEventStatement(ctx) {
    }
    exitRaiseEventStatement(ctx) {
    }
    enterAddHandlerStatement(ctx) {
    }
    exitAddHandlerStatement(ctx) {
    }
    enterRemoveHandlerStatement(ctx) {
    }
    exitRemoveHandlerStatement(ctx) {
    }
    enterAssignmentStatement(ctx) {
    }
    exitAssignmentStatement(ctx) {
    }
    enterRegularAssignmentStatement(ctx) {
    }
    exitRegularAssignmentStatement(ctx) {
    }
    enterCompoundAssignmentStatement(ctx) {
    }
    exitCompoundAssignmentStatement(ctx) {
    }
    enterCompoundBinaryOperator(ctx) {
    }
    exitCompoundBinaryOperator(ctx) {
    }
    enterMidAssignmentStatement(ctx) {
    }
    exitMidAssignmentStatement(ctx) {
    }
    enterInvocationStatement(ctx) {
    }
    exitInvocationStatement(ctx) {
    }
    enterConditionalStatement(ctx) {
    }
    exitConditionalStatement(ctx) {
    }
    enterIfStatement(ctx) {
    }
    exitIfStatement(ctx) {
    }
    enterBlockIfStatement(ctx) {
    }
    exitBlockIfStatement(ctx) {
    }
    enterElseIfStatement(ctx) {
    }
    exitElseIfStatement(ctx) {
    }
    enterElseStatement(ctx) {
    }
    exitElseStatement(ctx) {
    }
    enterLineIfThenStatement(ctx) {
    }
    exitLineIfThenStatement(ctx) {
    }
    enterSelectStatement(ctx) {
    }
    exitSelectStatement(ctx) {
    }
    enterCaseStatement(ctx) {
    }
    exitCaseStatement(ctx) {
    }
    enterCaseClauses(ctx) {
    }
    exitCaseClauses(ctx) {
    }
    enterCaseClause(ctx) {
    }
    exitCaseClause(ctx) {
    }
    enterComparisonOperator(ctx) {
    }
    exitComparisonOperator(ctx) {
    }
    enterCaseElseStatement(ctx) {
    }
    exitCaseElseStatement(ctx) {
    }
    enterLoopStatement(ctx) {
    }
    exitLoopStatement(ctx) {
    }
    enterWhileStatement(ctx) {
    }
    exitWhileStatement(ctx) {
    }
    enterDoLoopStatement(ctx) {
    }
    exitDoLoopStatement(ctx) {
    }
    enterDoTopLoopStatement(ctx) {
    }
    exitDoTopLoopStatement(ctx) {
    }
    enterDoBottomLoopStatement(ctx) {
    }
    exitDoBottomLoopStatement(ctx) {
    }
    enterWhileOrUntil(ctx) {
    }
    exitWhileOrUntil(ctx) {
    }
    enterForStatement(ctx) {
    }
    exitForStatement(ctx) {
    }
    enterLoopControlVariable(ctx) {
    }
    exitLoopControlVariable(ctx) {
    }
    enterNextExpressionList(ctx) {
    }
    exitNextExpressionList(ctx) {
    }
    enterForEachStatement(ctx) {
    }
    exitForEachStatement(ctx) {
    }
    enterErrorHandlingStatement(ctx) {
    }
    exitErrorHandlingStatement(ctx) {
    }
    enterStructuredErrorStatement(ctx) {
    }
    exitStructuredErrorStatement(ctx) {
    }
    enterTryStatement(ctx) {
    }
    exitTryStatement(ctx) {
    }
    enterFinallyStatement(ctx) {
    }
    exitFinallyStatement(ctx) {
    }
    enterCatchStatement(ctx) {
    }
    exitCatchStatement(ctx) {
    }
    enterThrowStatement(ctx) {
    }
    exitThrowStatement(ctx) {
    }
    enterUnstructuredErrorStatement(ctx) {
    }
    exitUnstructuredErrorStatement(ctx) {
    }
    enterErrorStatement(ctx) {
    }
    exitErrorStatement(ctx) {
    }
    enterOnErrorStatement(ctx) {
    }
    exitOnErrorStatement(ctx) {
    }
    enterErrorClause(ctx) {
    }
    exitErrorClause(ctx) {
    }
    enterResumeStatement(ctx) {
    }
    exitResumeStatement(ctx) {
    }
    enterResumeClause(ctx) {
    }
    exitResumeClause(ctx) {
    }
    enterBranchStatement(ctx) {
    }
    exitBranchStatement(ctx) {
    }
    enterGoToStatement(ctx) {
    }
    exitGoToStatement(ctx) {
    }
    enterExitStatement(ctx) {
    }
    exitExitStatement(ctx) {
    }
    enterExitKind(ctx) {
    }
    exitExitKind(ctx) {
    }
    enterContinueStatement(ctx) {
    }
    exitContinueStatement(ctx) {
    }
    enterContinueKind(ctx) {
    }
    exitContinueKind(ctx) {
    }
    enterStopStatement(ctx) {
    }
    exitStopStatement(ctx) {
    }
    enterEndStatement(ctx) {
    }
    exitEndStatement(ctx) {
    }
    enterReturnStatement(ctx) {
    }
    exitReturnStatement(ctx) {
    }
    enterArrayHandlingStatement(ctx) {
    }
    exitArrayHandlingStatement(ctx) {
    }
    enterRedimStatement(ctx) {
    }
    exitRedimStatement(ctx) {
    }
    enterRedimClauses(ctx) {
    }
    exitRedimClauses(ctx) {
    }
    enterRedimClause(ctx) {
    }
    exitRedimClause(ctx) {
    }
    enterEraseStatement(ctx) {
    }
    exitEraseStatement(ctx) {
    }
    enterEraseExpressions(ctx) {
    }
    exitEraseExpressions(ctx) {
    }
    enterUsingStatement(ctx) {
    }
    exitUsingStatement(ctx) {
    }
    enterUsingResources(ctx) {
    }
    exitUsingResources(ctx) {
    }
    enterAwaitStatement(ctx) {
    }
    exitAwaitStatement(ctx) {
    }
    enterYieldStatement(ctx) {
    }
    exitYieldStatement(ctx) {
    }
    enterExpression(ctx) {
    }
    exitExpression(ctx) {
    }
    enterMemberAccessBase(ctx) {
    }
    exitMemberAccessBase(ctx) {
    }
    enterConstantExpression(ctx) {
    }
    exitConstantExpression(ctx) {
    }
    enterSimpleExpression(ctx) {
    }
    exitSimpleExpression(ctx) {
    }
    enterLiteralExpression(ctx) {
    }
    exitLiteralExpression(ctx) {
    }
    enterParenthesizedExpression(ctx) {
    }
    exitParenthesizedExpression(ctx) {
    }
    enterInstanceExpression(ctx) {
    }
    exitInstanceExpression(ctx) {
    }
    enterSimpleNameExpression(ctx) {
    }
    exitSimpleNameExpression(ctx) {
    }
    enterAddressOfExpression(ctx) {
    }
    exitAddressOfExpression(ctx) {
    }
    enterGetTypeExpression(ctx) {
    }
    exitGetTypeExpression(ctx) {
    }
    enterGetTypeTypeName(ctx) {
    }
    exitGetTypeTypeName(ctx) {
    }
    enterQualifiedOpenTypeName(ctx) {
    }
    exitQualifiedOpenTypeName(ctx) {
    }
    enterTypeArityList(ctx) {
    }
    exitTypeArityList(ctx) {
    }
    enterCommaList(ctx) {
    }
    exitCommaList(ctx) {
    }
    enterTypeOfIsExpression(ctx) {
    }
    exitTypeOfIsExpression(ctx) {
    }
    enterGetXmlNamespaceExpression(ctx) {
    }
    exitGetXmlNamespaceExpression(ctx) {
    }
    enterArgumentList(ctx) {
    }
    exitArgumentList(ctx) {
    }
    enterPositionalArgumentList(ctx) {
    }
    exitPositionalArgumentList(ctx) {
    }
    enterNamedArgumentList(ctx) {
    }
    exitNamedArgumentList(ctx) {
    }
    enterNewExpression(ctx) {
    }
    exitNewExpression(ctx) {
    }
    enterObjectCreationExpression(ctx) {
    }
    exitObjectCreationExpression(ctx) {
    }
    enterObjectCreationExpressionInitializer(ctx) {
    }
    exitObjectCreationExpressionInitializer(ctx) {
    }
    enterObjectMemberInitializer(ctx) {
    }
    exitObjectMemberInitializer(ctx) {
    }
    enterFieldInitializerList(ctx) {
    }
    exitFieldInitializerList(ctx) {
    }
    enterFieldInitializer(ctx) {
    }
    exitFieldInitializer(ctx) {
    }
    enterObjectCollectionInitializer(ctx) {
    }
    exitObjectCollectionInitializer(ctx) {
    }
    enterCollectionInitializer(ctx) {
    }
    exitCollectionInitializer(ctx) {
    }
    enterCollectionElementList(ctx) {
    }
    exitCollectionElementList(ctx) {
    }
    enterCollectionElement(ctx) {
    }
    exitCollectionElement(ctx) {
    }
    enterArrayExpression(ctx) {
    }
    exitArrayExpression(ctx) {
    }
    enterArrayCreationExpression(ctx) {
    }
    exitArrayCreationExpression(ctx) {
    }
    enterArrayLiteralExpression(ctx) {
    }
    exitArrayLiteralExpression(ctx) {
    }
    enterAnonymousObjectCreationExpression(ctx) {
    }
    exitAnonymousObjectCreationExpression(ctx) {
    }
    enterCastExpression(ctx) {
    }
    exitCastExpression(ctx) {
    }
    enterCastTarget(ctx) {
    }
    exitCastTarget(ctx) {
    }
    enterUnaryPlusExpression(ctx) {
    }
    exitUnaryPlusExpression(ctx) {
    }
    enterUnaryMinusExpression(ctx) {
    }
    exitUnaryMinusExpression(ctx) {
    }
    enterBooleanExpression(ctx) {
    }
    exitBooleanExpression(ctx) {
    }
    enterLambdaExpression(ctx) {
    }
    exitLambdaExpression(ctx) {
    }
    enterSingleLineLambda(ctx) {
    }
    exitSingleLineLambda(ctx) {
    }
    enterMultiLineLambda(ctx) {
    }
    exitMultiLineLambda(ctx) {
    }
    enterMultiLineFunctionLambda(ctx) {
    }
    exitMultiLineFunctionLambda(ctx) {
    }
    enterMultiLineSubLambda(ctx) {
    }
    exitMultiLineSubLambda(ctx) {
    }
    enterLambdaModifier(ctx) {
    }
    exitLambdaModifier(ctx) {
    }
    enterQueryExpression(ctx) {
    }
    exitQueryExpression(ctx) {
    }
    enterFromOrAggregateQueryOperator(ctx) {
    }
    exitFromOrAggregateQueryOperator(ctx) {
    }
    enterQueryOperator(ctx) {
    }
    exitQueryOperator(ctx) {
    }
    enterJoinOrGroupJoinQueryOperator(ctx) {
    }
    exitJoinOrGroupJoinQueryOperator(ctx) {
    }
    enterCollectionRangeVariableDeclarationList(ctx) {
    }
    exitCollectionRangeVariableDeclarationList(ctx) {
    }
    enterCollectionRangeVariableDeclaration(ctx) {
    }
    exitCollectionRangeVariableDeclaration(ctx) {
    }
    enterExpressionRangeVariableDeclarationList(ctx) {
    }
    exitExpressionRangeVariableDeclarationList(ctx) {
    }
    enterExpressionRangeVariableDeclaration(ctx) {
    }
    exitExpressionRangeVariableDeclaration(ctx) {
    }
    enterFromQueryOperator(ctx) {
    }
    exitFromQueryOperator(ctx) {
    }
    enterJoinQueryOperator(ctx) {
    }
    exitJoinQueryOperator(ctx) {
    }
    enterJoinConditionList(ctx) {
    }
    exitJoinConditionList(ctx) {
    }
    enterJoinCondition(ctx) {
    }
    exitJoinCondition(ctx) {
    }
    enterLetQueryOperator(ctx) {
    }
    exitLetQueryOperator(ctx) {
    }
    enterSelectQueryOperator(ctx) {
    }
    exitSelectQueryOperator(ctx) {
    }
    enterDistinctQueryOperator(ctx) {
    }
    exitDistinctQueryOperator(ctx) {
    }
    enterWhereQueryOperator(ctx) {
    }
    exitWhereQueryOperator(ctx) {
    }
    enterPartitionQueryOperator(ctx) {
    }
    exitPartitionQueryOperator(ctx) {
    }
    enterOrderByQueryOperator(ctx) {
    }
    exitOrderByQueryOperator(ctx) {
    }
    enterOrderExpressionList(ctx) {
    }
    exitOrderExpressionList(ctx) {
    }
    enterOrderExpression(ctx) {
    }
    exitOrderExpression(ctx) {
    }
    enterOrdering(ctx) {
    }
    exitOrdering(ctx) {
    }
    enterGroupByQueryOperator(ctx) {
    }
    exitGroupByQueryOperator(ctx) {
    }
    enterAggregateQueryOperator(ctx) {
    }
    exitAggregateQueryOperator(ctx) {
    }
    enterGroupJoinQueryOperator(ctx) {
    }
    exitGroupJoinQueryOperator(ctx) {
    }
    enterConditionalExpression(ctx) {
    }
    exitConditionalExpression(ctx) {
    }
    enterXMLLiteralExpression(ctx) {
    }
    exitXMLLiteralExpression(ctx) {
    }
    enterXMLEmbeddedExpression(ctx) {
    }
    exitXMLEmbeddedExpression(ctx) {
    }
    enterXMLDocument(ctx) {
    }
    exitXMLDocument(ctx) {
    }
    enterXMLDocumentPrologue(ctx) {
    }
    exitXMLDocumentPrologue(ctx) {
    }
    enterXMLVersion(ctx) {
    }
    exitXMLVersion(ctx) {
    }
    enterXMLVersionNumberValue(ctx) {
    }
    exitXMLVersionNumberValue(ctx) {
    }
    enterXMLEncoding(ctx) {
    }
    exitXMLEncoding(ctx) {
    }
    enterXMLEncodingNameValue(ctx) {
    }
    exitXMLEncodingNameValue(ctx) {
    }
    enterXMLEncodingName(ctx) {
    }
    exitXMLEncodingName(ctx) {
    }
    enterXMLEncodingNameCharacter(ctx) {
    }
    exitXMLEncodingNameCharacter(ctx) {
    }
    enterXMLStandalone(ctx) {
    }
    exitXMLStandalone(ctx) {
    }
    enterXMLYesNoValue(ctx) {
    }
    exitXMLYesNoValue(ctx) {
    }
    enterXMLYesNo(ctx) {
    }
    exitXMLYesNo(ctx) {
    }
    enterXMLMisc(ctx) {
    }
    exitXMLMisc(ctx) {
    }
    enterXMLDocumentBody(ctx) {
    }
    exitXMLDocumentBody(ctx) {
    }
    enterXMLElement(ctx) {
    }
    exitXMLElement(ctx) {
    }
    enterXMLEmptyElement(ctx) {
    }
    exitXMLEmptyElement(ctx) {
    }
    enterXMLElementStart(ctx) {
    }
    exitXMLElementStart(ctx) {
    }
    enterXMLElementEnd(ctx) {
    }
    exitXMLElementEnd(ctx) {
    }
    enterXMLContent(ctx) {
    }
    exitXMLContent(ctx) {
    }
    enterXMLNestedContent(ctx) {
    }
    exitXMLNestedContent(ctx) {
    }
    enterXMLAttribute(ctx) {
    }
    exitXMLAttribute(ctx) {
    }
    enterXMLAttributeName(ctx) {
    }
    exitXMLAttributeName(ctx) {
    }
    enterXMLAttributeValue(ctx) {
    }
    exitXMLAttributeValue(ctx) {
    }
    enterXMLAttributeDoubleQuoteValueCharacter(ctx) {
    }
    exitXMLAttributeDoubleQuoteValueCharacter(ctx) {
    }
    enterXMLAttributeSingleQuoteValueCharacter(ctx) {
    }
    exitXMLAttributeSingleQuoteValueCharacter(ctx) {
    }
    enterXMLReference(ctx) {
    }
    exitXMLReference(ctx) {
    }
    enterXMLEntityReference(ctx) {
    }
    exitXMLEntityReference(ctx) {
    }
    enterXMLCharacterReference(ctx) {
    }
    exitXMLCharacterReference(ctx) {
    }
    enterXMLNamespaceAttributeName(ctx) {
    }
    exitXMLNamespaceAttributeName(ctx) {
    }
    enterXMLPrefixedNamespaceAttributeName(ctx) {
    }
    exitXMLPrefixedNamespaceAttributeName(ctx) {
    }
    enterXMLDefaultNamespaceAttributeName(ctx) {
    }
    exitXMLDefaultNamespaceAttributeName(ctx) {
    }
    enterXMLNamespaceName(ctx) {
    }
    exitXMLNamespaceName(ctx) {
    }
    enterXMLNamespaceNameCharacter(ctx) {
    }
    exitXMLNamespaceNameCharacter(ctx) {
    }
    enterXMLQualifiedNameOrExpression(ctx) {
    }
    exitXMLQualifiedNameOrExpression(ctx) {
    }
    enterXMLQualifiedName(ctx) {
    }
    exitXMLQualifiedName(ctx) {
    }
    enterXMLPrefixedName(ctx) {
    }
    exitXMLPrefixedName(ctx) {
    }
    enterXMLUnprefixedName(ctx) {
    }
    exitXMLUnprefixedName(ctx) {
    }
    enterXMLProcessingInstruction(ctx) {
    }
    exitXMLProcessingInstruction(ctx) {
    }
    enterXMLComment(ctx) {
    }
    exitXMLComment(ctx) {
    }
    enterXMLCDATASection(ctx) {
    }
    exitXMLCDATASection(ctx) {
    }
    enterAwaitOperatorExpression(ctx) {
    }
    exitAwaitOperatorExpression(ctx) {
    }
}
exports.default = vbParserListener;
