// @harvest-forward/runtime/dist/src/plugins/information-architecture/oo/Class.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Class = void 0;
const Permanence_1 = require("../../../system/Permanence");
const SystemComponentComparator_1 = require("../../../system/SystemComponentComparator");
const ObjectType_1 = require("../information/ObjectType");
const Field_1 = require("./Field");
const Method_1 = require("./Method");
const Property_1 = require("./Property");
class Class extends ObjectType_1.ObjectType {
    constructor(nameSpace, name, description, role) {
        super('Class', nameSpace, name, description, role);
        this.isClass = true;
        this.updateable = true;
        this.permanence = Permanence_1.Permanence.transient;
        this._fields = {};
        this._properties = {};
        this._methods = {};
    }
    get isObject() {
        return !this.isExternal;
    }
    get identifiable() {
        return this.identifyingProperty != null;
    }
    addChild(child) {
        super.addChild(child);
        if (child instanceof Property_1.Property) {
            this._properties[child.name] = child;
        }
        else if (child instanceof Method_1.Method) {
            this._methods[child.name] = child;
        }
        else if (child instanceof Field_1.Field) {
            this._fields[child.name] = child;
        }
    }
    removeChild(child) {
        if (child instanceof Property_1.Property) {
            delete this._properties[child.name];
        }
        else if (child instanceof Method_1.Method) {
            delete this._methods[child.name];
        }
        else if (child instanceof Field_1.Field) {
            delete this._fields[child.name];
        }
        return super.removeChild(child);
    }
    get fieldsMap() {
        return this._fields;
    }
    get fields() {
        return Object.values(this._fields).sort(SystemComponentComparator_1.SystemComponentComparator);
    }
    get propertiesMap() {
        return this._properties;
    }
    get properties() {
        return Object.values(this._properties).sort(SystemComponentComparator_1.SystemComponentComparator);
    }
    get methodsMap() {
        return this._methods;
    }
    get methods() {
        return Object.values(this._methods).sort(SystemComponentComparator_1.SystemComponentComparator);
    }
    get classReferences() {
        return this.allReferences.filter((item) => item.objectTypeName === 'Class');
    }
}
exports.Class = Class;
